{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import importlib, sys, os\n",
    "\n",
    "sys.path.append(r'C:\\Users\\owysocky\\Documents\\GitHub\\CCE_scGeneRAI\\source\\investigate_and_compare')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Define gene list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['KRAS', 'GATA3', 'CDH1', 'MYC', 'BRAF', 'ATM', 'TP53', 'APC', 'PTEN', 'RB1']\n"
     ]
    }
   ],
   "source": [
    "gene_list =  ['KRAS', 'GATA3', 'CDH1', 'MYC', 'BRAF', 'ATM', 'TP53', 'APC', 'PTEN', 'RB1']\n",
    "print(gene_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene Enrichment using gProfiler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "source",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "native",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "name",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "p_value",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "significant",
         "rawType": "bool",
         "type": "boolean"
        },
        {
         "name": "description",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "term_size",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "query_size",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "intersection_size",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "effective_domain_size",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "precision",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "recall",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "query",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "parents",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "61535a08-ad10-4b65-ad63-adfbab9202e1",
       "rows": [
        [
         "0",
         "KEGG",
         "KEGG:05219",
         "Bladder cancer",
         "1.1046729596474314e-10",
         "True",
         "Bladder cancer",
         "41",
         "10",
         "6",
         "8484",
         "0.6",
         "0.14634146341463414",
         "query_1",
         "['KEGG:00000']"
        ],
        [
         "1",
         "WP",
         "WP:WP2828",
         "Bladder cancer",
         "1.7189477547051771e-10",
         "True",
         "Bladder cancer",
         "40",
         "10",
         "6",
         "8722",
         "0.6",
         "0.15",
         "query_1",
         "['WP:000000']"
        ],
        [
         "2",
         "HP",
         "HP:0006740",
         "Transitional cell carcinoma of the bladder",
         "8.404839328090956e-09",
         "True",
         "The presence of a carcinoma of the urinary bladder with origin in a transitional epithelial cell.",
         "31",
         "10",
         "6",
         "5080",
         "0.6",
         "0.1935483870967742",
         "query_1",
         "['HP:0002862']"
        ],
        [
         "3",
         "HP",
         "HP:0002862",
         "Bladder carcinoma",
         "1.2626027342811956e-08",
         "True",
         "The presence of a carcinoma of the urinary bladder.",
         "33",
         "10",
         "6",
         "5080",
         "0.6",
         "0.18181818181818182",
         "query_1",
         "['HP:0009725']"
        ],
        [
         "4",
         "HP",
         "HP:0009725",
         "Bladder neoplasm",
         "1.532120038838035e-08",
         "True",
         "The presence of a neoplasm of the urinary bladder.",
         "34",
         "10",
         "6",
         "5080",
         "0.6",
         "0.17647058823529413",
         "query_1",
         "['HP:0010786']"
        ],
        [
         "5",
         "KEGG",
         "KEGG:05216",
         "Thyroid cancer",
         "1.813203326614217e-08",
         "True",
         "Thyroid cancer",
         "37",
         "10",
         "5",
         "8484",
         "0.5",
         "0.13513513513513514",
         "query_1",
         "['KEGG:00000']"
        ],
        [
         "6",
         "WP",
         "WP:WP1971",
         "Integrated cancer pathway",
         "8.434361666637605e-08",
         "True",
         "Integrated cancer pathway",
         "44",
         "10",
         "5",
         "8722",
         "0.5",
         "0.11363636363636363",
         "query_1",
         "['WP:000000']"
        ],
        [
         "7",
         "HP",
         "HP:0012126",
         "Stomach cancer",
         "2.733133699539587e-06",
         "True",
         "A cancer arising in any part of the stomach.",
         "33",
         "10",
         "5",
         "5080",
         "0.5",
         "0.15151515151515152",
         "query_1",
         "['HP:0006753']"
        ],
        [
         "8",
         "HP",
         "HP:0100743",
         "Neoplasm of the rectum",
         "1.39309695624091e-05",
         "True",
         "Neoplasm of the rectum",
         "45",
         "10",
         "5",
         "5080",
         "0.5",
         "0.1111111111111111",
         "query_1",
         "['HP:0002034', 'HP:0100834']"
        ],
        [
         "9",
         "HP",
         "HP:0009733",
         "Glioma",
         "1.947601609409882e-05",
         "True",
         "The presence of a glioma, which is a neoplasm of the central nervous system originating from a glial cell (astrocytes or oligodendrocytes).",
         "48",
         "10",
         "5",
         "5080",
         "0.5",
         "0.10416666666666667",
         "query_1",
         "['HP:0030063', 'HP:0100705', 'HP:0100836']"
        ],
        [
         "10",
         "HP",
         "HP:0012174",
         "Glioblastoma multiforme",
         "0.00010394838095811023",
         "True",
         "A tumor arising from glia in the central nervous system with macroscopic regions of necrosis and hemorrhage. Microscopically, glioblastoma multiforme is characterized by regions of pseudopalisading necrosis, pleomorphic nuclei and cells, and microvascular proliferation.",
         "24",
         "10",
         "4",
         "5080",
         "0.4",
         "0.16666666666666666",
         "query_1",
         "['HP:0009733']"
        ],
        [
         "11",
         "HP",
         "HP:0200008",
         "Intestinal polyposis",
         "0.0018605712011670209",
         "True",
         "The presence of multiple polyps in the intestine.",
         "48",
         "10",
         "4",
         "5080",
         "0.4",
         "0.08333333333333333",
         "query_1",
         "['HP:0005266']"
        ],
        [
         "12",
         "WP",
         "WP:WP2815",
         "Mammary gland development involution stage 4 of 4",
         "1.7112534089430503e-05",
         "True",
         "Mammary gland development involution stage 4 of 4",
         "10",
         "10",
         "3",
         "8722",
         "0.3",
         "0.3",
         "query_1",
         "['WP:000000']"
        ],
        [
         "13",
         "WP",
         "WP:WP1742",
         "TP53 network",
         "0.0001374362462010647",
         "True",
         "TP53 network",
         "19",
         "10",
         "3",
         "8722",
         "0.3",
         "0.15789473684210525",
         "query_1",
         "['WP:000000']"
        ],
        [
         "14",
         "GO:BP",
         "GO:0010165",
         "response to X-ray",
         "0.0003274388973378734",
         "True",
         "\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of X-ray radiation. An X-ray is a form of electromagnetic radiation with a wavelength in the range of 10 nanometers to 100 picometers (corresponding to frequencies in the range 30 PHz to 3 EHz).\" [GOC:sm, Wikipedia:X-ray]",
         "29",
         "10",
         "3",
         "21017",
         "0.3",
         "0.10344827586206896",
         "query_1",
         "['GO:0010212']"
        ],
        [
         "15",
         "WP",
         "WP:WP5385",
         "Growth factors and hormones in beta cell proliferation",
         "0.0003672129506481551",
         "True",
         "Growth factors and hormones in beta cell proliferation",
         "26",
         "10",
         "3",
         "8722",
         "0.3",
         "0.11538461538461539",
         "query_1",
         "['WP:000000']"
        ],
        [
         "16",
         "WP",
         "WP:WP3982",
         "miRNA regulation of p53 pathway in prostate cancer",
         "0.0006329432069965097",
         "True",
         "miRNA regulation of p53 pathway in prostate cancer",
         "31",
         "10",
         "3",
         "8722",
         "0.3",
         "0.0967741935483871",
         "query_1",
         "['WP:000000']"
        ],
        [
         "17",
         "GO:BP",
         "GO:0060251",
         "regulation of glial cell proliferation",
         "0.0007542599304691131",
         "True",
         "\"Any process that modulates the frequency, rate or extent of glial cell proliferation.\" [GOC:dph, GOC:tb]",
         "38",
         "10",
         "3",
         "21017",
         "0.3",
         "0.07894736842105263",
         "query_1",
         "['GO:0014009', 'GO:0042127']"
        ],
        [
         "18",
         "WP",
         "WP:WP5215",
         "Targeted agents in triple negative breast cancer",
         "0.0011828716455459269",
         "True",
         "Targeted agents in triple negative breast cancer",
         "38",
         "10",
         "3",
         "8722",
         "0.3",
         "0.07894736842105263",
         "query_1",
         "['WP:000000']"
        ],
        [
         "19",
         "REAC",
         "REAC:R-HSA-5689896",
         "Ovarian tumor domain proteases",
         "0.0013969513165221496",
         "True",
         "Ovarian tumor domain proteases",
         "39",
         "10",
         "3",
         "10916",
         "0.3",
         "0.07692307692307693",
         "query_1",
         "['REAC:R-HSA-5688426']"
        ],
        [
         "20",
         "HP",
         "HP:0010816",
         "Epidermal nevus",
         "0.0015760195530852808",
         "True",
         "Epidermal naevi are due to an overgrowth of the epidermis and may be present at birth (50%) or develop during childhood.",
         "12",
         "10",
         "3",
         "5080",
         "0.3",
         "0.25",
         "query_1",
         "['HP:0003764']"
        ],
        [
         "21",
         "WP",
         "WP:WP5300",
         "TROP2 regulatory signaling",
         "0.0018503236558656652",
         "True",
         "TROP2 regulatory signaling",
         "44",
         "10",
         "3",
         "8722",
         "0.3",
         "0.06818181818181818",
         "query_1",
         "['WP:000000']"
        ],
        [
         "22",
         "WP",
         "WP:WP4928",
         "MAPK pathway in congenital thyroid cancer",
         "7.957027277320243e-05",
         "True",
         "MAPK pathway in congenital thyroid cancer",
         "16",
         "10",
         "3",
         "8722",
         "0.3",
         "0.1875",
         "query_1",
         "['WP:000000']"
        ]
       ],
       "shape": {
        "columns": 14,
        "rows": 23
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:05219</td>\n",
       "      <td>Bladder cancer</td>\n",
       "      <td>1.104673e-10</td>\n",
       "      <td>True</td>\n",
       "      <td>Bladder cancer</td>\n",
       "      <td>41</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>8484</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.146341</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP2828</td>\n",
       "      <td>Bladder cancer</td>\n",
       "      <td>1.718948e-10</td>\n",
       "      <td>True</td>\n",
       "      <td>Bladder cancer</td>\n",
       "      <td>40</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0006740</td>\n",
       "      <td>Transitional cell carcinoma of the bladder</td>\n",
       "      <td>8.404839e-09</td>\n",
       "      <td>True</td>\n",
       "      <td>The presence of a carcinoma of the urinary bla...</td>\n",
       "      <td>31</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.193548</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0002862]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0002862</td>\n",
       "      <td>Bladder carcinoma</td>\n",
       "      <td>1.262603e-08</td>\n",
       "      <td>True</td>\n",
       "      <td>The presence of a carcinoma of the urinary bla...</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0009725]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0009725</td>\n",
       "      <td>Bladder neoplasm</td>\n",
       "      <td>1.532120e-08</td>\n",
       "      <td>True</td>\n",
       "      <td>The presence of a neoplasm of the urinary blad...</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.176471</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0010786]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:05216</td>\n",
       "      <td>Thyroid cancer</td>\n",
       "      <td>1.813203e-08</td>\n",
       "      <td>True</td>\n",
       "      <td>Thyroid cancer</td>\n",
       "      <td>37</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>8484</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.135135</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP1971</td>\n",
       "      <td>Integrated cancer pathway</td>\n",
       "      <td>8.434362e-08</td>\n",
       "      <td>True</td>\n",
       "      <td>Integrated cancer pathway</td>\n",
       "      <td>44</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.113636</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0012126</td>\n",
       "      <td>Stomach cancer</td>\n",
       "      <td>2.733134e-06</td>\n",
       "      <td>True</td>\n",
       "      <td>A cancer arising in any part of the stomach.</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.151515</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0006753]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0100743</td>\n",
       "      <td>Neoplasm of the rectum</td>\n",
       "      <td>1.393097e-05</td>\n",
       "      <td>True</td>\n",
       "      <td>Neoplasm of the rectum</td>\n",
       "      <td>45</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.111111</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0002034, HP:0100834]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0009733</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>1.947602e-05</td>\n",
       "      <td>True</td>\n",
       "      <td>The presence of a glioma, which is a neoplasm ...</td>\n",
       "      <td>48</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.104167</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0030063, HP:0100705, HP:0100836]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0012174</td>\n",
       "      <td>Glioblastoma multiforme</td>\n",
       "      <td>1.039484e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>A tumor arising from glia in the central nervo...</td>\n",
       "      <td>24</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.166667</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0009733]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0200008</td>\n",
       "      <td>Intestinal polyposis</td>\n",
       "      <td>1.860571e-03</td>\n",
       "      <td>True</td>\n",
       "      <td>The presence of multiple polyps in the intestine.</td>\n",
       "      <td>48</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0005266]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP2815</td>\n",
       "      <td>Mammary gland development involution stage 4 of 4</td>\n",
       "      <td>1.711253e-05</td>\n",
       "      <td>True</td>\n",
       "      <td>Mammary gland development involution stage 4 of 4</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP1742</td>\n",
       "      <td>TP53 network</td>\n",
       "      <td>1.374362e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>TP53 network</td>\n",
       "      <td>19</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.157895</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GO:BP</td>\n",
       "      <td>GO:0010165</td>\n",
       "      <td>response to X-ray</td>\n",
       "      <td>3.274389e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Any process that results in a change in state...</td>\n",
       "      <td>29</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>21017</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.103448</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0010212]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP5385</td>\n",
       "      <td>Growth factors and hormones in beta cell proli...</td>\n",
       "      <td>3.672130e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>Growth factors and hormones in beta cell proli...</td>\n",
       "      <td>26</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.115385</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP3982</td>\n",
       "      <td>miRNA regulation of p53 pathway in prostate ca...</td>\n",
       "      <td>6.329432e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>miRNA regulation of p53 pathway in prostate ca...</td>\n",
       "      <td>31</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.096774</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GO:BP</td>\n",
       "      <td>GO:0060251</td>\n",
       "      <td>regulation of glial cell proliferation</td>\n",
       "      <td>7.542599e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Any process that modulates the frequency, rat...</td>\n",
       "      <td>38</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>21017</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.078947</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0014009, GO:0042127]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP5215</td>\n",
       "      <td>Targeted agents in triple negative breast cancer</td>\n",
       "      <td>1.182872e-03</td>\n",
       "      <td>True</td>\n",
       "      <td>Targeted agents in triple negative breast cancer</td>\n",
       "      <td>38</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.078947</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>REAC</td>\n",
       "      <td>REAC:R-HSA-5689896</td>\n",
       "      <td>Ovarian tumor domain proteases</td>\n",
       "      <td>1.396951e-03</td>\n",
       "      <td>True</td>\n",
       "      <td>Ovarian tumor domain proteases</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>10916</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[REAC:R-HSA-5688426]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HP</td>\n",
       "      <td>HP:0010816</td>\n",
       "      <td>Epidermal nevus</td>\n",
       "      <td>1.576020e-03</td>\n",
       "      <td>True</td>\n",
       "      <td>Epidermal naevi are due to an overgrowth of th...</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>5080</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HP:0003764]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP5300</td>\n",
       "      <td>TROP2 regulatory signaling</td>\n",
       "      <td>1.850324e-03</td>\n",
       "      <td>True</td>\n",
       "      <td>TROP2 regulatory signaling</td>\n",
       "      <td>44</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.068182</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP4928</td>\n",
       "      <td>MAPK pathway in congenital thyroid cancer</td>\n",
       "      <td>7.957027e-05</td>\n",
       "      <td>True</td>\n",
       "      <td>MAPK pathway in congenital thyroid cancer</td>\n",
       "      <td>16</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>8722</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.187500</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   source              native  \\\n",
       "0    KEGG          KEGG:05219   \n",
       "1      WP           WP:WP2828   \n",
       "2      HP          HP:0006740   \n",
       "3      HP          HP:0002862   \n",
       "4      HP          HP:0009725   \n",
       "5    KEGG          KEGG:05216   \n",
       "6      WP           WP:WP1971   \n",
       "7      HP          HP:0012126   \n",
       "8      HP          HP:0100743   \n",
       "9      HP          HP:0009733   \n",
       "10     HP          HP:0012174   \n",
       "11     HP          HP:0200008   \n",
       "12     WP           WP:WP2815   \n",
       "13     WP           WP:WP1742   \n",
       "14  GO:BP          GO:0010165   \n",
       "15     WP           WP:WP5385   \n",
       "16     WP           WP:WP3982   \n",
       "17  GO:BP          GO:0060251   \n",
       "18     WP           WP:WP5215   \n",
       "19   REAC  REAC:R-HSA-5689896   \n",
       "20     HP          HP:0010816   \n",
       "21     WP           WP:WP5300   \n",
       "22     WP           WP:WP4928   \n",
       "\n",
       "                                                 name       p_value  \\\n",
       "0                                      Bladder cancer  1.104673e-10   \n",
       "1                                      Bladder cancer  1.718948e-10   \n",
       "2          Transitional cell carcinoma of the bladder  8.404839e-09   \n",
       "3                                   Bladder carcinoma  1.262603e-08   \n",
       "4                                    Bladder neoplasm  1.532120e-08   \n",
       "5                                      Thyroid cancer  1.813203e-08   \n",
       "6                           Integrated cancer pathway  8.434362e-08   \n",
       "7                                      Stomach cancer  2.733134e-06   \n",
       "8                              Neoplasm of the rectum  1.393097e-05   \n",
       "9                                              Glioma  1.947602e-05   \n",
       "10                            Glioblastoma multiforme  1.039484e-04   \n",
       "11                               Intestinal polyposis  1.860571e-03   \n",
       "12  Mammary gland development involution stage 4 of 4  1.711253e-05   \n",
       "13                                       TP53 network  1.374362e-04   \n",
       "14                                  response to X-ray  3.274389e-04   \n",
       "15  Growth factors and hormones in beta cell proli...  3.672130e-04   \n",
       "16  miRNA regulation of p53 pathway in prostate ca...  6.329432e-04   \n",
       "17             regulation of glial cell proliferation  7.542599e-04   \n",
       "18   Targeted agents in triple negative breast cancer  1.182872e-03   \n",
       "19                     Ovarian tumor domain proteases  1.396951e-03   \n",
       "20                                    Epidermal nevus  1.576020e-03   \n",
       "21                         TROP2 regulatory signaling  1.850324e-03   \n",
       "22          MAPK pathway in congenital thyroid cancer  7.957027e-05   \n",
       "\n",
       "    significant                                        description  term_size  \\\n",
       "0          True                                     Bladder cancer         41   \n",
       "1          True                                     Bladder cancer         40   \n",
       "2          True  The presence of a carcinoma of the urinary bla...         31   \n",
       "3          True  The presence of a carcinoma of the urinary bla...         33   \n",
       "4          True  The presence of a neoplasm of the urinary blad...         34   \n",
       "5          True                                     Thyroid cancer         37   \n",
       "6          True                          Integrated cancer pathway         44   \n",
       "7          True       A cancer arising in any part of the stomach.         33   \n",
       "8          True                             Neoplasm of the rectum         45   \n",
       "9          True  The presence of a glioma, which is a neoplasm ...         48   \n",
       "10         True  A tumor arising from glia in the central nervo...         24   \n",
       "11         True  The presence of multiple polyps in the intestine.         48   \n",
       "12         True  Mammary gland development involution stage 4 of 4         10   \n",
       "13         True                                       TP53 network         19   \n",
       "14         True  \"Any process that results in a change in state...         29   \n",
       "15         True  Growth factors and hormones in beta cell proli...         26   \n",
       "16         True  miRNA regulation of p53 pathway in prostate ca...         31   \n",
       "17         True  \"Any process that modulates the frequency, rat...         38   \n",
       "18         True   Targeted agents in triple negative breast cancer         38   \n",
       "19         True                     Ovarian tumor domain proteases         39   \n",
       "20         True  Epidermal naevi are due to an overgrowth of th...         12   \n",
       "21         True                         TROP2 regulatory signaling         44   \n",
       "22         True          MAPK pathway in congenital thyroid cancer         16   \n",
       "\n",
       "    query_size  intersection_size  effective_domain_size  precision    recall  \\\n",
       "0           10                  6                   8484        0.6  0.146341   \n",
       "1           10                  6                   8722        0.6  0.150000   \n",
       "2           10                  6                   5080        0.6  0.193548   \n",
       "3           10                  6                   5080        0.6  0.181818   \n",
       "4           10                  6                   5080        0.6  0.176471   \n",
       "5           10                  5                   8484        0.5  0.135135   \n",
       "6           10                  5                   8722        0.5  0.113636   \n",
       "7           10                  5                   5080        0.5  0.151515   \n",
       "8           10                  5                   5080        0.5  0.111111   \n",
       "9           10                  5                   5080        0.5  0.104167   \n",
       "10          10                  4                   5080        0.4  0.166667   \n",
       "11          10                  4                   5080        0.4  0.083333   \n",
       "12          10                  3                   8722        0.3  0.300000   \n",
       "13          10                  3                   8722        0.3  0.157895   \n",
       "14          10                  3                  21017        0.3  0.103448   \n",
       "15          10                  3                   8722        0.3  0.115385   \n",
       "16          10                  3                   8722        0.3  0.096774   \n",
       "17          10                  3                  21017        0.3  0.078947   \n",
       "18          10                  3                   8722        0.3  0.078947   \n",
       "19          10                  3                  10916        0.3  0.076923   \n",
       "20          10                  3                   5080        0.3  0.250000   \n",
       "21          10                  3                   8722        0.3  0.068182   \n",
       "22          10                  3                   8722        0.3  0.187500   \n",
       "\n",
       "      query                               parents  \n",
       "0   query_1                          [KEGG:00000]  \n",
       "1   query_1                           [WP:000000]  \n",
       "2   query_1                          [HP:0002862]  \n",
       "3   query_1                          [HP:0009725]  \n",
       "4   query_1                          [HP:0010786]  \n",
       "5   query_1                          [KEGG:00000]  \n",
       "6   query_1                           [WP:000000]  \n",
       "7   query_1                          [HP:0006753]  \n",
       "8   query_1              [HP:0002034, HP:0100834]  \n",
       "9   query_1  [HP:0030063, HP:0100705, HP:0100836]  \n",
       "10  query_1                          [HP:0009733]  \n",
       "11  query_1                          [HP:0005266]  \n",
       "12  query_1                           [WP:000000]  \n",
       "13  query_1                           [WP:000000]  \n",
       "14  query_1                          [GO:0010212]  \n",
       "15  query_1                           [WP:000000]  \n",
       "16  query_1                           [WP:000000]  \n",
       "17  query_1              [GO:0014009, GO:0042127]  \n",
       "18  query_1                           [WP:000000]  \n",
       "19  query_1                  [REAC:R-HSA-5688426]  \n",
       "20  query_1                          [HP:0003764]  \n",
       "21  query_1                           [WP:000000]  \n",
       "22  query_1                           [WP:000000]  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import gprofiler_code as ge\n",
    "importlib.reload(ge)\n",
    "\n",
    "ge_analyser = ge.GE_Analyser(gene_list)\n",
    "ge_results = ge_analyser.run_GE_on_nodes(user_threshold=0.002)\n",
    "ge_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Based on Gene Enrichment analysis, the entities:\n",
      "['KRAS', 'GATA3', 'CDH1', 'MYC', 'BRAF', 'ATM', 'TP53', 'APC', 'PTEN', 'RB1']\n",
      "the following processes or pathways were found to be enriched:\n",
      "\n",
      "-- Bladder cancer\n",
      "-- Bladder cancer\n",
      "-- Transitional cell carcinoma of the bladder; described as The presence of a carcinoma of the urinary bladder with origin in a transitional epithelial cell.\n",
      "-- Bladder carcinoma; described as The presence of a carcinoma of the urinary bladder.\n",
      "-- Bladder neoplasm; described as The presence of a neoplasm of the urinary bladder.\n",
      "-- Thyroid cancer\n",
      "-- Integrated cancer pathway\n",
      "-- Stomach cancer; described as A cancer arising in any part of the stomach.\n",
      "-- Neoplasm of the rectum\n",
      "-- Glioma; described as The presence of a glioma, which is a neoplasm of the central nervous system originating from a glial cell (astrocytes or oligodendrocytes).\n",
      "-- Glioblastoma multiforme; described as A tumor arising from glia in the central nervous system with macroscopic regions of necrosis and hemorrhage. Microscopically, glioblastoma multiforme is characterized by regions of pseudopalisading necrosis, pleomorphic nuclei and cells, and microvascular proliferation.\n",
      "-- Intestinal polyposis; described as The presence of multiple polyps in the intestine.\n",
      "-- Mammary gland development involution stage 4 of 4\n",
      "-- TP53 network\n",
      "-- response to X-ray; described as \"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of X-ray radiation. An X-ray is a form of electromagnetic radiation with a wavelength in the range of 10 nanometers to 100 picometers (corresponding to frequencies in the range 30 PHz to 3 EHz).\" [GOC:sm, Wikipedia:X-ray]\n",
      "-- Growth factors and hormones in beta cell proliferation\n",
      "-- miRNA regulation of p53 pathway in prostate cancer\n",
      "-- regulation of glial cell proliferation; described as \"Any process that modulates the frequency, rate or extent of glial cell proliferation.\" [GOC:dph, GOC:tb]\n",
      "-- Targeted agents in triple negative breast cancer\n",
      "-- Ovarian tumor domain proteases\n",
      "-- Epidermal nevus; described as Epidermal naevi are due to an overgrowth of the epidermis and may be present at birth (50%) or develop during childhood.\n",
      "-- TROP2 regulatory signaling\n",
      "-- MAPK pathway in congenital thyroid cancer\n"
     ]
    }
   ],
   "source": [
    "ge_verbalised = ge_analyser.verbalize_enrichment_results()\n",
    "print(ge_verbalised)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get gene info from CIVIC db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import civic_evidence_code\n",
    "importlib.reload(civic_evidence_code)\n",
    "\n",
    "path = r'C:\\Users\\owysocky\\Documents\\GitHub\\CCE_scGeneRAI\\resources\\civicdb'\n",
    "\n",
    "\n",
    "analyzer = civic_evidence_code.CivicEvidenceAnalyzer(path, gene_list)\n",
    "analyzer.create_feature_details_dict()\n",
    "\n",
    "details_dict = analyzer.add_evidence_to_dict()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Feature: NRAS\n",
      "Description: Mutations in the RAS family of proteins have frequently been observed across cancer types. The amino acid positions G12, G13 and Q61 account for the overwhelming majority of these mutations. The isoforms, despite their raw similarity, also behave very differently when expressed in non-native tissue types, likely due to differences in the C-terminal hyper-variable regions. Mis-regulation of isoform expression has been shown to be a driving event in cancer, as well as missense mutations at the three hotspots previously mentioned. While highly recurrent in cancer, targeting these RAS mutants has also been very elusive, and has not yet become common practice in the clinic.\n",
      "Summary: This is an N-ras oncogene encoding a membrane protein that shuttles between the Golgi apparatus and the plasma membrane. This shuttling is regulated through palmitoylation and depalmitoylation by the ZDHHC9-GOLGA7 complex. The encoded protein, which has intrinsic GTPase activity, is activated by a guanine nucleotide-exchange factor and inactivated by a GTPase activating protein. Mutations in this gene have been associated with somatic rectal cancer, follicular thyroid cancer, autoimmune lymphoproliferative syndrome, Noonan syndrome, and juvenile myelomonocytic leukemia. [provided by RefSeq, Jun 2011].\n",
      "Molecular Profiles:\n",
      "\t- While the NRAS G12 region is a widely studied recurrent region in cancer, its impact on clinical action is still debated.\n",
      "\t- While the NRAS G12 region is a widely studied recurrent region in cancer, its impact on clinical action is still debated.\n",
      "\t- NRAS Q61 mutations have been found in multiple myeloma, gastrointestinal stromal tumors, melanoma, and others. Two melanoma patients, each harboring mutations at this locus (Q61L and Q61R), responded to treatment with the akylating agent temozolomide. However, in colorectal cancer patients, mutations at this locus have been shown to confer resistance to cetuximab. The prognostic impact of mutations at this locus is currently under study.\n",
      "\t- NRAS Q61 mutations have been found in multiple myeloma, gastrointestinal stromal tumors, melanoma, and others. A melanoma patient harboring a mutation at this locus responded to treatment with the akylating agent temozolomide. However, in colorectal cancer patients, mutations at this locus have been shown to confer resistance to cetuximab. The prognostic impact of mutations at this locus is currently under study.\n",
      "\t- NRAS Q61 mutations have been found in multiple myeloma, gastrointestinal stromal tumors, melanoma, and others. A melanoma patient harboring a mutation at this locus responded to treatment with the akylating agent temozolomide. However, in colorectal cancer patients, mutations at this locus have been shown to confer resistance to cetuximab. The prognostic impact of mutations at this locus is currently under study.\n",
      "Evidence:\n",
      "\t- Recent studies in mouse have also shown that NRAS Q61K is capable of promoting melanoma formation and cooperates with INK4A loss to accelerate tumor onset.\n",
      "\t- In an in vitro study, an OXCO-2 cell line expressing NRAS G12D mutation demonstrated acquired resistance after low level cetuximab exposure compared to OXCO-2 parental cells expressing NRAS wild-type. Sensitivity was determined by assessing cell viability.\n",
      "\t- In an in vitro study, a PC-9 cell line expressing NRAS Q61K mutation demonstrated decreased sensitivity to erlotinib treatment compared to PC-9 cells expressing NRAS wild-type. Sensitivity was determined by performing cell growth-inhibition assays.\n",
      "\t- In patients with stage IV melanoma, NRAS mutation was associated with reduced median survival compared to patients with wildtype NRAS.\n",
      "\t- In an in vitro study of single-cell clones derived from a cutaneous melanoma patient expressing a BRAF V600E mutation (a known sensitizing mutation to vemurafenib) and NRAS Q61R co-mutations, NRAS Q61R mutation was associated with resistance to vemurafenib treatment (IC50: >1000 nmol/L). Resistance was determined by assessing cell viability.\n",
      "\t- Of the 4,562 patients with lung cancers, 30 showed NRAS mutations, which corresponded to 9 different amino acid substitutions. Cell lines were developed for 6 of these mutations and 5/6 were sensitivity to MEK inhibitors selumetinib and trametinib.  One of the cell lines was Q61L, two of the cell lines were Q61R, and three of the cell lines were Q61K.\n",
      "\t- In a retrospective study of 44 malignant melanoma patients, NRAS Q61L was associated with acquired resistance to vemurafenib monotherapy in one patient with disease progressive tumors harboring NRAS Q61L and BRAF V600E (a known sensitizing mutation to vemurafenib) co-mutations, but no other genetic alterations.\n",
      "\t- In a retrospective study of 44 malignant melanoma patients, NRAS Q61R was associated with acquired resistance to vemurafenib monotherapy in three patients with disease progressive tumors harboring NRAS Q61R and BRAF V600E (a known sensitizing mutation to vemurafenib) co-mutations, but no other genetic alterations.\n",
      "\t- Of the 4,562 patients with lung cancers, 30 showed NRAS mutations, which corresponded to 9 different amino acid substitutions. Cell lines were developed for 6 of these mutations (one harboring Q61L, two Q61R, and three Q61K mutations) and 5/6 were sensitive to MEK inhibitors selumetinib and trametinib. One Q61K line was not sensitive to these inhibitors.\n",
      "\t- One patient participating in a large retrospective study of cetuximab in metastatic, treatment refractory colorectal cancer had a tumor which harbored NRAS G13R, was wildtype for KRAS, BRAF and PIK3CA, and had individual response data. This patient was was a 68 year old male treated with cetuximab + irinotecan as 3rd line therapy who experienced stable disease (PFS: 18 weeks; OS: 34 weeks). Authors considered stable disease as both clinical benefit and non response.\n",
      "\t- Preclinical study in melanoma cell lines. Cell lines (2-5/assay) with NRAS mutation were more sensitive to Amuvatinib (inhibitor targets multiple tyrosine kinases) than BRAF mutant cell lines in growth inhibition, tyrosine phosphorylation inhibition, cell cycle and apoptotic (Annexin V) assays.\n",
      "\t- In chemotherapy-refractory patients with colorectal cancer, NRAS mutation status (primarily Q61, N=13) was not informative for disease control rate, progression free survival or overall survival following cetuximab plus chemotherapy (N=480).\n",
      "\t- Preclinical study in melanoma cell lines. NRAS Q61K, in addition to the BRAF V600E mutation was associated with resistance to vemurafenib but sensitivity to AZD6244 (selumetinib) in vitro in the acquired resistant cell line M249-AR4 and in the patient-derived acquired resistant cell line M376.\n",
      "\t- From 31 patients with advanced cutaneous squamous cell carcinomas evaluated for mutations in RAS, BRAF, and EGFR genes, only 2 RAS mutated samples (6.5%) were identified. Patients with NRAS p.G12D (c.35G>A) and HRAS p.G13V (c.38G>T) mutations had partial and complete responses to single agent rituximab, respectively. RAS mutation status was not correlated with response.\n",
      "\t- In an in vitro study, a DiFi colorectal cancer cell line stably expressing NRAS G12V mutation demonstrated resistance to cetuximab treatment compared to DiFi cells expressing wild-type NRAS. Resistance was determined by assessing clonogenic survival.\n",
      "\t- NRAS mutation status does not impact overall, event-free or disease-free survival in patients with AML.\n",
      "\t- In a retrospective study of 44 malignant melanoma patients, NRAS G13R was associated with acquired resistance to vemurafenib monotherapy in one patient with disease progressive tumors harboring NRAS G13R and BRAF V600E (a known sensitizing mutation to vemurafenib) co-mutations, but no other genetic alterations.\n",
      "\t- Phase II study in 71 patients with NRAS or BRAF V600 mutant melanoma treated with MEK162. Six (20%) of 30 patients with NRAS-mutated melanoma had a partial response (three confirmed).\n",
      "\t- Mice that carried the Q61H variant had Malignant Pleural Mesothelioma and Lewis Lung Carincoma. The malignant cells where taken and each cell line was shown to readily promote lung colonization compared to NRAS WT cells. The NRAS variants drove lung metastasis via interaction with lung endothelial (CXCR1) and myeloid (CXCR2) receptors, which were required for colonization. However, the receptors were not required for tumor growth.\n",
      "\t- In an in vitro study, a HTH7 cell line expressing NRAS Q61R demonstrated reduced sensitivity to vemurafenib treatment, compared to BRAF V600E expressing cells (8505C, SWI1736, BHT101 and BCPAP). Sensitivity was determined by assessing cell proliferation (HTH7, EC50: greater or equal to 1000nM vs. 8505C, EC50: 57nM; SWI1736, EC50: 29nM; BHT101, EC50: 97nM; BCPAP, EC50: 78nM), and MAPK1 and MAP2K1 phosphorylation.\n",
      "\t- Follicular thyroid cancers harboring RAS mutations have limited aggressiveness in the absence of TERT mutations. 2/2 follicular thyroid cancer patients with co-mutations in NRAS (10 Q61R and 1 Q61K) or HRAS (2 Q61R and 1 Q61K) and TERT promoter mutations had persistent or recurrent disease compared to 1/11 patients with only a RAS mutation (100% vs 8%, P=0.03).\n",
      "\t- In patients with follicular thyroid carcinoma, those with mutations in NRAS at codon 61 did not have significantly different overall survival than patients with wild-type NRAS (p=0.36).\n",
      "\t- Established a novel mouse model for melanoma that shows for the first time that over expression of a mutant form of human N-ras (N-RasQ61K) in melanocytes results in a hyper pigmented skin phenotype and increases the formation of melanoma lesions and favors the acquisition of a metastatic behavior on an INK4a background.\n",
      "\t- In a phase1 trial, patients with advanced solid tumor harboring RAS or RAF mutation were treated with pan-RAF inhibitor Belvarafenib (HM99573/GDC-5573).\n",
      "In the dose-escalation cohort, 4 of 9 patients with NRAS mutant melanoma showed tumor response.\n",
      "In the expansion cohort, 2 of 9 patients with NRAS mutant melanoma showed tumor response.\n",
      "In total response rate was 33.3% (6/19).\n",
      "\t- In 28 out of 40 (70%) metastatic colorectal cancer tumors harboring mutations in KRAS, NRAS, BRAF, or PIK3CA and implanted into mice, the therapeutic combination of the MEK inhibitor AZD6244 and the PI3K/mTor inhibitor BEZ235 resulted in disease stabilization. Mean tumor growth was lower in the tumors treated with AZD6244+ BEZ235 (+77% vs. +267%, P=1E-6) compared to AZD6244 alone and (+77% vs. +222%, P=0.0014) BEZ235 alone. NRAS mutations were in 4/40 tumors (three Q61K, one G12D).\n",
      "\t- Two patients participating in a large retrospective trial of cetuximab in metastatic, treatment refractory colorectal cancer had tumors which harbored NRAS G12C and were wild type for KRAS, PIK3CA, and BRAF. Both experienced stable disease, which the authors classified as non-response. The two patients whose tumors harbored NRAS G12C included a 63 year old female treated with cetuximab and irinotecan as a 3rd line treatment who had PFS of 34 weeks and OS of 55 weeks; and a 38 year old female treated with cetuximab, oxaliplatin and 5FU as a 2nd line therapy who had PFS of 12 weeks.\n",
      "\t- Inhibition of CDK4 and 6, regulators of the G1/S cell cycle checkpoint, arrests the cell cycle in the G1 phase, suppresses DNA synthesis, and inhibits cancer cell growth. There is an ongoing phase 1/2 trial of binimetinib with the CDK4/6 inhibitor ribociclib (LEE011) in mutant NRAS melanoma patients (NCT01781572). Early results from the 21 evaluable patients were very promising, with 7 patients achieving a partial response (33%) and 11 patients with stable disease (52%)\n",
      "\t- The presence of NRAS mutation in AML patients does not impact diease prognosis (resistant disease, disease-free survival, complete remission rate, relapse rate, induction death).\n",
      "\t- NRAS mutations have also been observed in melanoma and biliary tract, bladder, cervix, liver, lung, myeloid leukemia, ovarian, pancreatic, and thyroid cancer.\n",
      "\t- One patient participating in a large retrospective study of cetuximab in metastatic, treatment refractory colorectal cancer had a tumor which harbored NRAS Q61H was wildtype for KRAS, BRAF and PIK3CA. This patient was was a 73 year old male treated with cetuximab and irinotecan as 4th line therapy who experienced progressive disease and had PFS of 10 weeks and OS of 36 weeks.\n",
      "\t- NRAS mutations define a distinct subset of lung cancers (1%) with potential sensitivity to MEK inhibitors. Although NRAS mutations are more common in current/former smokers, the types of mutations are not those classically associated with smoking.\n",
      "\t- In a retrospective study of 44 malignant melanoma patients, NRAS Q61K was associated with acquired resistance to vemurafenib monotherapy in two patients with disease progressive tumors harboring NRAS Q61K and BRAF V600E (a known sensitizing mutation to vemurafenib) co-mutations, but no other genetic alterations.\n",
      "\t- The Medical Research Council (MRC) COIN trial consisted of patients with histologically confirmed adenocarcinoma of the colon or rectum, inoperable metastatic or locoregional measurable disease, and who had not received chemotherapy for metastatic disease.\n",
      "\n",
      "Regardless of treatment, median overall survival was shorter in patients who had mutations in BRAF, KRAS or NRAS (n=706, 13.6 months, IQR 7.923.1) than in patients whose tumors were wild type for all three (n=581, 20.1 months, IQR 11.531.7; log-rank test p<0.0001).\n",
      "\t- In an in vitro study, a M249 cell line expressing BRAF V600E (a known sensitizing mutation to vemurafenib) and NRAS Q61K co-mutations, demonstrated resistance to vemurafenib treatment (GI50: > 10 mol/L) compared to parental M249 cells expressing BRAF V600E and wild-type NRAS. Resistance was determined by assessing cell viability and phosphorylation of MAP2K1/2 and MAPK3/1.\n",
      "\t- Melanoma associated with NRAS Q61 mutation was more often associated with those at the extremity than those at the trunk.\n",
      "\t- Preclinical study in 16 NRAS mutant cell lines (melanoma, lung carcinoma, neuroblastoma with various Q61, G13, G12 mutations). The combination of metformin and trametinib shown synergistic effects in most tested cell lines (no antagonistic effects observed) and a melanoma xenograft model. Metformin-induced PI3K pathway suppression was identified as a potential mechanism.\n",
      "\t- In a retrospective study, authors analyzed BRAF V600 mutated melanoma metastases derived from 10 patients treated with the combination of dabrafenib and trametinib for resistance mechanisms and genetic correlates of response.\n",
      "An acquired resistance mechanism activating the MAPK pathway was identified in 9 tumors; BRAF amplification (n=4), MEK1/2 mutation (n=3), NRAS mutations (n=3). Among them, mutual exclusive alterations were BRAF amplification (n=4), MEK2 C125S (n=1), and NRAS Q61K (n=2).\n",
      "\t- One patient participating in a large retrospective study of cetuximab in metastatic, treatment refractory colorectal cancer had a tumor which harbored NRAS G13R, was wildtype for KRAS, BRAF and PIK3CA, and had individual response data. This patient was was a 50 year old male treated with panitumumab monotherapy as 3rd line therapy (discontinued for toxic/biochemical reasons) who experienced stable disease (PFS: 22 weeks; OS: 74 weeks). Authors considered stable disease as both clinical benefit and non-response\n",
      "\t- NEMO is an ongoing, randomised, open-label phase 3 study of patients with advanced, unresectable, AJCC IIIC/IV NRAS-mutant melanoma. 402 patients that were previously untreated or progressed on or after previous immunotherapy were randomised to receive either binimetinib (269) or dacarbazine (133). The study's primary endpoint was progression free survival. Median follow-up was 1.7 months (IQR 1.4-4.1). Median progression-free survival was 2.8 months (95% CI 2.8-3.6) in the binimetinib group and 1.5 months (1.5-1.7) in the dacarbazine group, demonstrating improved progression-free survival of binimetinib compared with dacarbazine.\n",
      "\t- In an in vitro study, a M249 R4 cell line expressing BRAF V600E (a known sensitizing mutation to vemurafenib) and NRAS Q61K co-mutations demonstrated acquired resistance to vemurafenib treatment. Resistance was determined by assessing cell survival.\n",
      "\t- Patients with colorectal cancer harboring NRAS mutation have poorer survival outcome and worse prognosis than patients with wildtype NRAS.\n",
      "\t- In six NRAS mutant melanoma cell lines, concomitant use of MEK inhibitor and PI3K/mTOR inhibitor resulted in greater decrease in cell viability than either alone, with the synergistic effect being marked in 4/6 cell lines. MEK inhibitor JTP-74057 and PI3K/mTOR inhibitor GSK2126458 together induced synergistic increase in apoptotic Annexin V+/PI+ cells in these cell lines. Tumor xenografts in nude mice were established with 5 of the 6 cell lines. Effects of single agent treatment with JTP-74057 or GSK2126458 ranged from minor inhibition to tumor stasis, while dual inhibitor treatment showed synergistic effects in each case, with marked tumor reductions in 2/5 cases and the authors report one compete response in a MaMel301 cell line xenograft mouse.\n",
      "\t- Two patients participating in a large retrospective study of cetuximab in metastatic, treatment refractory colorectal cancer had tumors which harbored NRAS Q61L and were wildtype for KRAS, BRAF and PIK3CA. Both patients experienced stable disease, which the authors classified as non-response. The two patients whose tumors harbored NRAS Q61L included a 53 year old female treated with cetuximab and irinotecan as a 6th line treatment who had PFS of 12 weeks and OS of 25 weeks; and a 64 year old female treated with cetuximab and ironotecan as a 3rd line therapy who had PFS of 12 weeks and OS of 46 weeks.\n",
      "\t- In 11 melanoma patients harboring BRAF mutations, one patient with a Q61K NRAS mutation acquired resistance to the BRAF inhibitor vemurafenib. Q61K and/or A146T NRAS mutations have also shown resistance to BRAF inhibitors in vitro. However, the combination of BRAF and MEK inhibitors have successfully hindered cell growth in cell lines with these mutations.\n",
      "\t- In a clinical study of 49 patients, 10 patients with Breast Cancer (BC) and wild-type NRAS gene were treated with Cetuximab. After 16 weeks, no clinical activity was observed. PFS of 6.7 weeks (95% CI, 1.7 -7.9) and OS of 14.1 weeks (95% CI, 4.3-30.1) was noted. No adverse effects of cetuximab treatment was reported in the cohort.\n",
      "\t- In 28 out of 40 (70%) metastatic colorectal cancer tumors implanted into mice, the therapeutic combination of the MEK inhibitor AZD6244 and the PI3K/mTor inhibitor BEZ235 resulted in disease stabilization. Mean tumor growth was lower in the tumors treated with AZD6244+ BEZ235 (+77% vs. +267%, P=1E-6) compared to AZD6244 alone and (+77% vs. +222%, P=0.0014) BEZ235 alone.\n",
      "\t- Retrospective analysis of thyroid carcinoma patients from a single institution identified 33/58 patients with follicular carcinomas had any RAS mutations (NRAS, KRAS, or HRAS at codons 12 or 61), 22 of which were NRAS Q61. NRAS Q61 was associated with aggressive disease in the form of distant metastases. Patients with RAS mutations had significantly shorter overall survival than those without (P=0.034, Kaplan-Meier). However, NRAS Q61 alone was not significantly associated with death (P=0.36, Chi-squared).\n",
      "\t- In this NCI-MATCH trial, 47 patients harboring codon 12, 13, or 61 NRAS mutated refractory solid tumors were treated with binimetinib. Of the 47 patients, >50% of the patients had metastatic colorectal adenocarcinoma (n=24). Around 30% of the the cohort had to discontinue due to binimetinib associated toxicity. The ORR was 2.1% (90% CI; 0.1-9.7), the 6-month PFS was 29.2% (90% CI; 19.4-44.0), the OS was 10.5 months (95% CI; 5.3-13.2), and the PFS was 3.5 months (95% CI; 1.8-5.8). Additionally, it was noted that patients with codon 61 mutated tumors had longer OS and PFS, compared to those with codon 12 or 13 mutated tumors. Overall, no appreciable antitumor activity was reported.\n",
      "\t- In an in vitro study of single-cell clones derived from a cutaneous melanoma patient expressing a BRAF V600E mutation (a known sensitizing mutation to vemurafenib) and NRAS Q61K co-mutations, NRAS Q61K mutation was associated with resistance to vemurafenib treatment (IC50: >1000 nmol/L). Resistance was determined by assessing cell viability.\n",
      "\t- In a retrospective study of 30 metastatic melanoma patients, a patient harboring NRAS G13R mutation was reported to be resistant to vemurafenib treatment. The patient was initially found to harbor a BRAF V600E mutation (a known vemurafenib sensitizing mutation) and had achieved a partial response with vemurafenib monotherapy prior to acquiring resistance.\n",
      "\t- Likely due to increased reliance of mutant NRAS on HSP90 for stabilization, inhibition of HSP90 by 17-AAG was shown to be effective in a patient with metastatic malignant melanoma with an NRAS G13D mutation.\n",
      "\t- One patient participating in a large retrospective trial of cetuximab in metastatic, treatment refractory colorectal cancer had a tumor which harbored NRAS G12D and was wild type for KRAS, PIK3CA, and BRAF. This patient was a 64 year old male treated with cetuximab and irinotecan as 5th line therapy who experienced stable disease and had PFS of 64 weeks and OS of 84 weeks.\n",
      "\t- In an open-label, phase II trial, 18 patients with histiocytosis of any mutational status were enrolled. The overall response rate was 89% (90% CI: 73 - 100). Responses were durable, with no acquired resistance at the time of publishing. At one year, 100% of responses were ongoing and 94% of patients remained progression-free. Cobimetinib treatment was efficacious regardless of genotype, and responses were observed in patients with ARAF, BRAF, RAF1, NRAS, KRAS, MEK1, and MEK2 mutations.\n",
      "\t- In chemotherapy-resistant colorectal cancer patients treated with cetuximab-based therapy, NRAS mutations were associated with a lower objective response (7.7% vs. 38.1%, odds ratio (OR):0.14, 95% CI:0.007-0.70, P=0.013, univariate analysis; OR: 0.105, 95% CI:0.0056-0.582, P=0.0065, multivariate analysis) compared to wildtype NRAS patients. Patients with NRAS mutations but no KRAS mutations experienced reduced overall survival (HR:1.98, 95% CI:1.08-3.62, P=0.042, multivariate analysis).\n",
      "\t- In a retrospective study of 44 relapsed melanoma patients harboring BRAF V600E or V600K (known BRAF inhibitor sensitizing mutations), MAPK pathway reactivation mechanisms were implicated in acquired resistance to BRAF inhibitor (vemurafenib or dabrafenib) monotherapy. The following MAPK pathway variants were found in progressive tumors with patient matched baseline tumors (some patients donated samples from multiple geographically/temporally distinct progressive tumors): NRAS mutations (G12D, G12R, G13R, Q61K, Q61R, Q61L) in 13/71 (18%) of progressive tumors, KRAS mutations (G12C, G12R, Q61H) in 5/71 (7%) of progressive tumors, mutant BRAF amplification (2-15 fold or 4-75 copies) in 11/57 (19%) of progressive tumors, mutant BRAF alternative splice variants (novel exon boundaries between 1/9, 1/11, 3/9) in 6/48 (13%) of progressive tumor RNA.\n",
      "\t- In an in vitro study, an OXCO-2 cell line expressing NRAS G12C mutation demonstrated acquired resistance after low level cetuximab exposure compared to OXCO-2 parental cells expressing NRAS wild-type. Sensitivity was determined by assessing cell viability.\n",
      "\t- Patients with a triple mutation of NPM1, DNMT3A, and NRAS (G12/13) (three-way interaction p=0.04) showed a relatively benign prognosis with an 8-year survival rate of 75% and a 10-year survival rate of 55%.  Conversely, patients with a concomitant NPM1/DNMT3A co-mutation, in the absence of an NRAS (G12/13) mutation, showed a clinically unfavorable 10-year survival rate of 30%.\n",
      "\t- In 11 melanoma patients harboring BRAF mutations, one patient with a Q61K NRAS mutation acquired resistance to the BRAF inhibitor vemurafenib. Q61K and/or A146T NRAS  mutations have also shown resistance to BRAF inhibitors in vitro. However, the combination of BRAF and MEK inhibitors have successfully hindered cell growth in cell lines with these mutations.\n",
      "\t- Five patients participating in a large retrospective study of cetuximab in metastatic, treatment refractory colorectal cancer had tumors which harbored NRAS Q61R and were wildtype for KRAS, BRAF and PIK3CA. Overall, one patient experienced stable disease, and the other four experienced progressive disease.The five patients whose tumors harbored NRAS Q61R included a 65 year old male treated with cetuximab monotherapy as a 3rd line treatment who experienced progressive disease and had PFS of 6 weeks and OS of 41 weeks; a 47 year old female treated with cetuximab and ironotecan as a 5th line therapy who experienced progressive disease and had PFS of 4 weeks and OS of 10 weeks; a 60 year old female treated with cetuximab and ironotecan as a 3rd line therapy who experienced stable disease and had PFS of 24 weeks and OS of 38 weeks; a 47 year old female treated with cetuximab and ironotecan as a 3rd line therapy who experienced progressive disease and had PFS of 14 weeks and OS of 18 weeks; and a 77 year old male treated with cetuximab monotherapy as a 1st line therapy who experienced progressive disease and had PFS of 7 weeks and OS of 67 weeks.\n",
      "\t- Four patients participating in a large retrospective study of cetuximab in metastatic, treatment refractory colorectal cancer had tumors which harbored NRAS Q61K and were wildtype for KRAS, BRAF and PIK3CA. One patient experienced partial response and three patients experienced progressive disease. The four patients whose tumors harbored NRAS Q61K included an 80 year old male treated with cetuximab and irinotecan as a 3rd line treatment who experienced progressive disease and had PFS of 6 weeks and OS of 43 weeks; a 51 year old male treated with cetuximab and ironotecan as a 3rd line therapy who experienced a partial response and had PFS of 24 weeks and OS of 54 weeks; a 64 year old male treated with cetuximab and ironotecan as a 5th line therapy who experienced progressive disease and had PFS of 8 weeks and OS of 33 weeks; and a 65 year old female treated with cetuximab, oxaliplatin and 5FU as a 2nd line therapy who experienced progressive disease and had PFS of 31 weeks and OS of 50 weeks.\n",
      "\t- Phase 1 trial in 57 patients with advanced cancers treated with selumetinib. 4 melanoma patients had NRAS Q61 mutations. One patient with NRAS mutation had 70% tumor shrinkage after three cycles of selumetinib (AZD6244) but developed symptomatic brain metastases before confirmatory scans could be performed.\n",
      "\t- Phase 1 trial of trametinib and docetaxel in patients with advanced melanoma lacking BRAF V600 mutations. 15 patients were enrolled (14 BRAF wt). Six patients had a partial response, including four out of eight patients with NRAS mutations.\n",
      "\t- In-vitro study in 5 neuroblastoma cell lines (2 with NRAS Q61K mutation). The combination of mTOR and MEK Inhibitors synergistically blocked cell growth in NRAS mutant but not wild type cell lines. Single agent MEK inhibition (AZD6244/selumetinib, MEK162 or PD0325901) and single agent mTOR inhibition (Everolimus or AZD8055) also blocked cell growth in NRAS mutant cell lines, whereas single agent PI3K inhibitors or MEK Inhibitors in combination with PIK3CA/AKT inhibitors did not show synergistic effects.\n",
      "\t- Of the 4,562 patients with lung cancers, 30 showed NRAS mutations, which corresponded to 9 different amino acid substitutions. Cell lines were developed for 6 of these mutations and 5/6 were sensitivity to MEK inhibitors selumetinib and trametinib.  One of the cell lines was Q61L, two of the cel lines were Q61R, and three of the cell lines were Q61K.\n",
      "\t- The researchers quantified ctDNA in time-intervals during the treatment of 6 patients with stage IV melanoma. The patient carrying the NRAS Q61H mutation was started on ipilimumab and responded well. As seen in figure 4, treatment led to a significant reduction in tumor burden, falling RECIST scores, and a rapid (86%) decline in ctDNA levels which correlates to a positive treatment response.\n",
      "\t- A cohort of patient-derived xenografts (PDX) from 85 patients with metastatic colorectal cancer was created. PDX were treated with cetuximab and mechanisms of resistance investigated. None of the xenografts harboring KRAS (N=18), NRAS (N=7) or BRAF (N=3) mutations showed a response to cetuximab whereas 1 out of 4 xenografts with a PIK3CA mutation responded to cetuximab.\n",
      "\t- In an in vitro study, the YULOVY and YUFULO cell lines expressing NRAS Q61L were associated with reduced sensitivity to vemurafenib (PLX4032) treatment, as compared to cells expressing wild-type BRAF. Resistance was determined by assessing cellular proliferation.\n",
      "\t- In a clinical study of 49 patients, 10 patients with Non-Small Cell Lung Cancer (NSCLC) and wild-type NRAS gene were treated with Cetuximab. After 16 weeks, no clinical activity was observed. A PFS of 8 weeks (95% CI, 5.9-8.4) and OS of 22.7 weeks (95% CI, 8.3-26.4) were noted, and a grade 3 adverse event was reported in this cohort, likely associated with cetuximab.\n",
      "\t- In a clinical study of 49 patients, 29 patients with Ovarian Cancer and wild-type NRAS gene were treated with Cetuximab. After 16 weeks, no clinical activity was observed. A PFS of 8 weeks (95% CI, 6.0-12.0) and OS of 21.6 weeks (95% CI, 12.3-41.7) were noted. At least one grade 3 adverse event was reported in three patients, and four patients had at least one serious adverse event likely associated with cetuximab.\n",
      "\t- In a melanoma patient with Q61R mutation, treatment with temozolomide resulted in overall survival of 16 months.\n",
      "\t- In an in vitro study, an OXCO-2 cell line expressing NRAS G13D mutation demonstrated acquired resistance after low level cetuximab exposure compared to OXCO-2 parental cells expressing NRAS wild-type. Sensitivity was determined by assessing cell viability.\n",
      "\t- In this phase II study of the arm E of the NCI-COG pediatric MATCH trial, 20 patients (median age: 14) were treated with Selumetinib. High-grade glioma (n=7) and rhabdomyosarcoma (n=7) were the most common types of cancers in this cohort. There was no objective response (PR or CR), the best overall response being SD, in three patients. The 6-month PFS rate was 15% (95% CI; 4-34). 25% of the patients had grade 3 or higher AEs, possibly attributed to selumetinib treatment.\n",
      "\t- One 56 year old female with advanced recurrent low grade serous ovarian cancer with NRAS Q61K mutation had a partial response to trametinib, with short follow-up.\n",
      "\t- In a panel of 7 NRAS mutant melanoma cell lines, all 7 were found to be sensitive to the MEK inhibitor MEK162, whereas 3 of 5 melanoma cell lines that were wild type for BRAF and NRAS were not sensitive to MEK162. In further tests using two of the seven NRAS mutant cell lines (YUDOSO Q16K/wt and YUKIM Q16R), colony forming capacity was reduced with increased MEK162 treatment, and MEK162 induced apoptotic markers in these cells.\n",
      "\t- In a retrospective study of 368 chemotherapy-refractory metastatic colorectal cancer, patients with a NRAS mutation (n=13) treated with cetuximab in combination with standard chemotherapy were associated with a lower response rate  (7.7%  vs 38.1% , OR: 0.14, 95% CI: 0.007-0.70, P=0.013) as compared to patients with wild-type NRAS (n=289).\n",
      "\t- Phase 1 trial in 97 patients with melanoma (81 cutaneous or unknown, 16 uveal). Seven patients were found to harbor an NRAS mutation and two of them experienced stable disease (one 48 weeks on treatment). Patients with concurrent BRAFV600wt/NRASwt (n=20) had a trend of higher response rates (20%) than BRAFV600wt/NRAS-mutant patients (n=11; RR=0%; p=0.27), as well as a trend of higher percentage of patients on study at Week 24 or at 1 year (40% vs. 18% [p=0.26] and 30% vs. 0% [p=0.07], respectively).\n",
      "\t- In patients with multiple myeloma, the presence of a mutation in NRAS was not significantly associated with progression-free or overall survival.\n",
      "\t- Phase 2 study of MEK162 in patients with advanced melanoma and NRAS or BRAF mutation. 71 patients were enrolled, 6/30 patients with NRAS mutation had a partial response (Gln61Leu 1 Patient, Gln61Lys 9 Patients, Gln61Arg 15 Patients). Another 13 patients had stable disease (63% disease control rate).\n",
      "\t- This was a retrospective study of 605 patients with metastatic, chemotherapy refractory colorectal cancer treated with cetuximab + chemotherapy. Patients with KRAS wildtype, NRAS assessable tumors, and individual response data (n=301) were genotyped for NRAS mutations. Patients harboring NRAS mutant tumors (n=13) had a significantly lower response rate than did patients with wildtype NRAS (n=289; 1/13 [7.7%] vs 110/289 [77.9%]; p =.013). However, there was no significant difference between patients with mutant or wildtype NRAS in disease control rate, progression free survival, or overall survival. Authors note that their small sample size of patients with NRAS mutant tumors made it difficult to assess the effect of NRAS mutation on survival thus the large confidence intervals. Authors further note that the close relationship between KRAS and NRAS, and the lack of NRAS response to another EGFR inhibitor (panitumumab) support the inclusion of NRAS genotyping prior to cetuximab treatment. Authors suggest screening for mutations in the following gene order: KRAS, BRAF, NRAS, PIK3CA exon 21 (listed as PIK3CA exon 20 in the source).\n",
      "\t- In a retrospective study of 148 treatment naive metastatic colorectal cancer patients, patients with RAS mutations (n=10), including KRAS A146T, KRAS G13D, NRAS G12D and NRAS Q179*, treated with FOLFOX4 plus cetuximab were associated with decreased overall survival (16.3mo vs. 28.5mo, HR:0.43, 95% CI:0.20-0.89, P=0.020) and decreased progression free survival (7.2mo vs. 9.7mo, HR:0.56, 95% CI:0.27-1.16, P=0.11), as compared to patients with wild-type KRAS and NRAS.\n",
      "\t- In an open-label, phase II trial, 18 patients with histiocytosis of any mutational status were enrolled. Patients received cobimetinib at a starting dose of 60 mg daily for 21 days of each 28-day cycle. The overall response rate was 89% (90% CI: 73 - 100). Responses were durable, with no acquired resistance at the time of publishing. At one year, 100% of responses were ongoing and 94% of patients remained progression-free. Cobimetinib treatment was efficacious regardless of genotype, and responses were observed in patients with ARAF, BRAF, RAF1, NRAS, KRAS, MEK1, and MEK2 mutations.\n",
      "\t- Vemurafenib resistance is associated with gain of Q61 mutation in NRAS.\n",
      "\t- In this Phase II study, 20 patients harboring activating RAS gene mutations (KRAS = 8, NRAS = 3. HRAS = 1), BRAF V600E mutations, and NF1 inactivating mutations (n=7) were treated with Selumetinib. No objected response was reported in these patients, and the six-month PFS was 15% (95% CI; 4-34). 25% (n=5) of the patients experienced a grade 3 or higher adverse effects, likely associated with selumetinib. Overall, selumetinib had limited efficacy in this cohort.\n",
      "\t- A total of 657 melanoma samples were analyzed for the influence of NRAS copy number gain (CNG) and the influence on overall survival and efficacy on Binimetinib. NRAS CNG was mutually exclusive with NRAS mutations and significantly associated with shorter overall survival. Treatment of cells with MEK inhibitor Binimetinib inhibited tumor growth.\n",
      "\t- Chemotherapy-refractory patients with KRAS wt metastatic colorectal cancer harboring NRAS mutation (primarily Q61, n=10) have a significantly lower response rate to cetuximab than patients wildtype for NRAS (7.1% [1/14] vs 38.1% [110/289]).\n",
      "\t- In this phase III, randomized, open-label trial, 377 patients with wild-type KRAS exon two were randomized 1:1 to receive panitumumab 6mg/kg IV on day 1 of each 14-day cycle plus BSC or BSC alone. The median overall survival was 10.0 months in the panitumumab plus BSC cohort versus 7.4 months in the BSC-only cohort  (HR, 0.73; 95% CI: 0.57 - 0.93; p = 0.0096). Progression-free survival was also improved in the panitumumab group with median PFS being 3.6 (95% CI: 3.4 - 5.3) months for panitumumab plus BSC versus 1.7 (95% CI: 1.6 - 1.9) months for BSC alone (HR, 0.51; 95% CI: 0.41 - 0.64; p < 0.0001). Patients with wildtype RAS tumours (KRAS + NRAS) also benefited from panitumumab with a median overall survival of  10.0 (95% CI: 8.7 - 11.6) months versus 6.9 (95% CI: 5.2 - 7.9) months (HR, 0.70; 95% CI: 0.53 - 0.93; p = 0.0135). Lastly, the overall response rate for the panitumumab plus BSC cohort was 31.0% (95% CI: 23.5 - 39.3) versus 2.3% (95% CI: 0.5 - 6.7) for BSC alone (OR = 20.0; 95% CI: 5.9 - 101.6; p < 0.0001.\n",
      "\t- Phase I expansion and pharmacodynamic study of the oral MEK inhibitor RO4987655 (CH4987655). Among 8 patients with NRAS mutated melanoma, 3 experienced clinical benefit: 2 stable disease and 1 partial response (1 Patient received study treatment for 323 days). Of 10 NRAS wt patients, 4 experienced clinical benefit. Authors note that RO4987655 demonstrates clinical efficacy in BRAF wildtype melanomas with NRAS mutations.\n",
      "\t- NCCN guidelines indicate colorectal cancer patients with NRAS mutations should not be treated with EGFR inhibitors cetuximab or panitumumab.\n",
      "\t- In a melanoma patient with NRAS Q61L mutation, treatment with temozolomide resulted in disease free survival of 14 months.\n",
      "_______________\n",
      "Feature: APC\n",
      "Description: nan\n",
      "Summary: This gene encodes a tumor suppressor protein that acts as an antagonist of the Wnt signaling pathway. It is also involved in other processes including cell migration and adhesion, transcriptional activation, and apoptosis. Defects in this gene cause familial adenomatous polyposis (FAP), an autosomal dominant pre-malignant disease that usually progresses to malignancy. Mutations in the APC gene have been found to occur in most colorectal cancers, where disease-associated mutations tend to be clustered in a small region designated the mutation cluster region (MCR) and result in a truncated protein product. [provided by RefSeq, Jun 2022].\n",
      "Molecular Profiles:\n",
      "Evidence:\n",
      "\t- In a phase 3 trial (NCT03785964) involving 142 adults with progressing desmoid tumors, the gamma-secretase inhibitor nirogacestat demonstrated a significant progression-free survival benefit compared to placebo (hazard ratio = 0.29; 95% CI, 0.15 to 0.55). The objective response rate was significantly higher with nirogacestat than placebo (41% vs. 8%).  Nirogacestat also provided significant improvements in secondary patient-reported outcomes, including pain, symptom burden, physical functioning, and health-related quality of life.  Desmoid tumors have activation of the WT pathway (with ~90% having CTNNB1 mutations and 10% harboring APC mutations) which is associated with activation of the NOTCH pathway.  According to the subgroup analysis, patients with either APC mutations or  CTNNB1 mutations showed improved progression-free survival with nirogacestat compared to placebo.  However, there are no desmoid tumor patients without CTNNB1 or APC mutations making it impossible to conclude that these mutations are the specific reason desmoid tumors respond to nirogacestat.\n",
      "\t- Canonical Wnt signaling in colon carcinoma cells with APC mutation was inhibited by JW55 treatment.\n",
      "\t- While evidence from multiple clinical studies establishing a strong association between loss of function mutations in APC and response to 5-fluorouracil is lacking, some studies suggest APC mutations may be associated with poor prognosis on 5-fluorouracil therapy. Evidence from a clinical study has shown that in colorectal cancer (CRC) patients treated with 5- fluorouracil, those with APC mutation positive tumors had a significantly increased risk for cancer related death (HR=6.9), compared to patients without APC mutations (HR=0.34). Preclinical studies have also shown that APC-mutant CRC cells display a limited response to 5-fluorouracil induced apoptosis, when compared to cells with wild-type APC.\n",
      "\t- G007-LK inhibits in vivo tumor growth in a subset of APC-mutant colorectal cancers xenograft models but may be limited by intestinal toxicity.\n",
      "_______________\n",
      "Feature: CDH1\n",
      "Description: nan\n",
      "Summary: This gene encodes a classical cadherin of the cadherin superfamily. Alternative splicing results in multiple transcript variants, at least one of which encodes a preproprotein that is proteolytically processed to generate the mature glycoprotein. This calcium-dependent cell-cell adhesion protein is comprised of five extracellular cadherin repeats, a transmembrane region and a highly conserved cytoplasmic tail. Mutations in this gene are correlated with gastric, breast, colorectal, thyroid and ovarian cancer. Loss of function of this gene is thought to contribute to cancer progression by increasing proliferation, invasion, and/or metastasis. The ectodomain of this protein mediates bacterial adhesion to mammalian cells and the cytoplasmic domain is required for internalization. This gene is present in a gene cluster with other members of the cadherin family on chromosome 16. [provided by RefSeq, Nov 2015].\n",
      "Molecular Profiles:\n",
      "Evidence:\n",
      "\t- 285 intestinal-type tumors and 183 diffuse-type tumors were evaluated for mutational profiles.  CDH1 mutations are associated with diffuse type mucinous gastric carcinoma, which indicate worse prognosis when compared to intestinal type (p<0.001).\n",
      "\t- In 42 families with hereditary diffuse gastric cancer, CDH1 sequencing revealed germline mutations including A298T, W409R, and R732Q. In vitro assays in CHO-K1 cells demonstrated that all three missense mutations reduced cell-cell adhesion (from particle diameter 1067 um to 26, 32 and 29 um respectively in aggregation assay) and increased invasion (from 1.2% to 13.5, 22.1 and 13.2%, respectively in collagen invasion assay), supporting these as loss-of-function oncogenic alterations.\n",
      "_______________\n",
      "Feature: GATA3\n",
      "Description: nan\n",
      "Summary: This gene encodes a protein which belongs to the GATA family of transcription factors. The protein contains two GATA-type zinc fingers and is an important regulator of T-cell development and plays an important role in endothelial cell biology. Defects in this gene are the cause of hypoparathyroidism with sensorineural deafness and renal dysplasia. [provided by RefSeq, Nov 2009].\n",
      "Molecular Profiles:\n",
      "Evidence:\n",
      "\t- This case study is on a 60-year-old man with a 7-year history of fatigue that was noted to have a palpable abdominal mass while being seen for gastroesophageal reflux disease. A CT scan of the abdomen and pelvis showed a perirenal soft-tissue mass on the left side and similar soft tissue in the perinephric region surrounding the renal pelvis on the right side. A core-needle biopsy of the renal mass on the left side revealed infiltration of fibroadipose tissue with sheets of large histiocytic forms that were positive for S100 but negative for factor XIIIa. The immunophenotype and the presence of emperipolesis were consistent with RosaiDorfman disease. The patient received prednisone but had no response. This was followed by targeted next-generation sequencing of DNA from the renal mass with a 447-gene panel which identified a hotspot activating mutation in codon 12 of KRAS (p.G12R) at a variant-allele frequency of 8% (of 456 reads) within overall tumour cellularity of 20% (determined by a morphologic assessment); mutations were also found in GATA3 and GNAS that were of unknown clinical significance but thought to be germline in origin. KRAS-activating mutations result in constitutive activation of the mitogen-activated protein kinase (MAPK) pathway that may be blocked by MAPK kinase (MEK) inhibitors. Therefore, we initiated treatment with cobimetinib (60 mg daily) for days 1 through 21 of a 28-day cycle. A repeat CT scan after 2 months of treatment with cobimetinib showed a substantial response.\n",
      "_______________\n",
      "Feature: MYC\n",
      "Description: nan\n",
      "Summary: This gene is a proto-oncogene and encodes a nuclear phosphoprotein that plays a role in cell cycle progression, apoptosis and cellular transformation. The encoded protein forms a heterodimer with the related transcription factor MAX. This complex binds to the E box DNA consensus sequence and regulates the transcription of specific target genes. Amplification of this gene is frequently observed in numerous human cancers. Translocations involving this gene are associated with Burkitt lymphoma and multiple myeloma in human patients. There is evidence to show that translation initiates both from an upstream, in-frame non-AUG (CUG) and a downstream AUG start site, resulting in the production of two isoforms with distinct N-termini. [provided by RefSeq, Aug 2017].\n",
      "Molecular Profiles:\n",
      "Evidence:\n",
      "\t- This study found the clinical BTK/ERBB2 kinase inhibitor, ibrutinib, downregulated ERK-mediated signal transduction, cMYC Ser-62 phosphorylation and levels of MYC protein, and elicited G1 cell cycle arrest and apoptosis, in both MYC and ERBB2 amplified tumour cells, suggesting that this drug could be used to treat biomarker-selected groups of patients with oesophageal cancer.\n",
      "\t- An 82-year old male (smoker: 60 pack-years) was diagnosed with lung adenocarcinoma. Initial tumor tissue testing revealed MET amplification (GCN = 13.5). The patient was enrolled in a phase 2 trial and given capmatinib (400 mg, twice daily). However, disease continued to progress through treatment. Next-generation sequencing showed MYC amplification (estimated copy number = 5). After further treatment, multiple brain metastases developed and the patient passed away. The authors suggest a MYC inhibitor such as ICX-101 to be used after capmatinib administration to inhibit resistance of cells to capmatinib.\n",
      "\t- BET inhibition by JQ1 downregulates MYC transcription, followed by genome-wide downregulation of Myc-dependent target genes. In experimental models of multiple myeloma, a Myc-dependent hematologic malignancy, JQ1 produces a potent antiproliferative effect associated with cell-cycle arrest and cellular senescence.\n",
      "\t- In 2016 the WHO added a new category of lymphoma, separate from diffuse large B-cell lymphoma, termed high-grade B-cell lymphoma with MYC AND [BCL2 OR BCL6] translocations. These represent less than 10% of cases of diffuse large B-cell lymphoma. They may be referred to as double or triple hit lymphomas. Patients with these combination of somatic translocation variant have a poor prognosis when treated with standard therapy and increased risk of central nervous system involvement and progression. This variant specifically refers to somatic mutations in the genome typically detected by FISH or genomic assays and are distinct from expression biomarkers involving these same genes (typically defined by IHC).\n",
      "\t- A preclinical study to evaluate the efficacy of targeting the overexpressed CHEK1 cell cycle chekpoint kinase, including both miRNA and Prexasertib-induced inhibition. Both in vitro and in vivo, Prexasertib had strong single agent efficacy and augmented the effect of Olaparib or Cisplatin in either combination, including platinum-resistant models. CHEK1 and MYC overexpression were predictors of response.\n",
      "\t- Collectively data from expression analysis of 102 esophageal squamous cell carcinoma (ESCC) sample tissues established MYC and HSP90 as potential targets with overexpression of MYC observed in over half the patient samples and strong levels of HSP90 in nearly 47% cases of all tumors. In vitro, the relative protein expression analysis of 5 ESCC cell lines displayed concurrent overexpression of HSP90 and MYC in both KYSE-150 and Eca-109 cell lines, while overexpression of HSP90 was solely found in EC-1, TE-13 and HESS. Treatment of STA-9090 an HSP90 inhibitor, inhibited ESCC proliferation with significant sensitivity shown in cell lines overexpressing HSP90 and MYC proteins (KYSE-150 with a IC50 value of 29.32 nM and Eca-109 with a value of 69.44nM). In vivo experiments also showed benefit of STA-9090 in MYC expressing PDX models, and no beneficial effect of STA-9090 in MYC-low ESCC PDX models. These findings are indicative of potential use in ESCC tissues with the patient populations possessing overexpression of HSP90 and MYC proteins.\n",
      "\t- MYC and BCL2 rearrangements together are defining diagnostic features of this new entity\n",
      "\t- Group 3 medulloblastomas with MYC amplifications are often associated with large cell/anaplastic histology, young patients, and overall poor prognosis.\n",
      "\t- MYC translocations, often (but not always) involving immunoglobulin loci, are common in DLBCL and one of two defining features of high-grade B-cell lymphoma with MYC and BCL2 rearrangements\n",
      "\t- A patient diagnosed with lung adenocarcinoma had cells cultured to create patient-derived cell lines. Such cells exhibited MET amplification and showed overexpression of MYC. Capmatinib was administered with doses ranging from 0 - 10 uM with drug resistance, consistent with drug resistance in vivo. However, ICX-101, a MYC inhibitor, showed sensitivity and was tested in combination and sequentially with capmatinib. The authors report that when cell lines were treated with capmatinib and subsequently with ICX-101, there was a synergistic anti-tumor effect. The authors suggest a MYC inhibitor could be utilized to inhibit resistance to capmatinib.\n",
      "_______________\n",
      "Feature: RB1\n",
      "Description: nan\n",
      "Summary: The protein encoded by this gene is a negative regulator of the cell cycle and was the first tumor suppressor gene found. The encoded protein also stabilizes constitutive heterochromatin to maintain the overall chromatin structure. The active, hypophosphorylated form of the protein binds transcription factor E2F1. Defects in this gene are a cause of childhood cancer retinoblastoma (RB), bladder cancer, and osteogenic sarcoma. [provided by RefSeq, Jul 2008].\n",
      "Molecular Profiles:\n",
      "Evidence:\n",
      "\t- In Capan-1 cell lines engineered to reduce pRB expression (by transfection with p16), resensitization to chemotherapy-induced apoptosis was observed. The effect was asserted to be specific to pRB depletion because, in two separate cell lines, also transfected with p16, that retained high pRB expression, chemotherapy resistance was still observed.\n",
      "\t- Three patients with hormone receptor positive metastastic breast cancer, treated with letrozol/ribociclib or fulvestrant/palbociclib, underwent RB1 gene sequencing before, during and after acquiring resistance to therapy. ctDNA analyses revealed the emergence of RB1 mutations after but not before progression on CDK4/6 Inhibitors.\n",
      "\t- Study evaluated a total of 50 small cell lung cancer patients with extensive stage disease. Multivariate analysis identified RB1 as a factor in predicting response to first-line chemotherapy (odds ratio 5.58 compared to WT, p=.038). Kaplan-Meier survival analysis revealed patients with RB1 mutations exhibited better overall survival (p = .04, 11.7 versus 9.1 months, CI = 1.1-28.2).\n",
      "\t- The study investigated the sensitivity of RB1-mutant osteosarcoma cell lines to PARPi. It found that cell viability, measured by AUC values after 5 days, was significantly lower in RB1-defective cell lines compared to RB1-normal lines when treated with three different PARPi: olaparib, niraparib, and talazoparib. Further, the study confirmed this selective sensitivity using clonogenic assays for all three PARPi in RB1-defective osteosarcoma cells. Overall, the study establishes an association between PARPi sensitivity and RB1-defective status in otherwise PARPi insensitive osteosarcoma.\n",
      "\t- We have used polyclonal anti-synthetic peptide serum to study the role of retinoblastoma gene (RB) inactivation in a variety of human tumor cell lines. Our analysis indicates that inactivation of the RB protein, p105-Rb, is universal in retinoblastoma cells, vindicating the predictions of the Knudson \"two-hit\" hypothesis. In addition, our analysis has shown that inactivations of the RB gene are nearly as frequent in a more common human tumor, small cell lung carcinoma. One-third of bladder carcinomas surveyed also carry altered or absent p105-Rb. Other human tumors by contrast demonstrate only infrequent inactivation of the RB gene. These results suggest that inactivation of the RB gene is a critical step in the pathogenesis of a subset of human tumors.\n",
      "\t- In this phase III trial (PALOMA-3), patients with ER+, HER2- advanced breast cancer were treated with CDK4/6 inhibitors. Patients were randomized to receive palbociclib plus fulvestrant (P+F) or placebo plus fulvestrant (F). A paired ctDNA exome sequencing was conducted to analyze clonal composition. RB1 mutations were identified in 6/127 patients (p=0.041), part of the P + F arm, while no RB1 mutations were identified in the patients treated with fulvestrant alone, at EOT (end of treatment). ESR1 mutations were noted in 25.1% at day 1, and 31.3% of the patients at EOT. An increased proportion of PIK3CA mutations were noted at EOT compared to day 1  (19% v/s 26.7%; p = 0.00069). Additionally, the proportions of patients with new PIK3CA mutations did not differ between the two treatment groups, which suggests that these mutations may play a role in fulvestrant resistance. Strong evidence was reported for positive selection for ESR1 Y537S through the treatment in both treatment groups (p=0.0037), suggesting this mutation may promote resistance to fulvestrant. The data also indicates that resistance to fulvestrant may be a major genetic driver of resistance to the combination treatment.\n",
      "\t- A drug screen of triple-negative breast cancer (TNBC) cell lines found several therapies to be synthetically lethal with RB1 loss of function. Cell lines with endogenous expression of RB1 (BT20, MB453, MCF7, and MB231) were resistant to PLK inhibitor volasertib, while those with RB1 loss (MB468, BT549, AW23) or experimental knockdown had increased sensitivity. In addition, inhibition of CDK4/6 upstream of RB1 reduced sensitivity to volasertib in RB1-positive cells (MB231).\n",
      "\t- This study reportedly sequenced 1001 DLBCL tumors using whole exome sequencing. They reported 150 recurrently mutated genes. Some concerns about the validity of the results have been expressed in subsequent studies. Only genes that have been confirmed by other similar studies were added to CIViC at this time.\n",
      "\t- The p110alpha-specific PI3K inhibitor BYL719 (Alpelisib) was used on a panel of five PI3K inhibitor sensitive and five resistant breast cancer cell lines. Sensitive cells (IC50<400nM) showed reduced phospho-Rb (p-Rb) western blot levels in response to BYL719 treatment, while resistant cell lines (IC50>800nM) showed significantly less reduction in p-Rb after BYL719. Paired biopsies before and after treatment were taken from a set of eight patients classified as either responder or non-responder to BYL719. Responder samples showed significantly reduced p-Rb staining after treatment in comparison to non-responder biopsies. In biopsies taken from a group of three responders who progressed, p-Rb levels were increased after progression.\n",
      "\t- A phase II clinical study was conducted with 29 patients with Non-Small-Cell Lung Cancer harboring CDKN2A alterations and wild-type Rb. The study reported one patient with a partial response, and six patients with SD for a duration of at least 16 weeks. The disease control rate was 31% (90% CI, 19-40). The PFS and OS observed were 8.1 weeks (95% CI, 7.1 - 16.0) and 21.6 weeks (95% CI. 14.1-41.1) respectively. At least one grade 3 or 4 adverse event or serious adverse event was reported in 11 patients, likely associated with palbociclib.\n",
      "\t- Positive RB1 expression (2+ by IHC) was associated with adverse outcomes in a cohort of 73 cases with stage I and II NSCLC (P<0.001) in this retrospective study. This finding was independent from other genomic and clinical factors explored by the authors.\n",
      "\t- Preclinical study in breast cancer cell lines treated with CDK4/6 inhibitor palbociclib. Palbociclib-resistant xenografts from one cell line (PDX244) were developed in-vivo. In resistant xenografts, western blot analysis showed a decrease of pRb protein levels in 4 of 7 CDK4/6-acquired resistant tumors and a sustained expression of the E2F target cyclin E2, in contrast to CDK4/6 inhibitor-sensitive PDX244. Genomic characterization of PDX244LR1 showed the acquisition of an RB1 frameshift mutation (RB1 p.M695fs*26).\n",
      "\t- A drug screen of triple-negative breast cancer (TNBC) cell lines found several therapies to be synthetically lethal with RB1 loss of function. Cell lines with endogenous expression of RB1 (BT20, MB453, MCF7, and MB231) were resistant to CHK1 inhibitor AZD7762, while those with RB1 loss (MB468, BT549, AW23) or experimental knockdown had increased sensitivity. In addition, inhibition of CDK4/6 upstream of RB1 reduced sensitivity to AZD7762 in RB1-positive cells (MB231). Cell cycle analysis indicated that CHK1 inhibition in RB deficient cells resulted in the appearance of fragmented nuclei and S phase collapse, which indicates irreparable DNA damage during S phase.\n",
      "\t- A drug screen of triple-negative breast cancer (TNBC) cell lines found several therapies to be synthetically lethal with RB1 loss of function. Cell lines with endogenous expression of RB1 (BT20, MB453, MCF7, and MB231) were resistant to gemcitabine, while those with RB1 loss (MB468, BT549, AW23) or experimental knockdown had increased sensitivity. In addition, inhibition of CDK4/6 upstream of RB1 reduced sensitivity to gemcitabine in RB1-positive cells (MB231).\n",
      "\t- A panel of 21 glioblastoma cell lines harboring various mutations was treated with palbociclib. 16 of the cell lines with intact Rb showed decreased proliferation (bromodeoxyuridine incorporation, BrdU flow cytometry) and growth arrest characteristic of senescence, while the 5 cell lines that did not show a palbociclib response had homozygous deletion of Rb. Rb positive and palbociclib responsive GBM cell line U87MG was treated with shRNA against Rb. shRb treated cells showed restoration of cell proliferation (BrdU incorporation) under palbociclib treatment, indicating resistance.\n",
      "\t- In this study, the sensitivity of RB1 mutated cancer cells to AURKAi LY3295668 has been investigated.  Enhanced cytotoxicity was reported in RB1 depleted lung and breast cancer cells in response to the AURKAi. However, the same cytotoxicity wasn't observed in AURKA gene knockout cells.\n",
      "\t- In 3 ER positive and 2 ER negative breast cancer cell lines with intact Rb, significant growth inhibition, as measured by BrdU incorporation, was seen with incubation in CDK4/6 inhibitor palbociclib (PD-0332991). In contrast, two ER negative cell lines lacking detectable Rb were fully resistant to palbociclib. Knockdown of Rb in 3 of the tested Rb positive cell lines led to moderate palbociclib resistance in short term incubation experiments. \n",
      "In longer term (15 day) palbociclib incubations of ER negative MDA-MB-231 cells and ER positive MCF-7 cells, significant resistance to palbociclib was seen in Rb knockdown conditions while wild-type Rb cells maintained sensitivity.\n",
      "_______________\n",
      "\n"
     ]
    }
   ],
   "source": [
    "civicdb_knowledge = analyzer.verbalize_civicdb_knowledge()\n",
    "print(civicdb_knowledge )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PharmaKB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pharmabk_code as pbk\n",
    "importlib.reload(pbk)\n",
    "path = 'C:\\\\Users\\\\owysocky\\\\Documents\\\\GitHub\\\\CCE_scGeneRAI\\\\resources\\\\pharmgkb'\n",
    "\n",
    "pharmakb_analyzer = pbk.Pharmakb_Analyzer(gene_list)\n",
    "pharmakb_analyzer.get_pharmakb_knowledge(files_path = path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Based on PharmGKB knowledge, the following information was found for the genes in the differential graph:\n",
      "\n",
      "Gene: ATM; Drug(s): metformin; Statement: Allele A is not associated with response to metformin in women with Polycystic Ovary Syndrome as compared to allele C.; Notes: The authors evaluated whether women with PCOS would respond to one year of treatment with metformin either with or without oral contraception. Although there was an overall improvement in metabolic measures in subjects due to metformin, this response was not associated with any genetic variant investigated.; (PMID: 28834135).Notes: The authors evaluated whether women with PCOS would respond to one year of treatment with metformin either with or without oral contraception. Although there was an overall improvement in metabolic measures in subjects due to metformin, this response was not associated with any genetic variant investigated.; (PMID: 28834135).\n",
      "Gene: ATM; Drug(s): metformin; Statement: Allele A is not associated with decreased clinical benefit to metformin in people with Diabetes Mellitus, Type 2 as compared to allele C.; Notes: \"Other SNPs (rs4621031, rs34399035, rs1800058, and rs11212617) had no significant impact on metformin response. \"; (PMID: 39612420).Notes: \"Other SNPs (rs4621031, rs34399035, rs1800058, and rs11212617) had no significant impact on metformin response. \"; (PMID: 39612420).\n",
      "Gene: ATM; Drug(s): metformin; Statement: Allele T is not associated with decreased clinical benefit to metformin in people with Diabetes Mellitus, Type 2 as compared to allele C.; Notes: \"Other SNPs (rs4621031, rs34399035, rs1800058, and rs11212617) had no significant impact on metformin response. \" This was not polymorphic (all CC) in this population; (PMID: 39612420).Notes: \"Other SNPs (rs4621031, rs34399035, rs1800058, and rs11212617) had no significant impact on metformin response. \" This was not polymorphic (all CC) in this population; (PMID: 39612420).\n",
      "Gene: ATM; Drug(s): metformin; Statement: Allele C is associated with decreased concentrations of metformin in people with Diabetes Mellitus, Type 2 as compared to allele A.; Notes: Authors state is \"close to significance with application of the Bonferroni method\"; (PMID: 29790415).Notes: Authors state is \"close to significance with application of the Bonferroni method\"; (PMID: 29790415).\n",
      "Gene: ATM; Drug(s): Iodine (131I) compounds, radiotherapy; Statement: Genotype AG is associated with decreased likelihood of Fatigue when treated with Iodine (131I) compounds and radiotherapy in people with Thyroid Neoplasms as compared to genotype GG.; Notes: \"patients with GA genotype of the ATM rs11212570 were more unlikely to develop fatigue compared to GG genotype carriers (for GA genotype: OR=0.42 (0.190.94), p<0.05) (Table 7).\" Only one patient was AA homozygous.; (PMID: 37708952).Notes: \"patients with GA genotype of the ATM rs11212570 were more unlikely to develop fatigue compared to GG genotype carriers (for GA genotype: OR=0.42 (0.190.94), p<0.05) (Table 7).\" Only one patient was AA homozygous.; (PMID: 37708952).\n",
      "Gene: ATM; Drug(s): Iodine (131I) compounds, radiotherapy; Statement: Genotypes AA + AG is associated with increased likelihood of Pain when treated with Iodine (131I) compounds and radiotherapy in people with Thyroid Neoplasms as compared to genotype GG.; Notes: nan; (PMID: 30430914).Notes: nan; (PMID: 30430914).\n",
      "Gene: ATM; Drug(s): metformin; Statement: Allele C is not associated with response to metformin in people with Diabetes Mellitus, Type 2 as compared to allele A.; Notes: in Iranian patients. Response to metformin was defined by HbA1c and fasting blood sugar (FBS) values.; (PMID: 27386433).Notes: in Iranian patients. Response to metformin was defined by HbA1c and fasting blood sugar (FBS) values.; (PMID: 27386433).\n",
      "Gene: ATM; Drug(s): metformin; Statement: Allele C is associated with increased response to metformin in people with Diabetes Mellitus, Type 2 as compared to allele A.; Notes: as measured by decrease in Z score (which includes HbA1c and daily dose of insulin)   at 16 weeks, authors report as trend that was not significant after Bonferroni.; (PMID: 29790415).Notes: as measured by decrease in Z score (which includes HbA1c and daily dose of insulin)   at 16 weeks, authors report as trend that was not significant after Bonferroni.; (PMID: 29790415).\n",
      "Gene: ATM; Drug(s): Iodine (131I) compounds, radiotherapy; Statement: Genotype TT is associated with increased likelihood of gastrointestinal toxicity when treated with Iodine (131I) compounds and radiotherapy in people with Thyroid Neoplasms as compared to genotype CC.; Notes: Authors say this is significant and that recessive models CT+TT vs CC and CT+CC vs TT are not significantly different.; (PMID: 30430914).Notes: Authors say this is significant and that recessive models CT+TT vs CC and CT+CC vs TT are not significantly different.; (PMID: 30430914).\n",
      "Gene: ATM; Drug(s): cyclophosphamide, doxorubicin, fluorouracil; Statement: Genotypes AA + AG are associated with increased risk of Nausea when treated with cyclophosphamide, doxorubicin and fluorouracil in women with Breast Neoplasms as compared to genotype GG.; Notes: Grade 1-3 nausea.; (PMID: 29507678).Notes: Grade 1-3 nausea.; (PMID: 29507678).\n",
      "Gene: ATM; Drug(s): cyclophosphamide, doxorubicin, fluorouracil; Statement: Genotype AA is associated with increased risk of Vomiting when treated with cyclophosphamide, doxorubicin and fluorouracil in women with Breast Neoplasms as compared to genotypes AG + GG.; Notes: Early grade 3 vomiting.; (PMID: 29507678).Notes: Early grade 3 vomiting.; (PMID: 29507678).\n",
      "Gene: ATM; Drug(s): metformin; Statement: Genotype AC is associated with increased response to metformin in people with Diabetes Mellitus, Type 2 as compared to genotypes AA + CC.; Notes: Patients with the AC genotype of rs11212617 achieved significantly lower FPG than that of patients with AA or CC homozygote of rs11212617, including 30FPG and respectively). Association between genotypes and the efficacy of metformin with regression analysis, which adjusted for several variables, including age, BMI, sex, the dose of metformin, education, tea drink, smoking, and sweet. Patients were eligible for the study if they had newly diagnosed T2DM and could not achieve an appropriate FPG level.; (PMID: 34545025).Notes: Patients with the AC genotype of rs11212617 achieved significantly lower FPG than that of patients with AA or CC homozygote of rs11212617, including 30FPG and respectively). Association between genotypes and the efficacy of metformin with regression analysis, which adjusted for several variables, including age, BMI, sex, the dose of metformin, education, tea drink, smoking, and sweet. Patients were eligible for the study if they had newly diagnosed T2DM and could not achieve an appropriate FPG level.; (PMID: 34545025).\n",
      "Gene: BRAF; Drug(s): FOLFIRI, FOLFOX, XELOX; Statement: Allele T is associated with decreased overall survival when treated with FOLFIRI, FOLFOX or XELOX in people with Colorectal Neoplasms as compared to genotype AA.; Notes: \"BRAF V600E wild-type status correlated with better survival than BRAF V600E patients (HR = 0.28, 95% CI 0.0870.909)\"; (PMID: 38248103).Notes: \"BRAF V600E wild-type status correlated with better survival than BRAF V600E patients (HR = 0.28, 95% CI 0.0870.909)\"; (PMID: 38248103).\n",
      "Gene: BRAF; Drug(s): dabrafenib; Statement: Allele T is associated with increased sensitivity to dabrafenib as compared to allele A.; Notes: Dabrafenib had IC50 of 3.2nM against wild-type human BRAF (A allele) and IC50 of 0.6nM against V600E BRAF.; (PMID: 23844038).Notes: Dabrafenib had IC50 of 3.2nM against wild-type human BRAF (A allele) and IC50 of 0.6nM against V600E BRAF.; (PMID: 23844038).\n",
      "Gene: BRAF; Drug(s): selumetinib; Statement: Genotypes AT + TT is associated with increased progression-free survival when treated with selumetinib in people with Thyroid Neoplasms as compared to genotype AA.; Notes: Patients with tumors containing the BRAF V600E genotype (12 of 26) had an improved median PFS of 33 weeks compared with a median PFS of 11 weeks in patients with tumors lacking this mutation, although this comparison did not reach statistical significance with CIs overlapping throughout the follow-up period.; (PMID: 22241789).Notes: Patients with tumors containing the BRAF V600E genotype (12 of 26) had an improved median PFS of 33 weeks compared with a median PFS of 11 weeks in patients with tumors lacking this mutation, although this comparison did not reach statistical significance with CIs overlapping throughout the follow-up period.; (PMID: 22241789).\n",
      "Gene: BRAF; Drug(s): dabrafenib; Statement: Allele G is associated with increased sensitivity to dabrafenib as compared to allele A.; Notes: Dabrafenib had IC50 of 3.2nM against wild-type human BRAF (A allele) and IC50 of 0.5nM against V600K BRAF.; (PMID: 23844038).Notes: Dabrafenib had IC50 of 3.2nM against wild-type human BRAF (A allele) and IC50 of 0.5nM against V600K BRAF.; (PMID: 23844038).\n",
      "Gene: GATA3; Drug(s): prednisolone; Statement: Genotype AA is associated with decreased clinical benefit to prednisolone in children with Precursor Cell Lymphoblastic Leukemia-Lymphoma as compared to genotypes AC + CC.; Notes: nan; (PMID: 36714653).Notes: nan; (PMID: 36714653).\n",
      "Gene: GATA3; Drug(s): prednisolone; Statement: Genotype AA is associated with increased risk of Sepsis when treated with prednisolone in children with Precursor Cell Lymphoblastic Leukemia-Lymphoma as compared to genotypes AC + CC.; Notes: \"AA genotype had more episodes of sepsis than those with the CC or CA genotype (P=0.019). However, we did not identify a significant association between the rs3824662 genotype and other side effects, such as myelosuppression,\" \"AA genotype remained statistically significantly associated with sepsis after adjusting clinical variables (i.e., age, white blood cell count, and risk group)\"; (PMID: 36714653).Notes: \"AA genotype had more episodes of sepsis than those with the CC or CA genotype (P=0.019). However, we did not identify a significant association between the rs3824662 genotype and other side effects, such as myelosuppression,\" \"AA genotype remained statistically significantly associated with sepsis after adjusting clinical variables (i.e., age, white blood cell count, and risk group)\"; (PMID: 36714653).\n",
      "Gene: GATA3; Drug(s): Glucarpidase; Statement: Genotypes AA + AC are associated with increased risk of nephrotoxicity requiring treatment with Glucarpidase in children with Precursor Cell Lymphoblastic Leukemia-Lymphoma as compared to genotype CC.; Notes: Patients were receiving methotrexate treatment for ALL. Each A allele increases the odds of the patient requiring glucarpidase treatment for kidney injury.; (PMID: 33788417).Notes: Patients were receiving methotrexate treatment for ALL. Each A allele increases the odds of the patient requiring glucarpidase treatment for kidney injury.; (PMID: 33788417).\n",
      "Gene: GATA3; Drug(s): asparaginase, cyclophosphamide, cytarabine, daunorubicin, dexamethasone, doxorubicin, leucovorin, mercaptopurine, methotrexate, pegaspargase, prednisone, thioguanine, vincristine; Statement: Allele A is associated with increased likelihood of relapse when treated with asparaginase, cyclophosphamide, cytarabine, daunorubicin, dexamethasone, doxorubicin, leucovorin, mercaptopurine, methotrexate, pegaspargase, prednisone, thioguanine and vincristine in children with Precursor Cell Lymphoblastic Leukemia-Lymphoma as compared to allele C.; Notes: The A allele of this SNP was also associated with susceptibility to Philadelphia chromosome-like ALL (statistics were adjusted for genetic ancestry).; (PMID: 24141364).Notes: The A allele of this SNP was also associated with susceptibility to Philadelphia chromosome-like ALL (statistics were adjusted for genetic ancestry).; (PMID: 24141364).\n",
      "Gene: GATA3; Drug(s): asparaginase, cyclophosphamide, cytarabine, daunorubicin, dexamethasone, doxorubicin, leucovorin, mercaptopurine, methotrexate, pegaspargase, prednisone, thioguanine, vincristine; Statement: Allele A is associated with increased minimal residual disease when treated with asparaginase, cyclophosphamide, cytarabine, daunorubicin, dexamethasone, doxorubicin, leucovorin, mercaptopurine, methotrexate, pegaspargase, prednisone, thioguanine and vincristine in children with Precursor Cell Lymphoblastic Leukemia-Lymphoma as compared to allele C.; Notes: The A allele was associated with higher MRD level at the end of induction therapy. The A allele of this SNP was also associated with susceptibility to Philadelphia chromosome-like ALL and increased risk of relapse.; (PMID: 24141364).Notes: The A allele was associated with higher MRD level at the end of induction therapy. The A allele of this SNP was also associated with susceptibility to Philadelphia chromosome-like ALL and increased risk of relapse.; (PMID: 24141364).\n",
      "Gene: KRAS; Drug(s): cetuximab; Statement: Allele T is not associated with increased response to cetuximab in people with Colorectal Neoplasms.; Notes: Allele complemented to plus chromosomal strand. For patients with G13D, cetuximab did not appear to provide a clinically relevant benefit.; (PMID: 26371285).Notes: Allele complemented to plus chromosomal strand. For patients with G13D, cetuximab did not appear to provide a clinically relevant benefit.; (PMID: 26371285).\n",
      "Gene: KRAS; Drug(s): cetuximab, panitumumab; Statement: Allele C is not associated with response to cetuximab or panitumumab in people with Colorectal Neoplasms as compared to allele A.; Notes: Meta-analysis with 5 studies. The authors did not provide the exact number of patients but stated that \"the median number of patients per analysis was 110 (range 50 - 740)\". Most definitions of response were variations of the RECIST criteria. Please note that alleles have been complemented to the plus chromosomal strand.; (PMID: 27897268).Notes: Meta-analysis with 5 studies. The authors did not provide the exact number of patients but stated that \"the median number of patients per analysis was 110 (range 50 - 740)\". Most definitions of response were variations of the RECIST criteria. Please note that alleles have been complemented to the plus chromosomal strand.; (PMID: 27897268).\n",
      "Gene: KRAS; Drug(s): cetuximab, panitumumab; Statement: Genotypes AC + CC is not associated with response to cetuximab or panitumumab in people with Colorectal Neoplasms as compared to genotype AA.; Notes: In the results section the uncorrected p value listed looks good but when looking at Table 3 can see the corrected p value is not significant and authors state result is not significant.; (PMID: 25210463).Notes: In the results section the uncorrected p value listed looks good but when looking at Table 3 can see the corrected p value is not significant and authors state result is not significant.; (PMID: 25210463).\n",
      "Gene: KRAS; Drug(s): capecitabine, cetuximab, oxaliplatin; Statement: Genotype AC is associated with increased response to capecitabine, cetuximab and oxaliplatin in people with Rectal Neoplasms as compared to genotype AA.; Notes: C allele carriers (there were no homozygotes) had a statistically significantly higher rate of complete response (CR) after neoadjuvant therapy and a trend for better 5-year progression-free survival (PFS) and overall survival. Both CR and survival outcomes were independent of the use of cetuximab. Patients received neoadjuvant CAPOX followed by chemoradiotherapy, surgery and adjuvant CAPOX plus or minus cetuximab; (PMID: 26162609).Notes: C allele carriers (there were no homozygotes) had a statistically significantly higher rate of complete response (CR) after neoadjuvant therapy and a trend for better 5-year progression-free survival (PFS) and overall survival. Both CR and survival outcomes were independent of the use of cetuximab. Patients received neoadjuvant CAPOX followed by chemoradiotherapy, surgery and adjuvant CAPOX plus or minus cetuximab; (PMID: 26162609).\n",
      "Gene: KRAS; Drug(s): cetuximab, panitumumab; Statement: Allele A is associated with increased overall survival when treated with cetuximab or panitumumab in people with Colorectal Neoplasms.; Notes: compared to patients with other KRAS mutations. Study showed increased survival for patients with codon 13 mutations compared to those with other KRAS mutations when treated with anti-EGFR antibodies. Patients with codon 13 mutations had similar survival as those with no KRAS mutations when treated with anti-EGFR antibodies. Authors did not specify exactly which codon 13 mutations were included (or may not have known since this was a meta-analysis).; (PMID: 23090619).Notes: compared to patients with other KRAS mutations. Study showed increased survival for patients with codon 13 mutations compared to those with other KRAS mutations when treated with anti-EGFR antibodies. Patients with codon 13 mutations had similar survival as those with no KRAS mutations when treated with anti-EGFR antibodies. Authors did not specify exactly which codon 13 mutations were included (or may not have known since this was a meta-analysis).; (PMID: 23090619).\n",
      "Gene: KRAS; Drug(s): cetuximab, panitumumab; Statement: Allele T is not associated with increased overall survival when treated with cetuximab or panitumumab in people with Colorectal Neoplasms.; Notes: compared with other KRAS mutations in this meta-analysis. Whereas patients with \"wild type\" KRAS has improved OS. Significant heterogeneity between studies was observed for hazard ratio estimates of the KRAS G13D tumours (allele complemented here to plus chromosomal strand).; (PMID: 26812186).Notes: compared with other KRAS mutations in this meta-analysis. Whereas patients with \"wild type\" KRAS has improved OS. Significant heterogeneity between studies was observed for hazard ratio estimates of the KRAS G13D tumours (allele complemented here to plus chromosomal strand).; (PMID: 26812186).\n",
      "Gene: KRAS; Drug(s): cetuximab; Statement: Allele T is associated with increased progression-free survival when treated with cetuximab in people with Colorectal Neoplasms.; Notes: compared to patients with other KRAS mutations. The median PFS was 4.5 months in patients with KRAS p.G13D mutations (n=9), compared with 2.8 months in those with other KRAS mutations (n=22)(allele complemented to plus chromosomal strand).; (PMID: 26623049).Notes: compared to patients with other KRAS mutations. The median PFS was 4.5 months in patients with KRAS p.G13D mutations (n=9), compared with 2.8 months in those with other KRAS mutations (n=22)(allele complemented to plus chromosomal strand).; (PMID: 26623049).\n",
      "Gene: KRAS; Drug(s): cetuximab; Statement: Allele T is not associated with increased overall survival when treated with cetuximab in people with Colorectal Neoplasms.; Notes: Study compared G13D (allele A, complemented to T on plus chromosomal strand) with other KRAS mutations. Full Text was not available.; (PMID: 22537608).Notes: Study compared G13D (allele A, complemented to T on plus chromosomal strand) with other KRAS mutations. Full Text was not available.; (PMID: 22537608).\n",
      "Gene: KRAS; Drug(s): cetuximab; Statement: Allele T is associated with increased overall survival when treated with cetuximab in people with Colorectal Neoplasms.; Notes: compared to patients with other KRAS mutations. Study showed increased survival for patients with the G13D mutation (allele A, Gly13Asp complemented to plus chromosomal strand = T) compared to those with other codon 13 mutations or mutations in codons 12, 61 or 146 when treated with cetuximab. Patients with G13D had similar survival as those with no KRAS mutations when treated with cetuximab.; (PMID: 20978259).Notes: compared to patients with other KRAS mutations. Study showed increased survival for patients with the G13D mutation (allele A, Gly13Asp complemented to plus chromosomal strand = T) compared to those with other codon 13 mutations or mutations in codons 12, 61 or 146 when treated with cetuximab. Patients with G13D had similar survival as those with no KRAS mutations when treated with cetuximab.; (PMID: 20978259).\n",
      "Gene: KRAS; Drug(s): cetuximab, irinotecan; Statement: Allele T is not associated with increased overall survival when treated with cetuximab and irinotecan in people with Colorectal Neoplasms.; Notes: Study compared G13D (allele A, complemented to T on plus chromosomal strand) against other KRAS mutations (G12D, G12V and G12C). Authors noted, the disease control rate was higher in KRAS p.G13D-mutant patients and a partial response was observed in one such patient.; (PMID: 23071293).Notes: Study compared G13D (allele A, complemented to T on plus chromosomal strand) against other KRAS mutations (G12D, G12V and G12C). Authors noted, the disease control rate was higher in KRAS p.G13D-mutant patients and a partial response was observed in one such patient.; (PMID: 23071293).\n",
      "Gene: KRAS; Drug(s): cetuximab; Statement: Allele A is not associated with increased progression-free survival when treated with cetuximab in people with Colorectal Neoplasms.; Notes: compared to treated with chemotherapy alone. Study compared patients with the G12V mutation (allele T, Gly12Val complemented to A on plus chromosomal strand) receiving chemotherapy (FOLFOX or FOLFIRI) and those receiving chemotherapy plus cetuximab, and there was no benefit to adding cetuximab.; (PMID: 22734028).Notes: compared to treated with chemotherapy alone. Study compared patients with the G12V mutation (allele T, Gly12Val complemented to A on plus chromosomal strand) receiving chemotherapy (FOLFOX or FOLFIRI) and those receiving chemotherapy plus cetuximab, and there was no benefit to adding cetuximab.; (PMID: 22734028).\n",
      "Gene: KRAS; Drug(s): cetuximab, panitumumab; Statement: Genotypes AC + CC is associated with increased progression-free survival when treated with cetuximab or panitumumab in people with Colorectal Neoplasms as compared to genotype AA.; Notes: Individuals with the AC and CC genotypes did not benefit from additional chemotherapy beyond monotherapy with anti-EGFR mAb,  but those with the AA genotype required additional chemotherapy to achieve the same rate of progression-free survival.; (PMID: 25183481).Notes: Individuals with the AC and CC genotypes did not benefit from additional chemotherapy beyond monotherapy with anti-EGFR mAb,  but those with the AA genotype required additional chemotherapy to achieve the same rate of progression-free survival.; (PMID: 25183481).\n",
      "Gene: KRAS; Drug(s): cetuximab; Statement: Genotypes AC + CC is not associated with increased event-free survival when treated with cetuximab in people with Colonic Neoplasms as compared to genotype AA.; Notes: Patients also received FOLFOX or FOLFIRI or both sequentially.; (PMID: 24727325).Notes: Patients also received FOLFOX or FOLFIRI or both sequentially.; (PMID: 24727325).\n",
      "Gene: KRAS; Drug(s): cetuximab, panitumumab; Statement: Allele C is associated with nonresponse to anti-EGFR-based treatment when treated with cetuximab and panitumumab in people with Adenocarcinoma as compared to genotype AA.; Notes: nan; (PMID: 23324806).Notes: nan; (PMID: 23324806).\n",
      "Gene: KRAS; Drug(s): cetuximab, fluorouracil, irinotecan, oxaliplatin; Statement: Allele T is associated with increased progression-free survival when treated with cetuximab, fluorouracil, irinotecan and oxaliplatin in people with Colorectal Neoplasms.; Notes: Study compared patients with the G13D mutation (allele A, Gly13Asp complemented here to plus chromosomal stand T) receiving chemotherapy (FOLFOX or FOLFIRI)  plus cetuximab and those receiving chemotherapy alone. Patients with different KRAS mutations did not benefit from the addition of cetuximab.; (PMID: 22734028).Notes: Study compared patients with the G13D mutation (allele A, Gly13Asp complemented here to plus chromosomal stand T) receiving chemotherapy (FOLFOX or FOLFIRI)  plus cetuximab and those receiving chemotherapy alone. Patients with different KRAS mutations did not benefit from the addition of cetuximab.; (PMID: 22734028).\n",
      "Gene: MYC; Drug(s): warfarin; Statement: Genotypes CT + TT is associated with increased dose of warfarin in people with heart valve replacement as compared to genotype CC.; Notes: This variant was found to be significant in the multivariate analysis and accounted for approximately 2.8% of variability in warfarin dose.; (PMID: 26249541).Notes: This variant was found to be significant in the multivariate analysis and accounted for approximately 2.8% of variability in warfarin dose.; (PMID: 26249541).\n",
      "Gene: MYC; Drug(s): warfarin; Statement: Genotype CC is associated with increased dose of warfarin in people with heart valve replacement as compared to genotypes CT + TT.; Notes: This variant was found to be significant in the multivariate analysis and accounted for approximately 2.4% of variability in warfarin dose.; (PMID: 26249541).Notes: This variant was found to be significant in the multivariate analysis and accounted for approximately 2.4% of variability in warfarin dose.; (PMID: 26249541).\n",
      "Gene: MYC; Drug(s): warfarin; Statement: Genotype CT is associated with decreased dose of warfarin in people with heart valve replacement as compared to genotype TT.; Notes: There were no CC homozygotes observed. \"Patients with TC alleles of c-Myc rs4645962 required a 30% dose reduction compared with patients carrying TT alleles.\"; (PMID: 26249541).Notes: There were no CC homozygotes observed. \"Patients with TC alleles of c-Myc rs4645962 required a 30% dose reduction compared with patients carrying TT alleles.\"; (PMID: 26249541).\n",
      "Gene: PTEN; Drug(s): carboplatin, cisplatin; Statement: Genotypes AC + CC are associated with decreased likelihood of Drug Toxicity when treated with carboplatin or cisplatin in people with Carcinoma, Non-Small-Cell Lung as compared to genotype AA.; Notes: Chemotherapy was primary-platinum based; most patients' regime was carboplatin-based. 16 SNPs were tested.  p is given as 0.036, and the adjusted p is not listed but the result is stated to be significant.  \"The Benjamini-Hochberg method was used to correct for multiple comparisons based on a FDR of 10%\".; (PMID: 20447721).Notes: Chemotherapy was primary-platinum based; most patients' regime was carboplatin-based. 16 SNPs were tested.  p is given as 0.036, and the adjusted p is not listed but the result is stated to be significant.  \"The Benjamini-Hochberg method was used to correct for multiple comparisons based on a FDR of 10%\".; (PMID: 20447721).\n",
      "Gene: PTEN; Drug(s): capecitabine, fluorouracil; Statement: Allele C is associated with increased response to capecitabine and fluorouracil in people with Neoplasm Metastasis as compared to allele T.; Notes: pfSNP identified 2800 SNPS associated with key-words: 5-FU, capecitabine and oxilaplatin and colorectal cancer. Various criteria were used to determine 1536 SNPs to genotype. These SNPs were genotyped in a discovery cohort (N=62) and tested in a validation cohort (N=27). Both cohorts consisted of patients with colorectal cancer metastasis in liver. Five SNPs (rs2289310 G>T; rs1047840 G>A;  rs17431184 T>C; rs17160359 G>T; rs2236722 A>G)  were identified as distinguishing the \"non-responder\" phenotype from the \"responder\" phenotype when using a logistic regression multivariate model. The AUC for the receiver operating characteristic curve of the 5 SNPs is 0.875. This logistic-based multivariate model is said to be able to identify 39.1% of non-responders.; (PMID: 25372392).Notes: pfSNP identified 2800 SNPS associated with key-words: 5-FU, capecitabine and oxilaplatin and colorectal cancer. Various criteria were used to determine 1536 SNPs to genotype. These SNPs were genotyped in a discovery cohort (N=62) and tested in a validation cohort (N=27). Both cohorts consisted of patients with colorectal cancer metastasis in liver. Five SNPs (rs2289310 G>T; rs1047840 G>A;  rs17431184 T>C; rs17160359 G>T; rs2236722 A>G)  were identified as distinguishing the \"non-responder\" phenotype from the \"responder\" phenotype when using a logistic regression multivariate model. The AUC for the receiver operating characteristic curve of the 5 SNPs is 0.875. This logistic-based multivariate model is said to be able to identify 39.1% of non-responders.; (PMID: 25372392).\n",
      "Gene: TP53; Drug(s): capecitabine, paclitaxel; Statement: Genotype GG is associated with decreased response to capecitabine and paclitaxel in people with Stomach Neoplasms as compared to genotype CC.; Notes: Outcomes were response rate, progression free survival, and overall survival. Alleles given as Pro/Arg, which Pro (G on + strand) associated with worse response.; (PMID: 26696550).Notes: Outcomes were response rate, progression free survival, and overall survival. Alleles given as Pro/Arg, which Pro (G on + strand) associated with worse response.; (PMID: 26696550).\n",
      "Gene: TP53; Drug(s): cisplatin, cyclophosphamide; Statement: Genotypes CG + GG are not associated with decreased risk of progression-free survival or overall survival when treated with cisplatin and cyclophosphamide in women with Ovarian Neoplasms as compared to genotype CC.; Notes: [stat_test: kaplan-meier].; (PMID: 22188361).Notes: [stat_test: kaplan-meier].; (PMID: 22188361).\n",
      "Gene: TP53; Drug(s): levonorgestrel; Statement: Genotype CG is associated with increased likelihood of adverse events when treated with levonorgestrel in women.; Notes: Author state that there was a significant \" rs1042522 in TP53 was significantly different between cases and controls.\" and \"Regarding rs1042522 as the risk variant, the subjects carrying rs1042522 are not recommended to use LNG contraception\" but do not state which allele is risk allele. There do not seem to be differences in allele frequency but there are with genotypes.; (PMID: 34783250).Notes: Author state that there was a significant \" rs1042522 in TP53 was significantly different between cases and controls.\" and \"Regarding rs1042522 as the risk variant, the subjects carrying rs1042522 are not recommended to use LNG contraception\" but do not state which allele is risk allele. There do not seem to be differences in allele frequency but there are with genotypes.; (PMID: 34783250).\n",
      "Gene: TP53; Drug(s): cisplatin, cyclophosphamide; Statement: Genotype GG is not associated with increased risk of Drug Toxicity when treated with cisplatin and cyclophosphamide in women with Ovarian Neoplasms as compared to genotypes CC + CG.; Notes: Clinical outcomes measured: neutropenia, anemia, thrombocitopenia, nephrotoxicity, neuropathy, emesis and ototoxicity [stat_test: fisher's exact test and chi-squared test]. An association was seen with neutropenia in a cohort of Russian patients previously genotyped in an earlier study (see [PMID: 19786980]).; (PMID: 22188361).Notes: Clinical outcomes measured: neutropenia, anemia, thrombocitopenia, nephrotoxicity, neuropathy, emesis and ototoxicity [stat_test: fisher's exact test and chi-squared test]. An association was seen with neutropenia in a cohort of Russian patients previously genotyped in an earlier study (see [PMID: 19786980]).; (PMID: 22188361).\n",
      "Gene: TP53; Drug(s): cisplatin, radiotherapy; Statement: Genotype CG is associated with increased overall survival when treated with cisplatin and radiotherapy in women with Uterine Cervical Neoplasms as compared to genotypes CC + GG.; Notes: \"increased median survival time was observed for patients carrying Arg/Pro genotype when compared with patients with Arg/Arg and Pro/Pro genotypes (126 months vs. 111 months)\". Note: This is a C/G SNP in a gene on the minus chromosomal strand. This was not significant for disease free survival or response rate.; (PMID: 29731921).Notes: \"increased median survival time was observed for patients carrying Arg/Pro genotype when compared with patients with Arg/Arg and Pro/Pro genotypes (126 months vs. 111 months)\". Note: This is a C/G SNP in a gene on the minus chromosomal strand. This was not significant for disease free survival or response rate.; (PMID: 29731921).\n",
      "Gene: TP53; Drug(s): docetaxel, paclitaxel; Statement: Genotype CC is not associated with risk of Peripheral Nervous System Diseases when treated with docetaxel and paclitaxel in women with Breast Neoplasms as compared to genotypes CG + GG.; Notes: The association was with risk of neuropathy of level 1 and higher. The scale of neuropathy is: 0 = normal, 1 = asymptomatic, weakness on physical examination, loss of reflexes, or parenthesis not interfering with daily function, 2 = weakness and sensory alterations interfering with daily function, 3 = weakness and sensory alterations interfering with activities of daily living or requirement bracing or assistive devices, and 4 = life threatening, paralysis, disabling.; (PMID: 27574448).Notes: The association was with risk of neuropathy of level 1 and higher. The scale of neuropathy is: 0 = normal, 1 = asymptomatic, weakness on physical examination, loss of reflexes, or parenthesis not interfering with daily function, 2 = weakness and sensory alterations interfering with daily function, 3 = weakness and sensory alterations interfering with activities of daily living or requirement bracing or assistive devices, and 4 = life threatening, paralysis, disabling.; (PMID: 27574448).\n",
      "Gene: TP53; Drug(s): cisplatin, paclitaxel; Statement: Genotypes CG + GG are associated with decreased response to cisplatin and paclitaxel in people with Stomach Neoplasms as compared to genotype CC.; Notes: Association reported for 72 ProPro and Arg/Pro (Pro is interpreted as C on coding strand, gene is on minus strand, and therefore risk allele is given as G on plus strand as shown above); (PMID: 19052714).Notes: Association reported for 72 ProPro and Arg/Pro (Pro is interpreted as C on coding strand, gene is on minus strand, and therefore risk allele is given as G on plus strand as shown above); (PMID: 19052714).\n",
      "Gene: TP53; Drug(s): oxaliplatin; Statement: Genotypes CG + GG is associated with decreased response to oxaliplatin in people with Colorectal Neoplasms as compared to genotype CC.; Notes: Patients receiving FOLFOX (oxaliplatin+fluorouracil+leucovorin) with or without bevacizumab. No significant association with progression-free survival or overall survival was seen.; (PMID: 29925895).Notes: Patients receiving FOLFOX (oxaliplatin+fluorouracil+leucovorin) with or without bevacizumab. No significant association with progression-free survival or overall survival was seen.; (PMID: 29925895).\n",
      "Gene: TP53; Drug(s): docetaxel, paclitaxel; Statement: Genotype CC is not associated with risk of Peripheral Nervous System Diseases when treated with docetaxel and paclitaxel in women with Breast Neoplasms as compared to genotypes CG + GG.; Notes: The significant association was with risk of neuropathy of level 2 and higher. The scale of neuropathy is: 0 = normal, 1 = asymptomatic, weakness on physical examination, loss of reflexes, or parenthesis not interfering with daily function, 2 = weakness and sensory alterations interfering with daily function, 3 = weakness and sensory alterations interfering with activities of daily living or requirement bracing or assistive devices, and 4 = life threatening, paralysis, disabling.; (PMID: 27574448).Notes: The significant association was with risk of neuropathy of level 2 and higher. The scale of neuropathy is: 0 = normal, 1 = asymptomatic, weakness on physical examination, loss of reflexes, or parenthesis not interfering with daily function, 2 = weakness and sensory alterations interfering with daily function, 3 = weakness and sensory alterations interfering with activities of daily living or requirement bracing or assistive devices, and 4 = life threatening, paralysis, disabling.; (PMID: 27574448).\n",
      "Gene: TP53; Drug(s): cisplatin, paclitaxel; Statement: Genotypes CG + GG are associated with decreased one year progression free survival when treated with cisplatin and paclitaxel in people with Stomach Neoplasms as compared to genotype GG.; Notes: However, the overall survival was not significantly different.; Association reported for 72 ProPro and Arg/Pro (Pro is interpreted as C on coding strand, gene is on minus strand, and therefore risk allele is given as G on plus strand as shown above); (PMID: 19052714).Notes: However, the overall survival was not significantly different.; Association reported for 72 ProPro and Arg/Pro (Pro is interpreted as C on coding strand, gene is on minus strand, and therefore risk allele is given as G on plus strand as shown above); (PMID: 19052714).\n",
      "Gene: TP53; Drug(s): fluorouracil; Statement: Genotype GG is associated with decreased overall survival when treated with fluorouracil in people with Stomach Neoplasms as compared to genotypes CC + CG.; Notes: This G/C snp is in TP53 which is on the minus strand, presenting the opportunity for the alleles to be easily confused.   The C>G change in the gene is represented by a G>C change on the positive chromosomal strand.; (PMID: 18357466).Notes: This G/C snp is in TP53 which is on the minus strand, presenting the opportunity for the alleles to be easily confused.   The C>G change in the gene is represented by a G>C change on the positive chromosomal strand.; (PMID: 18357466).\n",
      "Gene: TP53; Drug(s): cyclophosphamide, doxorubicin, fluorouracil; Statement: Genotype GG is associated with increased risk of Vomiting when treated with cyclophosphamide, doxorubicin and fluorouracil in women with Breast Neoplasms as compared to genotypes CC + CG.; Notes: Recurrent grade 1-3 vomiting. Paper states \"In our study homozygote variant CC (Pro/Pro) predisposed to recurrent vomiting grade 13. \" This gene is on the minus chromosomal stand, alleles in annotation have been complemented to plus chromosomal strand.; (PMID: 29507678).Notes: Recurrent grade 1-3 vomiting. Paper states \"In our study homozygote variant CC (Pro/Pro) predisposed to recurrent vomiting grade 13. \" This gene is on the minus chromosomal stand, alleles in annotation have been complemented to plus chromosomal strand.; (PMID: 29507678).\n",
      "Gene: TP53; Drug(s): cyclophosphamide, fluorouracil; Statement: Allele G is associated with increased likelihood of Drug Toxicity when treated with cyclophosphamide and fluorouracil in people with Breast Neoplasms as compared to allele C.; Notes: (Full text not available); Association reported for 72Pro (interpreted as C on coding strand, gene is on minus strand, and therefore risk allele is G on plus strand as shown above); (PMID: 20638924).Notes: (Full text not available); Association reported for 72Pro (interpreted as C on coding strand, gene is on minus strand, and therefore risk allele is G on plus strand as shown above); (PMID: 20638924).\n",
      "Gene: TP53; Drug(s): cyclophosphamide, epirubicin, fluorouracil; Statement: Allele T is associated with decreased risk of Neutropenia due to cyclophosphamide, epirubicin and fluorouracil in women with Breast Neoplasms as compared to allele C.; Notes: Association is designated as nominal as correction for multiple comparisons was not carried out.; (PMID: 29593529).Notes: Association is designated as nominal as correction for multiple comparisons was not carried out.; (PMID: 29593529).\n",
      "Gene: TP53; Drug(s): carboplatin, cisplatin; Statement: Allele C is not associated with increased severity of Drug Toxicity when treated with carboplatin or cisplatin in people with Lung Neoplasms as compared to allele G.; Notes: \"no significant association observed between grade 1-2 or grade 3-4 platinum-based chemotherapy induced toxicities like anemia and ...TP53 codon 72 genetic polymorphisms (Arg/Pro: OR=0.64, 95% CI=0.34-1.17, p>0.05; Pro/Pro: OR=0.46, 95% CI=0.15-1.42, p>0.05 and Arg/Pro+Pro/Pro: OR=0.62, 95% CI=0.34-1.15, p>0.05)\"; (PMID: 34967559).Notes: \"no significant association observed between grade 1-2 or grade 3-4 platinum-based chemotherapy induced toxicities like anemia and ...TP53 codon 72 genetic polymorphisms (Arg/Pro: OR=0.64, 95% CI=0.34-1.17, p>0.05; Pro/Pro: OR=0.46, 95% CI=0.15-1.42, p>0.05 and Arg/Pro+Pro/Pro: OR=0.62, 95% CI=0.34-1.15, p>0.05)\"; (PMID: 34967559).\n",
      "Gene: TP53; Drug(s): cisplatin, cyclophosphamide; Statement: Genotype GG is associated with increased severity of Neutropenia when treated with cisplatin and cyclophosphamide in people with Ovarian Neoplasms as compared to genotypes CC + CG.; Notes: nan; (PMID: 19786980).Notes: nan; (PMID: 19786980).\n"
     ]
    }
   ],
   "source": [
    "pharmakb_knowledge = pharmakb_analyzer.verbalise_pharmakb_knowledge()\n",
    "print(pharmakb_knowledge)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
