{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import importlib\n",
    "sys.path.append(r'C:\\Users\\owysocky\\Documents\\GitHub\\CCE_scGeneRAI\\source\\investigate_and_compare')\n",
    "import civicdb_gene_info\n",
    "importlib.reload(civicdb_gene_info)\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "id",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "name",
         "rawType": "object",
         "type": "string"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "467800e7-85fb-4549-a8e8-fc40db51a61a",
       "rows": [
        [
         "0",
         "1",
         "ALK"
        ],
        [
         "1",
         "2",
         "AKT1"
        ],
        [
         "2",
         "3",
         "ARAF"
        ],
        [
         "3",
         "4",
         "ABL1"
        ],
        [
         "4",
         "5",
         "BRAF"
        ],
        [
         "5",
         "6",
         "BRCA1"
        ],
        [
         "6",
         "7",
         "BRCA2"
        ],
        [
         "7",
         "8",
         "CCND1"
        ],
        [
         "8",
         "9",
         "CCND2"
        ],
        [
         "9",
         "10",
         "CCND3"
        ],
        [
         "10",
         "11",
         "CCNE1"
        ],
        [
         "11",
         "12",
         "CDK6"
        ],
        [
         "12",
         "13",
         "CDK4"
        ],
        [
         "13",
         "14",
         "CDKN2A"
        ],
        [
         "14",
         "15",
         "CEBPA"
        ],
        [
         "15",
         "18",
         "DNMT3A"
        ],
        [
         "16",
         "19",
         "EGFR"
        ],
        [
         "17",
         "20",
         "ERBB2"
        ],
        [
         "18",
         "21",
         "ESR1"
        ],
        [
         "19",
         "22",
         "FGFR2"
        ],
        [
         "20",
         "23",
         "FGFR3"
        ],
        [
         "21",
         "24",
         "FLT3"
        ],
        [
         "22",
         "25",
         "GATA2"
        ],
        [
         "23",
         "26",
         "IDH1"
        ],
        [
         "24",
         "27",
         "IDH2"
        ],
        [
         "25",
         "28",
         "JAK2"
        ],
        [
         "26",
         "29",
         "KIT"
        ],
        [
         "27",
         "30",
         "KRAS"
        ],
        [
         "28",
         "31",
         "MAP2K1"
        ],
        [
         "29",
         "32",
         "MEF2D"
        ],
        [
         "30",
         "33",
         "CSF1R"
        ],
        [
         "31",
         "34",
         "MGMT"
        ],
        [
         "32",
         "35",
         "NPM1"
        ],
        [
         "33",
         "36",
         "NRAS"
        ],
        [
         "34",
         "37",
         "PIK3CA"
        ],
        [
         "35",
         "38",
         "PDGFRA"
        ],
        [
         "36",
         "39",
         "PML"
        ],
        [
         "37",
         "41",
         "PTEN"
        ],
        [
         "38",
         "42",
         "RET"
        ],
        [
         "39",
         "43",
         "RUNX1"
        ],
        [
         "40",
         "44",
         "SF3B1"
        ],
        [
         "41",
         "45",
         "TP53"
        ],
        [
         "42",
         "46",
         "TSC1"
        ],
        [
         "43",
         "47",
         "TSC2"
        ],
        [
         "44",
         "48",
         "U2AF1"
        ],
        [
         "45",
         "49",
         "WT1"
        ],
        [
         "46",
         "50",
         "NOTCH1"
        ],
        [
         "47",
         "51",
         "DDR2"
        ],
        [
         "48",
         "52",
         "MET"
        ],
        [
         "49",
         "54",
         "EWSR1"
        ]
       ],
       "shape": {
        "columns": 2,
        "rows": 590
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ALK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>AKT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ARAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ABL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>32083</td>\n",
       "      <td>KLLN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>33873</td>\n",
       "      <td>MEF2B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>41522</td>\n",
       "      <td>PRNCR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>56400</td>\n",
       "      <td>COX2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>56415</td>\n",
       "      <td>ND1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>590 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        id    name\n",
       "0        1     ALK\n",
       "1        2    AKT1\n",
       "2        3    ARAF\n",
       "3        4    ABL1\n",
       "4        5    BRAF\n",
       "..     ...     ...\n",
       "585  32083    KLLN\n",
       "586  33873   MEF2B\n",
       "587  41522  PRNCR1\n",
       "588  56400    COX2\n",
       "589  56415     ND1\n",
       "\n",
       "[590 rows x 2 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load gene info\n",
    "path = r'C:\\Users\\owysocky\\Documents\\GitHub\\CCE_scGeneRAI\\resources\\civicdb'\n",
    "file_name = 'civic_all_features'\n",
    "\n",
    "gene_ids = pd.read_csv(path + '\\\\' + file_name + '.csv')[['id','name']]\n",
    "gene_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieved gene info for gene ID 1\n",
      "Retrieved gene info for gene ID 2\n",
      "Retrieved gene info for gene ID 3\n",
      "Retrieved gene info for gene ID 4\n",
      "Retrieved gene info for gene ID 5\n",
      "Retrieved gene info for gene ID 6\n",
      "Retrieved gene info for gene ID 7\n",
      "Retrieved gene info for gene ID 8\n",
      "Retrieved gene info for gene ID 9\n",
      "Retrieved gene info for gene ID 10\n",
      "Retrieved gene info for gene ID 11\n",
      "Retrieved gene info for gene ID 12\n",
      "Retrieved gene info for gene ID 13\n",
      "Retrieved gene info for gene ID 14\n",
      "Retrieved gene info for gene ID 15\n",
      "Retrieved gene info for gene ID 18\n",
      "Retrieved gene info for gene ID 19\n",
      "Retrieved gene info for gene ID 20\n",
      "Retrieved gene info for gene ID 21\n",
      "Retrieved gene info for gene ID 22\n",
      "Retrieved gene info for gene ID 23\n",
      "Retrieved gene info for gene ID 24\n",
      "Retrieved gene info for gene ID 25\n",
      "Retrieved gene info for gene ID 26\n",
      "Retrieved gene info for gene ID 27\n",
      "Retrieved gene info for gene ID 28\n",
      "Retrieved gene info for gene ID 29\n",
      "Retrieved gene info for gene ID 30\n",
      "Retrieved gene info for gene ID 31\n",
      "Retrieved gene info for gene ID 32\n",
      "Retrieved gene info for gene ID 33\n",
      "Retrieved gene info for gene ID 34\n",
      "Retrieved gene info for gene ID 35\n",
      "Retrieved gene info for gene ID 36\n",
      "Retrieved gene info for gene ID 37\n",
      "Retrieved gene info for gene ID 38\n",
      "Retrieved gene info for gene ID 39\n",
      "Retrieved gene info for gene ID 41\n",
      "Retrieved gene info for gene ID 42\n",
      "Retrieved gene info for gene ID 43\n",
      "Retrieved gene info for gene ID 44\n",
      "Retrieved gene info for gene ID 45\n",
      "Retrieved gene info for gene ID 46\n",
      "Retrieved gene info for gene ID 47\n",
      "Retrieved gene info for gene ID 48\n",
      "Retrieved gene info for gene ID 49\n",
      "Retrieved gene info for gene ID 50\n",
      "Retrieved gene info for gene ID 51\n",
      "Retrieved gene info for gene ID 52\n",
      "Retrieved gene info for gene ID 54\n",
      "Retrieved gene info for gene ID 55\n",
      "Retrieved gene info for gene ID 56\n",
      "Retrieved gene info for gene ID 57\n",
      "Retrieved gene info for gene ID 58\n",
      "Retrieved gene info for gene ID 59\n",
      "Retrieved gene info for gene ID 60\n",
      "Retrieved gene info for gene ID 61\n",
      "Retrieved gene info for gene ID 62\n",
      "Retrieved gene info for gene ID 63\n",
      "Retrieved gene info for gene ID 64\n",
      "Retrieved gene info for gene ID 65\n",
      "Retrieved gene info for gene ID 66\n",
      "Retrieved gene info for gene ID 67\n",
      "Retrieved gene info for gene ID 68\n",
      "Retrieved gene info for gene ID 69\n",
      "Retrieved gene info for gene ID 70\n",
      "Retrieved gene info for gene ID 71\n",
      "Retrieved gene info for gene ID 72\n",
      "Retrieved gene info for gene ID 73\n",
      "Retrieved gene info for gene ID 74\n",
      "Retrieved gene info for gene ID 75\n",
      "Retrieved gene info for gene ID 76\n",
      "Retrieved gene info for gene ID 77\n",
      "Retrieved gene info for gene ID 78\n",
      "Retrieved gene info for gene ID 79\n",
      "Retrieved gene info for gene ID 80\n",
      "Retrieved gene info for gene ID 101\n",
      "Retrieved gene info for gene ID 128\n",
      "Retrieved gene info for gene ID 130\n",
      "Retrieved gene info for gene ID 150\n",
      "Retrieved gene info for gene ID 154\n",
      "Retrieved gene info for gene ID 199\n",
      "Retrieved gene info for gene ID 221\n",
      "Retrieved gene info for gene ID 254\n",
      "Retrieved gene info for gene ID 259\n",
      "Retrieved gene info for gene ID 353\n",
      "Retrieved gene info for gene ID 355\n",
      "Retrieved gene info for gene ID 375\n",
      "Retrieved gene info for gene ID 389\n",
      "Retrieved gene info for gene ID 399\n",
      "Retrieved gene info for gene ID 436\n",
      "Retrieved gene info for gene ID 446\n",
      "Retrieved gene info for gene ID 450\n",
      "Retrieved gene info for gene ID 519\n",
      "Retrieved gene info for gene ID 524\n",
      "Retrieved gene info for gene ID 525\n",
      "Retrieved gene info for gene ID 533\n",
      "Retrieved gene info for gene ID 537\n",
      "Retrieved gene info for gene ID 549\n",
      "Retrieved gene info for gene ID 550\n",
      "Retrieved gene info for gene ID 566\n",
      "Retrieved gene info for gene ID 567\n",
      "Retrieved gene info for gene ID 570\n",
      "Retrieved gene info for gene ID 588\n",
      "Retrieved gene info for gene ID 593\n",
      "Retrieved gene info for gene ID 594\n",
      "Retrieved gene info for gene ID 608\n",
      "Retrieved gene info for gene ID 625\n",
      "Retrieved gene info for gene ID 634\n",
      "Retrieved gene info for gene ID 638\n",
      "Retrieved gene info for gene ID 732\n",
      "Retrieved gene info for gene ID 761\n",
      "Retrieved gene info for gene ID 778\n",
      "Retrieved gene info for gene ID 779\n",
      "Retrieved gene info for gene ID 826\n",
      "Retrieved gene info for gene ID 827\n",
      "Retrieved gene info for gene ID 855\n",
      "Retrieved gene info for gene ID 859\n",
      "Retrieved gene info for gene ID 864\n",
      "Retrieved gene info for gene ID 868\n",
      "Retrieved gene info for gene ID 888\n",
      "Retrieved gene info for gene ID 906\n",
      "Retrieved gene info for gene ID 908\n",
      "Retrieved gene info for gene ID 912\n",
      "Retrieved gene info for gene ID 913\n",
      "Retrieved gene info for gene ID 914\n",
      "Retrieved gene info for gene ID 916\n",
      "Retrieved gene info for gene ID 927\n",
      "Retrieved gene info for gene ID 981\n",
      "Retrieved gene info for gene ID 1193\n",
      "Retrieved gene info for gene ID 1204\n",
      "Retrieved gene info for gene ID 1234\n",
      "Retrieved gene info for gene ID 1239\n",
      "Retrieved gene info for gene ID 1278\n",
      "Retrieved gene info for gene ID 1283\n",
      "Retrieved gene info for gene ID 1285\n",
      "Retrieved gene info for gene ID 1290\n",
      "Retrieved gene info for gene ID 1342\n",
      "Retrieved gene info for gene ID 1390\n",
      "Retrieved gene info for gene ID 1396\n",
      "Retrieved gene info for gene ID 1411\n",
      "Retrieved gene info for gene ID 1417\n",
      "Retrieved gene info for gene ID 1428\n",
      "Retrieved gene info for gene ID 1463\n",
      "Retrieved gene info for gene ID 1510\n",
      "Retrieved gene info for gene ID 1526\n",
      "Retrieved gene info for gene ID 1536\n",
      "Retrieved gene info for gene ID 1559\n",
      "Retrieved gene info for gene ID 1563\n",
      "Retrieved gene info for gene ID 1567\n",
      "Retrieved gene info for gene ID 1590\n",
      "Retrieved gene info for gene ID 1603\n",
      "Retrieved gene info for gene ID 1627\n",
      "Retrieved gene info for gene ID 1634\n",
      "Retrieved gene info for gene ID 1646\n",
      "Retrieved gene info for gene ID 1651\n",
      "Retrieved gene info for gene ID 1655\n",
      "Retrieved gene info for gene ID 1701\n",
      "Retrieved gene info for gene ID 1704\n",
      "Retrieved gene info for gene ID 1705\n",
      "Retrieved gene info for gene ID 1719\n",
      "Retrieved gene info for gene ID 1721\n",
      "Retrieved gene info for gene ID 1728\n",
      "Retrieved gene info for gene ID 1733\n",
      "Retrieved gene info for gene ID 1734\n",
      "Retrieved gene info for gene ID 1735\n",
      "Retrieved gene info for gene ID 1736\n",
      "Retrieved gene info for gene ID 1737\n",
      "Retrieved gene info for gene ID 1741\n",
      "Retrieved gene info for gene ID 1742\n",
      "Retrieved gene info for gene ID 1753\n",
      "Retrieved gene info for gene ID 1762\n",
      "Retrieved gene info for gene ID 1763\n",
      "Retrieved gene info for gene ID 1764\n",
      "Retrieved gene info for gene ID 1767\n",
      "Retrieved gene info for gene ID 1768\n",
      "Retrieved gene info for gene ID 1769\n",
      "Retrieved gene info for gene ID 1771\n",
      "Retrieved gene info for gene ID 1810\n",
      "Retrieved gene info for gene ID 1811\n",
      "Retrieved gene info for gene ID 1827\n",
      "Retrieved gene info for gene ID 1842\n",
      "Retrieved gene info for gene ID 1843\n",
      "Retrieved gene info for gene ID 1844\n",
      "Retrieved gene info for gene ID 1872\n",
      "Retrieved gene info for gene ID 1873\n",
      "Retrieved gene info for gene ID 1883\n",
      "Retrieved gene info for gene ID 1885\n",
      "Retrieved gene info for gene ID 1887\n",
      "Retrieved gene info for gene ID 1911\n",
      "Retrieved gene info for gene ID 1920\n",
      "Retrieved gene info for gene ID 1925\n",
      "Retrieved gene info for gene ID 1926\n",
      "Retrieved gene info for gene ID 1937\n",
      "Retrieved gene info for gene ID 1938\n",
      "Retrieved gene info for gene ID 1947\n",
      "Retrieved gene info for gene ID 1950\n",
      "Retrieved gene info for gene ID 1955\n",
      "Retrieved gene info for gene ID 1956\n",
      "Retrieved gene info for gene ID 2073\n",
      "Retrieved gene info for gene ID 2186\n",
      "Retrieved gene info for gene ID 2188\n",
      "Retrieved gene info for gene ID 2189\n",
      "Retrieved gene info for gene ID 2239\n",
      "Retrieved gene info for gene ID 2308\n",
      "Retrieved gene info for gene ID 2312\n",
      "Retrieved gene info for gene ID 2317\n",
      "Retrieved gene info for gene ID 2319\n",
      "Retrieved gene info for gene ID 2438\n",
      "Retrieved gene info for gene ID 2473\n",
      "Retrieved gene info for gene ID 2478\n",
      "Retrieved gene info for gene ID 2526\n",
      "Retrieved gene info for gene ID 2527\n",
      "Retrieved gene info for gene ID 2529\n",
      "Retrieved gene info for gene ID 2535\n",
      "Retrieved gene info for gene ID 2537\n",
      "Retrieved gene info for gene ID 2538\n",
      "Retrieved gene info for gene ID 2556\n",
      "Retrieved gene info for gene ID 2578\n",
      "Retrieved gene info for gene ID 2591\n",
      "Retrieved gene info for gene ID 2593\n",
      "Retrieved gene info for gene ID 2596\n",
      "Retrieved gene info for gene ID 2606\n",
      "Retrieved gene info for gene ID 2607\n",
      "Retrieved gene info for gene ID 2608\n",
      "Retrieved gene info for gene ID 2622\n",
      "Retrieved gene info for gene ID 2657\n",
      "Retrieved gene info for gene ID 2677\n",
      "Retrieved gene info for gene ID 2682\n",
      "Retrieved gene info for gene ID 2715\n",
      "Retrieved gene info for gene ID 2723\n",
      "Retrieved gene info for gene ID 2747\n",
      "Retrieved gene info for gene ID 2778\n",
      "Retrieved gene info for gene ID 2784\n",
      "Retrieved gene info for gene ID 2822\n",
      "Retrieved gene info for gene ID 2836\n",
      "Retrieved gene info for gene ID 2840\n",
      "Retrieved gene info for gene ID 2888\n",
      "Retrieved gene info for gene ID 2899\n",
      "Retrieved gene info for gene ID 2967\n",
      "Retrieved gene info for gene ID 2970\n",
      "Retrieved gene info for gene ID 2975\n",
      "Retrieved gene info for gene ID 3017\n",
      "Retrieved gene info for gene ID 3043\n",
      "Retrieved gene info for gene ID 3083\n",
      "Retrieved gene info for gene ID 3090\n",
      "Retrieved gene info for gene ID 3091\n",
      "Retrieved gene info for gene ID 3094\n",
      "Retrieved gene info for gene ID 3153\n",
      "Retrieved gene info for gene ID 3215\n",
      "Retrieved gene info for gene ID 3236\n",
      "Retrieved gene info for gene ID 3255\n",
      "Retrieved gene info for gene ID 3274\n",
      "Retrieved gene info for gene ID 3342\n",
      "Retrieved gene info for gene ID 3359\n",
      "Retrieved gene info for gene ID 3362\n",
      "Retrieved gene info for gene ID 3364\n",
      "Retrieved gene info for gene ID 3375\n",
      "Retrieved gene info for gene ID 3376\n",
      "Retrieved gene info for gene ID 3389\n",
      "Retrieved gene info for gene ID 3448\n",
      "Retrieved gene info for gene ID 3465\n",
      "Retrieved gene info for gene ID 3466\n",
      "Retrieved gene info for gene ID 3475\n",
      "Retrieved gene info for gene ID 3480\n",
      "Retrieved gene info for gene ID 3485\n",
      "Retrieved gene info for gene ID 3512\n",
      "Retrieved gene info for gene ID 3527\n",
      "Retrieved gene info for gene ID 3529\n",
      "Retrieved gene info for gene ID 3532\n",
      "Retrieved gene info for gene ID 3537\n",
      "Retrieved gene info for gene ID 3549\n",
      "Retrieved gene info for gene ID 3553\n",
      "Retrieved gene info for gene ID 3564\n",
      "Retrieved gene info for gene ID 3577\n",
      "Retrieved gene info for gene ID 3586\n",
      "Retrieved gene info for gene ID 3628\n",
      "Retrieved gene info for gene ID 3629\n",
      "Retrieved gene info for gene ID 3659\n",
      "Retrieved gene info for gene ID 3672\n",
      "Retrieved gene info for gene ID 3730\n",
      "Retrieved gene info for gene ID 3731\n",
      "Retrieved gene info for gene ID 3737\n",
      "Retrieved gene info for gene ID 3738\n",
      "Retrieved gene info for gene ID 3741\n",
      "Retrieved gene info for gene ID 3742\n",
      "Retrieved gene info for gene ID 3777\n",
      "Retrieved gene info for gene ID 3801\n",
      "Retrieved gene info for gene ID 3852\n",
      "Retrieved gene info for gene ID 3867\n",
      "Retrieved gene info for gene ID 3870\n",
      "Retrieved gene info for gene ID 3878\n",
      "Retrieved gene info for gene ID 3885\n",
      "Retrieved gene info for gene ID 3887\n",
      "Retrieved gene info for gene ID 3935\n",
      "Retrieved gene info for gene ID 3936\n",
      "Retrieved gene info for gene ID 3976\n",
      "Retrieved gene info for gene ID 3983\n",
      "Retrieved gene info for gene ID 3984\n",
      "Retrieved gene info for gene ID 3985\n",
      "Retrieved gene info for gene ID 4046\n",
      "Retrieved gene info for gene ID 4103\n",
      "Retrieved gene info for gene ID 4111\n",
      "Retrieved gene info for gene ID 4123\n",
      "Retrieved gene info for gene ID 4137\n",
      "Retrieved gene info for gene ID 4173\n",
      "Retrieved gene info for gene ID 4176\n",
      "Retrieved gene info for gene ID 4244\n",
      "Retrieved gene info for gene ID 4285\n",
      "Retrieved gene info for gene ID 4286\n",
      "Retrieved gene info for gene ID 4289\n",
      "Retrieved gene info for gene ID 4290\n",
      "Retrieved gene info for gene ID 4326\n",
      "Retrieved gene info for gene ID 4327\n",
      "Retrieved gene info for gene ID 4337\n",
      "Retrieved gene info for gene ID 4371\n",
      "Retrieved gene info for gene ID 4384\n",
      "Retrieved gene info for gene ID 4386\n",
      "Retrieved gene info for gene ID 4409\n",
      "Retrieved gene info for gene ID 4411\n",
      "Retrieved gene info for gene ID 4419\n",
      "Retrieved gene info for gene ID 4463\n",
      "Retrieved gene info for gene ID 4465\n",
      "Retrieved gene info for gene ID 4503\n",
      "Retrieved gene info for gene ID 4511\n",
      "Retrieved gene info for gene ID 4517\n",
      "Retrieved gene info for gene ID 4529\n",
      "Retrieved gene info for gene ID 4532\n",
      "Retrieved gene info for gene ID 4542\n",
      "Retrieved gene info for gene ID 4546\n",
      "Retrieved gene info for gene ID 4566\n",
      "Retrieved gene info for gene ID 4616\n",
      "Retrieved gene info for gene ID 4630\n",
      "Retrieved gene info for gene ID 4645\n",
      "Retrieved gene info for gene ID 4659\n",
      "Retrieved gene info for gene ID 4682\n",
      "Retrieved gene info for gene ID 4685\n",
      "Retrieved gene info for gene ID 4686\n",
      "Retrieved gene info for gene ID 4690\n",
      "Retrieved gene info for gene ID 4692\n",
      "Retrieved gene info for gene ID 4695\n",
      "Retrieved gene info for gene ID 4753\n",
      "Retrieved gene info for gene ID 4760\n",
      "Retrieved gene info for gene ID 4762\n",
      "Retrieved gene info for gene ID 4763\n",
      "Retrieved gene info for gene ID 4765\n",
      "Retrieved gene info for gene ID 4767\n",
      "Retrieved gene info for gene ID 4791\n",
      "Retrieved gene info for gene ID 4795\n",
      "Retrieved gene info for gene ID 4868\n",
      "Retrieved gene info for gene ID 4875\n",
      "Retrieved gene info for gene ID 4941\n",
      "Retrieved gene info for gene ID 5013\n",
      "Retrieved gene info for gene ID 5051\n",
      "Retrieved gene info for gene ID 5052\n",
      "Retrieved gene info for gene ID 5087\n",
      "Retrieved gene info for gene ID 5112\n",
      "Retrieved gene info for gene ID 5128\n",
      "Retrieved gene info for gene ID 5167\n",
      "Retrieved gene info for gene ID 5176\n",
      "Retrieved gene info for gene ID 5177\n",
      "Retrieved gene info for gene ID 5200\n",
      "Retrieved gene info for gene ID 5210\n",
      "Retrieved gene info for gene ID 5225\n",
      "Retrieved gene info for gene ID 5354\n",
      "Retrieved gene info for gene ID 5356\n",
      "Retrieved gene info for gene ID 5365\n",
      "Retrieved gene info for gene ID 5409\n",
      "Retrieved gene info for gene ID 5418\n",
      "Retrieved gene info for gene ID 5499\n",
      "Retrieved gene info for gene ID 5514\n",
      "Retrieved gene info for gene ID 5516\n",
      "Retrieved gene info for gene ID 5519\n",
      "Retrieved gene info for gene ID 5520\n",
      "Retrieved gene info for gene ID 5525\n",
      "Retrieved gene info for gene ID 5534\n",
      "Retrieved gene info for gene ID 5572\n",
      "Retrieved gene info for gene ID 5581\n",
      "Retrieved gene info for gene ID 5627\n",
      "Retrieved gene info for gene ID 5646\n",
      "Retrieved gene info for gene ID 5649\n",
      "Retrieved gene info for gene ID 5654\n",
      "Retrieved gene info for gene ID 5717\n",
      "Retrieved gene info for gene ID 5723\n",
      "Retrieved gene info for gene ID 5728\n",
      "Retrieved gene info for gene ID 5731\n",
      "Retrieved gene info for gene ID 5737\n",
      "Retrieved gene info for gene ID 5742\n",
      "Retrieved gene info for gene ID 5752\n",
      "Retrieved gene info for gene ID 5761\n",
      "Retrieved gene info for gene ID 5777\n",
      "Retrieved gene info for gene ID 5814\n",
      "Retrieved gene info for gene ID 5828\n",
      "Retrieved gene info for gene ID 5845\n",
      "Retrieved gene info for gene ID 5848\n",
      "Retrieved gene info for gene ID 5852\n",
      "Retrieved gene info for gene ID 5870\n",
      "Retrieved gene info for gene ID 5876\n",
      "Retrieved gene info for gene ID 5939\n",
      "Retrieved gene info for gene ID 5970\n",
      "Retrieved gene info for gene ID 5971\n",
      "Retrieved gene info for gene ID 5972\n",
      "Retrieved gene info for gene ID 5988\n",
      "Retrieved gene info for gene ID 6009\n",
      "Retrieved gene info for gene ID 6023\n",
      "Retrieved gene info for gene ID 6054\n",
      "Retrieved gene info for gene ID 6071\n",
      "Retrieved gene info for gene ID 6073\n",
      "Retrieved gene info for gene ID 6104\n",
      "Retrieved gene info for gene ID 6107\n",
      "Retrieved gene info for gene ID 6119\n",
      "Retrieved gene info for gene ID 6143\n",
      "Retrieved gene info for gene ID 6144\n",
      "Retrieved gene info for gene ID 6146\n",
      "Retrieved gene info for gene ID 6152\n",
      "Retrieved gene info for gene ID 6281\n",
      "Retrieved gene info for gene ID 6317\n",
      "Retrieved gene info for gene ID 6329\n",
      "Retrieved gene info for gene ID 6340\n",
      "Retrieved gene info for gene ID 6380\n",
      "Retrieved gene info for gene ID 6394\n",
      "Retrieved gene info for gene ID 6415\n",
      "Retrieved gene info for gene ID 6523\n",
      "Retrieved gene info for gene ID 6534\n",
      "Retrieved gene info for gene ID 6538\n",
      "Retrieved gene info for gene ID 6559\n",
      "Retrieved gene info for gene ID 6572\n",
      "Retrieved gene info for gene ID 6590\n",
      "Retrieved gene info for gene ID 6592\n",
      "Retrieved gene info for gene ID 6614\n",
      "Retrieved gene info for gene ID 6652\n",
      "Retrieved gene info for gene ID 6676\n",
      "Retrieved gene info for gene ID 6772\n",
      "Retrieved gene info for gene ID 6856\n",
      "Retrieved gene info for gene ID 6862\n",
      "Retrieved gene info for gene ID 6906\n",
      "Retrieved gene info for gene ID 6945\n",
      "Retrieved gene info for gene ID 7009\n",
      "Retrieved gene info for gene ID 7014\n",
      "Retrieved gene info for gene ID 7024\n",
      "Retrieved gene info for gene ID 7041\n",
      "Retrieved gene info for gene ID 7044\n",
      "Retrieved gene info for gene ID 7074\n",
      "Retrieved gene info for gene ID 7084\n",
      "Retrieved gene info for gene ID 7222\n",
      "Retrieved gene info for gene ID 7282\n",
      "Retrieved gene info for gene ID 7297\n",
      "Retrieved gene info for gene ID 7339\n",
      "Retrieved gene info for gene ID 7351\n",
      "Retrieved gene info for gene ID 7356\n",
      "Retrieved gene info for gene ID 7411\n",
      "Retrieved gene info for gene ID 7451\n",
      "Retrieved gene info for gene ID 7502\n",
      "Retrieved gene info for gene ID 7608\n",
      "Retrieved gene info for gene ID 7704\n",
      "Retrieved gene info for gene ID 7728\n",
      "Retrieved gene info for gene ID 7777\n",
      "Retrieved gene info for gene ID 7780\n",
      "Retrieved gene info for gene ID 7909\n",
      "Retrieved gene info for gene ID 7936\n",
      "Retrieved gene info for gene ID 7948\n",
      "Retrieved gene info for gene ID 7953\n",
      "Retrieved gene info for gene ID 7954\n",
      "Retrieved gene info for gene ID 8032\n",
      "Retrieved gene info for gene ID 8160\n",
      "Retrieved gene info for gene ID 8197\n",
      "Retrieved gene info for gene ID 8253\n",
      "Retrieved gene info for gene ID 8262\n",
      "Retrieved gene info for gene ID 8292\n",
      "Retrieved gene info for gene ID 8331\n",
      "Retrieved gene info for gene ID 8351\n",
      "Retrieved gene info for gene ID 8407\n",
      "Retrieved gene info for gene ID 8450\n",
      "Retrieved gene info for gene ID 8508\n",
      "Retrieved gene info for gene ID 8516\n",
      "Retrieved gene info for gene ID 8524\n",
      "Retrieved gene info for gene ID 8552\n",
      "Retrieved gene info for gene ID 8553\n",
      "Retrieved gene info for gene ID 8614\n",
      "Retrieved gene info for gene ID 8787\n",
      "Retrieved gene info for gene ID 8813\n",
      "Retrieved gene info for gene ID 8878\n",
      "Retrieved gene info for gene ID 8908\n",
      "Retrieved gene info for gene ID 8950\n",
      "Retrieved gene info for gene ID 8996\n",
      "Retrieved gene info for gene ID 9054\n",
      "Retrieved gene info for gene ID 9121\n",
      "Retrieved gene info for gene ID 9171\n",
      "Retrieved gene info for gene ID 9189\n",
      "Retrieved gene info for gene ID 9216\n",
      "Retrieved gene info for gene ID 9236\n",
      "Retrieved gene info for gene ID 9261\n",
      "Retrieved gene info for gene ID 9326\n",
      "Retrieved gene info for gene ID 9427\n",
      "Retrieved gene info for gene ID 9533\n",
      "Retrieved gene info for gene ID 9538\n",
      "Retrieved gene info for gene ID 9588\n",
      "Retrieved gene info for gene ID 9705\n",
      "Retrieved gene info for gene ID 9722\n",
      "Retrieved gene info for gene ID 9935\n",
      "Retrieved gene info for gene ID 9957\n",
      "Retrieved gene info for gene ID 10024\n",
      "Retrieved gene info for gene ID 10239\n",
      "Retrieved gene info for gene ID 10573\n",
      "Retrieved gene info for gene ID 11261\n",
      "Retrieved gene info for gene ID 11289\n",
      "Retrieved gene info for gene ID 11314\n",
      "Retrieved gene info for gene ID 11322\n",
      "Retrieved gene info for gene ID 11335\n",
      "Retrieved gene info for gene ID 11492\n",
      "Retrieved gene info for gene ID 11587\n",
      "Retrieved gene info for gene ID 11761\n",
      "Retrieved gene info for gene ID 11781\n",
      "Retrieved gene info for gene ID 11900\n",
      "Retrieved gene info for gene ID 11932\n",
      "Retrieved gene info for gene ID 12074\n",
      "Retrieved gene info for gene ID 12112\n",
      "Retrieved gene info for gene ID 12121\n",
      "Retrieved gene info for gene ID 12133\n",
      "Retrieved gene info for gene ID 12146\n",
      "Retrieved gene info for gene ID 12271\n",
      "Retrieved gene info for gene ID 12367\n",
      "Retrieved gene info for gene ID 12422\n",
      "Retrieved gene info for gene ID 12555\n",
      "Retrieved gene info for gene ID 12754\n",
      "Retrieved gene info for gene ID 12881\n",
      "Retrieved gene info for gene ID 12903\n",
      "Retrieved gene info for gene ID 13005\n",
      "Retrieved gene info for gene ID 13180\n",
      "Retrieved gene info for gene ID 13185\n",
      "Retrieved gene info for gene ID 13276\n",
      "Retrieved gene info for gene ID 13465\n",
      "Retrieved gene info for gene ID 13556\n",
      "Retrieved gene info for gene ID 13648\n",
      "Retrieved gene info for gene ID 13783\n",
      "Retrieved gene info for gene ID 13880\n",
      "Retrieved gene info for gene ID 13948\n",
      "Retrieved gene info for gene ID 13968\n",
      "Retrieved gene info for gene ID 14079\n",
      "Retrieved gene info for gene ID 14089\n",
      "Retrieved gene info for gene ID 14160\n",
      "Retrieved gene info for gene ID 14195\n",
      "Retrieved gene info for gene ID 14284\n",
      "Retrieved gene info for gene ID 14303\n",
      "Retrieved gene info for gene ID 14368\n",
      "Retrieved gene info for gene ID 14525\n",
      "Retrieved gene info for gene ID 14591\n",
      "Retrieved gene info for gene ID 14842\n",
      "Retrieved gene info for gene ID 14923\n",
      "Retrieved gene info for gene ID 14995\n",
      "Retrieved gene info for gene ID 15004\n",
      "Retrieved gene info for gene ID 15013\n",
      "Retrieved gene info for gene ID 15315\n",
      "Retrieved gene info for gene ID 15318\n",
      "Retrieved gene info for gene ID 15344\n",
      "Retrieved gene info for gene ID 15403\n",
      "Retrieved gene info for gene ID 15676\n",
      "Retrieved gene info for gene ID 15955\n",
      "Retrieved gene info for gene ID 16160\n",
      "Retrieved gene info for gene ID 16180\n",
      "Retrieved gene info for gene ID 16394\n",
      "Retrieved gene info for gene ID 16516\n",
      "Retrieved gene info for gene ID 16656\n",
      "Retrieved gene info for gene ID 16661\n",
      "Retrieved gene info for gene ID 16802\n",
      "Retrieved gene info for gene ID 16872\n",
      "Retrieved gene info for gene ID 17016\n",
      "Retrieved gene info for gene ID 17340\n",
      "Retrieved gene info for gene ID 17939\n",
      "Retrieved gene info for gene ID 18353\n",
      "Retrieved gene info for gene ID 18364\n",
      "Retrieved gene info for gene ID 18420\n",
      "Retrieved gene info for gene ID 18440\n",
      "Retrieved gene info for gene ID 19838\n",
      "Retrieved gene info for gene ID 19896\n",
      "Retrieved gene info for gene ID 19959\n",
      "Retrieved gene info for gene ID 20168\n",
      "Retrieved gene info for gene ID 20480\n",
      "Retrieved gene info for gene ID 21380\n",
      "Retrieved gene info for gene ID 21634\n",
      "Retrieved gene info for gene ID 22231\n",
      "Retrieved gene info for gene ID 24374\n",
      "Retrieved gene info for gene ID 24929\n",
      "Retrieved gene info for gene ID 24938\n",
      "Retrieved gene info for gene ID 26599\n",
      "Retrieved gene info for gene ID 32083\n",
      "Retrieved gene info for gene ID 33873\n",
      "Retrieved gene info for gene ID 41522\n",
      "Retrieved gene info for gene ID 56400\n",
      "Retrieved gene info for gene ID 56415\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "fullName",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "name",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "gene_id",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "summary",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "Protein_Domains",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "a70b961f-4449-4422-9543-2dc37024eba6",
       "rows": [
        [
         "0",
         "ALK receptor tyrosine kinase",
         "ALK",
         "1",
         "This gene encodes a receptor tyrosine kinase, which belongs to the insulin receptor superfamily. This protein comprises an extracellular domain, an hydrophobic stretch corresponding to a single pass transmembrane region, and an intracellular kinase domain. It plays an important role in the development of the brain and exerts its effects on specific neurons in the nervous system. This gene has been found to be rearranged, mutated, or amplified in a series of tumours including anaplastic large cell lymphomas, neuroblastoma, and non-small cell lung cancer. The chromosomal rearrangements are the most common genetic alterations in this gene, which result in creation of multiple fusion genes in tumourigenesis, including ALK (chromosome 2)/EML4 (chromosome 2), ALK/RANBP2 (chromosome 2), ALK/ATIC (chromosome 2), ALK/TFG (chromosome 3), ALK/NPM1 (chromosome 5), ALK/SQSTM1 (chromosome 5), ALK/KIF5B (chromosome 10), ALK/CLTC (chromosome 17), ALK/TPM4 (chromosome 19), and ALK/MSN (chromosome X).[provided by RefSeq, Jan 2011].",
         "['Protein kinase domain', 'MAM domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class II, conserved site', 'Low-density lipoprotein (LDL) receptor class A repeat', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Concanavalin A-like lectin/glucanase domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'LDL receptor-like superfamily', 'Receptor Tyrosine Kinase']"
        ],
        [
         "1",
         "AKT serine/threonine kinase 1",
         "AKT1",
         "2",
         "This gene encodes one of the three members of the human AKT serine-threonine protein kinase family which are often referred to as protein kinase B alpha, beta, and gamma. These highly similar AKT proteins all have an N-terminal pleckstrin homology domain, a serine/threonine-specific kinase domain and a C-terminal regulatory domain. These proteins are phosphorylated by phosphoinositide 3-kinase (PI3K). AKT/PI3K forms a key component of many signalling pathways that involve the binding of membrane-bound ligands such as receptor tyrosine kinases, G-protein coupled receptors, and integrin-linked kinase. These AKT proteins therefore regulate a wide variety of cellular functions including cell proliferation, survival, metabolism, and angiogenesis in both normal and malignant cells. AKT proteins are recruited to the cell membrane by phosphatidylinositol 3,4,5-trisphosphate (PIP3) after phosphorylation of phosphatidylinositol 4,5-bisphosphate (PIP2) by PI3K. Subsequent phosphorylation of both threonine residue 308 and serine residue 473 is required for full activation of the AKT1 protein encoded by this gene. Phosphorylation of additional residues also occurs, for example, in response to insulin growth factor-1 and epidermal growth factor. Protein phosphatases act as negative regulators of AKT proteins by dephosphorylating AKT or PIP3. The PI3K/AKT signalling pathway is crucial for tumor cell survival. Survival factors can suppress apoptosis in a transcription-independent manner by activating AKT1 which then phosphorylates and inactivates components of the apoptotic machinery. AKT proteins also participate in the mammalian target of rapamycin (mTOR) signalling pathway which controls the assembly of the eukaryotic translation initiation factor 4F (eIF4E) complex and this pathway, in addition to responding to extracellular signals from growth factors and cytokines, is disregulated in many cancers. Mutations in this gene are associated with multiple types of cancer and excessive tissue growth including Proteus syndrome and Cowden syndrome 6, and breast, colorectal, and ovarian cancers. Multiple alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Jul 2020].",
         "['Protein kinase domain', 'AGC-kinase, C-terminal', 'Pleckstrin homology domain', 'Serine/threonine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'PH-like domain superfamily', 'Protein kinase, ATP binding site', 'Protein kinase, C-terminal', 'Protein kinase B alpha, catalytic domain', 'Protein Kinase B, pleckstrin homology domain']"
        ],
        [
         "2",
         "A-Raf proto-oncogene, serine/threonine kinase",
         "ARAF",
         "3",
         "This proto-oncogene belongs to the RAF subfamily of the Ser/Thr protein kinase family, and maybe involved in cell growth and development. Alternatively spliced transcript variants encoding different isoforms have been found for this gene.[provided by RefSeq, Jan 2012]",
         "['Protein kinase C-like, phorbol ester/diacylglycerol-binding domain', 'Raf-like Ras-binding', 'Diacylglycerol/phorbol-ester binding', 'Ubiquitin-like domain superfamily', 'C1-like domain superfamily', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Serine/threonine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Serine/threonine-protein kinase']"
        ],
        [
         "3",
         "ABL proto-oncogene 1, non-receptor tyrosine kinase",
         "ABL1",
         "4",
         "This gene is a protooncogene that encodes a protein tyrosine kinase involved in a variety of cellular processes, including cell division, adhesion, differentiation, and response to stress. The activity of the protein is negatively regulated by its SH3 domain, whereby deletion of the region encoding this domain results in an oncogene. The ubiquitously expressed protein has DNA-binding activity that is regulated by CDC2-mediated phosphorylation, suggesting a cell cycle function. This gene has been found fused to a variety of translocation partner genes in various leukemias, most notably the t(9;22) translocation that results in a fusion with the 5' end of the breakpoint cluster region gene (BCR; MIM:151410). Alternative splicing of this gene results in two transcript variants, which contain alternative first exons that are spliced to the remaining common exons. [provided by RefSeq, Aug 2014].",
         "['Protein kinase domain', 'SH2 domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'SH3 domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'F-actin binding', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase ABL, SH2 domain', 'SH3-like domain superfamily', 'SH2 domain superfamily', 'Non-receptor tyrosine kinases involved in cell signaling']"
        ],
        [
         "4",
         "B-Raf proto-oncogene, serine/threonine kinase",
         "BRAF",
         "5",
         "This gene encodes a protein belonging to the RAF family of serine/threonine protein kinases. This protein plays a role in regulating the MAP kinase/ERK signaling pathway, which affects cell division, differentiation, and secretion. Mutations in this gene, most commonly the V600E mutation, are the most frequently identified cancer-causing mutations in melanoma, and have been identified in various other cancers as well, including non-Hodgkin lymphoma, colorectal cancer, thyroid carcinoma, non-small cell lung carcinoma, hairy cell leukemia and adenocarcinoma of lung. Mutations in this gene are also associated with cardiofaciocutaneous, Noonan, and Costello syndromes, which exhibit overlapping phenotypes. A pseudogene of this gene has been identified on the X chromosome. [provided by RefSeq, Aug 2017].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Protein kinase C-like, phorbol ester/diacylglycerol-binding domain', 'Raf-like Ras-binding', 'Serine/threonine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Diacylglycerol/phorbol-ester binding', 'Ubiquitin-like domain superfamily', 'C1-like domain superfamily', 'Serine/threonine-protein kinase']"
        ],
        [
         "5",
         "BRCA1 DNA repair associated",
         "BRCA1",
         "6",
         "This gene encodes a 190 kD nuclear phosphoprotein that plays a role in maintaining genomic stability, and it also acts as a tumor suppressor. The BRCA1 gene contains 22 exons spanning about 110 kb of DNA. The encoded protein combines with other tumor suppressors, DNA damage sensors, and signal transducers to form a large multi-subunit protein complex known as the BRCA1-associated genome surveillance complex (BASC). This gene product associates with RNA polymerase II, and through the C-terminal domain, also interacts with histone deacetylase complexes. This protein thus plays a role in transcription, DNA repair of double-stranded breaks, and recombination. Mutations in this gene are responsible for approximately 40% of inherited breast cancers and more than 80% of inherited breast and ovarian cancers. Alternative splicing plays a role in modulating the subcellular localization and physiological function of this gene. Many alternatively spliced transcript variants, some of which are disease-associated mutations, have been described for this gene, but the full-length natures of only some of these variants has been described. A related pseudogene, which is also located on chromosome 17, has been identified. [provided by RefSeq, May 2020].",
         "['BRCT domain', 'Zinc finger, RING-type', 'Breast cancer type 1 susceptibility protein (BRCA1)', 'Zinc finger, RING/FYVE/PHD-type', 'Zinc finger, RING-type, conserved site', 'Zinc finger, C3HC4 RING-type', 'Zinc finger, RING-type, eukaryotic', 'BRCA1-associated', 'BRCT domain superfamily', 'BRCA1, serine-rich domain']"
        ],
        [
         "6",
         "BRCA2 DNA repair associated",
         "BRCA2",
         "7",
         "Inherited mutations in BRCA1 and this gene, BRCA2, confer increased lifetime risk of developing breast or ovarian cancer. Both BRCA1 and BRCA2 are involved in maintenance of genome stability, specifically the homologous recombination pathway for double-strand DNA repair. The largest exon in both genes is exon 11, which harbors the most important and frequent mutations in breast cancer patients. The BRCA2 gene was found on chromosome 13q12.3 in human. The BRCA2 protein contains several copies of a 70 aa motif called the BRC motif, and these motifs mediate binding to the RAD51 recombinase which functions in DNA repair. BRCA2 is considered a tumor suppressor gene, as tumors with BRCA2 mutations generally exhibit loss of heterozygosity (LOH) of the wild-type allele. [provided by RefSeq, May 2020].",
         "['BRCA2 repeat', 'Nucleic acid-binding, OB-fold', 'BRCA2, OB1', 'BRCA2, OB3', 'Tower domain', 'Breast cancer type 2 susceptibility protein, helical domain', 'Breast cancer type 2 susceptibility protein', 'BRCA2 helical domain superfamily', 'BRCA2, OB2 domain']"
        ],
        [
         "7",
         "cyclin D1",
         "CCND1",
         "8",
         "The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance throughout the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with and functions as a regulatory subunit of CDK4 or CDK6, whose activity is required for cell cycle G1/S transition. This protein has been shown to interact with tumor suppressor protein Rb and the expression of this gene is regulated positively by Rb. Mutations, amplification and overexpression of this gene, which alters cell cycle progression, are observed frequently in a variety of human cancers. [provided by RefSeq, Dec 2019].",
         "['Cyclin, C-terminal domain', 'Cyclin, N-terminal', 'Cyclin-like domain', 'Cyclin-like superfamily', 'Cyclin', 'Cyclins, cyclin-box']"
        ],
        [
         "8",
         "cyclin D2",
         "CCND2",
         "9",
         "The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance through the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with CDK4 or CDK6 and functions as a regulatory subunit of the complex, whose activity is required for cell cycle G1/S transition. This protein has been shown to interact with and be involved in the phosphorylation of tumor suppressor protein Rb. Knockout studies of the homologous gene in mouse suggest the essential roles of this gene in ovarian granulosa and germ cell proliferation. High level expression of this gene was observed in ovarian and testicular tumors. Mutations in this gene are associated with megalencephaly-polymicrogyria-polydactyly-hydrocephalus syndrome 3 (MPPH3). [provided by RefSeq, Sep 2014].",
         "['Cyclin, C-terminal domain', 'Cyclin, N-terminal', 'Cyclin-like domain', 'Cyclin-like superfamily', 'Cyclin', 'Cyclins, cyclin-box']"
        ],
        [
         "9",
         "cyclin D3",
         "CCND3",
         "10",
         "The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance through the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with and functions as a regulatory subunit of CDK4 or CDK6, whose activtiy is required for cell cycle G1/S transition. This protein has been shown to interact with and be involved in the phosphorylation of tumor suppressor protein Rb. The CDK4 activity associated with this cyclin was reported to be necessary for cell cycle progression through G2 phase into mitosis after UV radiation. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2008].",
         "['Cyclin, N-terminal', 'Cyclin-like superfamily', 'Cyclin', 'Cyclin, C-terminal domain', 'Cyclin-like domain', 'Cyclins, cyclin-box']"
        ],
        [
         "10",
         "cyclin E1",
         "CCNE1",
         "11",
         "The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance through the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with and functions as a regulatory subunit of CDK2, whose activity is required for cell cycle G1/S transition. This protein accumulates at the G1-S phase boundary and is degraded as cells progress through S phase. Overexpression of this gene has been observed in many tumors, which results in chromosome instability, and thus may contribute to tumorigenesis. This protein was found to associate with, and be involved in, the phosphorylation of NPAT protein (nuclear protein mapped to the ATM locus), which participates in cell-cycle regulated histone gene expression and plays a critical role in promoting cell-cycle progression in the absence of pRB. [provided by RefSeq, Apr 2016].",
         "['Cyclin, C-terminal domain', 'Cyclin, N-terminal', 'Cyclin-like domain', 'Cyclin-like superfamily', 'Cyclin', 'Cyclins, cyclin-box']"
        ],
        [
         "11",
         "cyclin dependent kinase 6",
         "CDK6",
         "12",
         "The protein encoded by this gene is a member of the CMGC family of serine/threonine protein kinases. This kinase is a catalytic subunit of the protein kinase complex that is important for cell cycle G1 phase progression and G1/S transition. The activity of this kinase first appears in mid-G1 phase, which is controlled by the regulatory subunits including D-type cyclins and members of INK4 family of CDK inhibitors. This kinase, as well as CDK4, has been shown to phosphorylate, and thus regulate the activity of, tumor suppressor protein Rb. Altered expression of this gene has been observed in multiple human cancers. A mutation in this gene resulting in reduced cell proliferation, and impaired cell motility and polarity, and has been identified in patients with primary microcephaly. [provided by RefSeq, Aug 2017].",
         "['Protein kinase domain', 'Serine/threonine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Cyclin-dependent kinase 6', 'Cyclin-dependent kinase']"
        ],
        [
         "12",
         "cyclin dependent kinase 4",
         "CDK4",
         "13",
         "The protein encoded by this gene is a member of the Ser/Thr protein kinase family. This protein is highly similar to the gene products of S. cerevisiae cdc28 and S. pombe cdc2. It is a catalytic subunit of the protein kinase complex that is important for cell cycle G1 phase progression. The activity of this kinase is restricted to the G1-S phase, which is controlled by the regulatory subunits D-type cyclins and CDK inhibitor p16(INK4a). This kinase was shown to be responsible for the phosphorylation of retinoblastoma gene product (Rb). Mutations in this gene as well as in its related proteins including D-type cyclins, p16(INK4a) and Rb were all found to be associated with tumorigenesis of a variety of cancers. Multiple polyadenylation sites of this gene have been reported. [provided by RefSeq, Jul 2008].",
         "['Protein kinase domain', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Cyclin-dependent kinase', 'Serine/threonine-protein kinase, active site']"
        ],
        [
         "13",
         "cyclin dependent kinase inhibitor 2A",
         "CDKN2A",
         "14",
         "This gene generates several transcript variants which differ in their first exons. At least three alternatively spliced variants encoding distinct proteins have been reported, two of which encode structurally related isoforms known to function as inhibitors of CDK4 kinase. The remaining transcript includes an alternate first exon located 20 Kb upstream of the remainder of the gene; this transcript contains an alternate open reading frame (ARF) that specifies a protein which is structurally unrelated to the products of the other variants. This ARF product functions as a stabilizer of the tumor suppressor protein p53 as it can interact with, and sequester, the E3 ubiquitin-protein ligase MDM2, a protein responsible for the degradation of p53. In spite of the structural and functional differences, the CDK inhibitor isoforms and the ARF product encoded by this gene, through the regulatory roles of CDK4 and p53 in cell cycle G1 progression, share a common functionality in cell cycle G1 control. This gene is frequently mutated or deleted in a wide variety of tumors, and is known to be an important tumor suppressor gene. [provided by RefSeq, Sep 2012].",
         "['Ankyrin repeat-containing domain superfamily', 'Ankyrin Repeat and Cyclin-Dependent Kinase Inhibitor', 'Tumor suppressor ARF']"
        ],
        [
         "14",
         "CCAAT enhancer binding protein alpha",
         "CEBPA",
         "15",
         "This intronless gene encodes a transcription factor that contains a basic leucine zipper (bZIP) domain and recognizes the CCAAT motif in the promoters of target genes. The encoded protein functions in homodimers and also heterodimers with CCAAT/enhancer-binding proteins beta and gamma. Activity of this protein can modulate the expression of genes involved in cell cycle regulation as well as in body weight homeostasis. Mutation of this gene is associated with acute myeloid leukemia. The use of alternative in-frame non-AUG (GUG) and AUG start codons results in protein isoforms with different lengths. Differential translation initiation is mediated by an out-of-frame, upstream open reading frame which is located between the GUG and the first AUG start codons. [provided by RefSeq, Dec 2013].",
         "['Basic-leucine zipper domain', 'CCAAT/enhancer-binding protein, chordates', 'CCAAT/enhancer-binding protein C/EBP', 'Basic-leucine zipper domain superfamily']"
        ],
        [
         "15",
         "DNA methyltransferase 3 alpha",
         "DNMT3A",
         "18",
         "CpG methylation is an epigenetic modification that is important for embryonic development, imprinting, and X-chromosome inactivation. Studies in mice have demonstrated that DNA methylation is required for mammalian development. This gene encodes a DNA methyltransferase that is thought to function in de novo methylation, rather than maintenance methylation. The protein localizes to the cytoplasm and nucleus and its expression is developmentally regulated. [provided by RefSeq, Mar 2016].",
         "['PWWP domain', 'C-5 cytosine methyltransferase', 'DNA methylase, C-5 cytosine-specific, active site', 'ADD domain', 'S-adenosyl-L-methionine-dependent methyltransferase superfamily', 'DNMT3, cysteine rich ADD domain, GATA1-like zinc finger', 'DNA (cytosine-5)-methyltransferase 3A, ADD domain', 'DNMT3, ADD domain, PHD zinc finger']"
        ],
        [
         "16",
         "epidermal growth factor receptor",
         "EGFR",
         "19",
         "The protein encoded by this gene is a transmembrane glycoprotein that is a member of the protein kinase superfamily. This protein is a receptor for members of the epidermal growth factor family. EGFR is a cell surface protein that binds to epidermal growth factor, thus inducing receptor dimerization and tyrosine autophosphorylation leading to cell proliferation. Mutations in this gene are associated with lung cancer. EGFR is a component of the cytokine storm which contributes to a severe form of Coronavirus Disease 2019 (COVID-19) resulting from infection with severe acute respiratory syndrome coronavirus-2 (SARS-CoV-2). [provided by RefSeq, Jul 2020].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Furin-like repeat', 'Tyrosine-protein kinase, active site', 'Growth factor receptor cysteine-rich domain superfamily', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Growth factor receptor domain 4', 'Epidermal growth factor receptor-like, transmembrane-juxtamembrane segment', 'Receptor Tyrosine Kinase', 'Receptor L-domain', 'Furin-like cysteine-rich domain', 'Tyrosine protein kinase, EGF/ERB/XmrK receptor', 'Receptor L-domain superfamily']"
        ],
        [
         "17",
         "erb-b2 receptor tyrosine kinase 2",
         "ERBB2",
         "20",
         "This gene encodes a member of the epidermal growth factor (EGF) receptor family of receptor tyrosine kinases. This protein has no ligand binding domain of its own and therefore cannot bind growth factors. However, it does bind tightly to other ligand-bound EGF receptor family members to form a heterodimer, stabilizing ligand binding and enhancing kinase-mediated activation of downstream signalling pathways, such as those involving mitogen-activated protein kinase and phosphatidylinositol-3 kinase. Allelic variations at amino acid positions 654 and 655 of isoform a (positions 624 and 625 of isoform b) have been reported, with the most common allele, Ile654/Ile655, shown here. Amplification and/or overexpression of this gene has been reported in numerous cancers, including breast and ovarian tumors. Alternative splicing results in several additional transcript variants, some encoding different isoforms and others that have not been fully characterized. [provided by RefSeq, Jul 2008].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Epidermal growth factor receptor-like, transmembrane-juxtamembrane segment', 'Receptor Tyrosine Kinase', 'Furin-like cysteine-rich domain', 'Furin-like repeat', 'Growth factor receptor cysteine-rich domain superfamily', 'Growth factor receptor domain 4', 'Receptor L-domain superfamily', 'Receptor L-domain', 'Tyrosine protein kinase, EGF/ERB/XmrK receptor']"
        ],
        [
         "18",
         "estrogen receptor 1",
         "ESR1",
         "21",
         "This gene encodes an estrogen receptor and ligand-activated transcription factor. The canonical protein contains an N-terminal ligand-independent transactivation domain, a central DNA binding domain, a hinge domain, and a C-terminal ligand-dependent transactivation domain. The protein localizes to the nucleus where it may form either a homodimer or a heterodimer with estrogen receptor 2. The protein encoded by this gene regulates the transcription of many estrogen-inducible genes that play a role in growth, metabolism, sexual development, gestation, and other reproductive functions and is expressed in many non-reproductive tissues. The receptor encoded by this gene plays a key role in breast cancer, endometrial cancer, and osteoporosis. This gene is reported to have dozens of transcript variants due to the use of alternate promoters and alternative splicing, however, the full-length nature of many of these variants remain uncertain. [provided by RefSeq, Jul 2020].",
         "['Estrogen receptor', 'Estrogen receptor, N-terminal', 'Nuclear hormone receptor, ligand-binding domain', 'Zinc finger, nuclear hormone receptor-type', 'Nuclear hormone receptor', 'Zinc finger, NHR/GATA-type', 'Estrogen receptor/oestrogen-related receptor', 'Oestrogen-type nuclear receptor final C-terminal domain', 'Nuclear hormone receptor-like domain superfamily', 'Nuclear hormone receptor family NR3 subfamily']"
        ],
        [
         "19",
         "fibroblast growth factor receptor 2",
         "FGFR2",
         "22",
         "The protein encoded by this gene is a member of the fibroblast growth factor receptor family, where amino acid sequence is highly conserved between members and throughout evolution. FGFR family members differ from one another in their ligand affinities and tissue distribution. A full-length representative protein consists of an extracellular region, composed of three immunoglobulin-like domains, a single hydrophobic membrane-spanning segment and a cytoplasmic tyrosine kinase domain. The extracellular portion of the protein interacts with fibroblast growth factors, setting in motion a cascade of downstream signals, ultimately influencing mitogenesis and differentiation. This particular family member is a high-affinity receptor for acidic, basic and/or keratinocyte growth factor, depending on the isoform. Mutations in this gene are associated with Crouzon syndrome, Pfeiffer syndrome, Craniosynostosis, Apert syndrome, Jackson-Weiss syndrome, Beare-Stevenson cutis gyrata syndrome, Saethre-Chotzen syndrome, and syndromic craniosynostosis. Multiple alternatively spliced transcript variants encoding different isoforms have been noted for this gene. [provided by RefSeq, Jan 2009].",
         "['Immunoglobulin subtype 2', 'Immunoglobulin subtype', 'Immunoglobulin-like domain', 'Immunoglobulin-like fold', 'Immunoglobulin-like domain superfamily', 'Immunoglobulin I-set', 'Fibroblast growth factor receptor-like', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Fibroblast growth factor receptor family', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Receptor Tyrosine Kinase']"
        ],
        [
         "20",
         "fibroblast growth factor receptor 3",
         "FGFR3",
         "23",
         "This gene encodes a member of the fibroblast growth factor receptor (FGFR) family, with its amino acid sequence being highly conserved between members and among divergent species. FGFR family members differ from one another in their ligand affinities and tissue distribution. A full-length representative protein would consist of an extracellular region, composed of three immunoglobulin-like domains, a single hydrophobic membrane-spanning segment and a cytoplasmic tyrosine kinase domain. The extracellular portion of the protein interacts with fibroblast growth factors, setting in motion a cascade of downstream signals, ultimately influencing mitogenesis and differentiation. This particular family member binds acidic and basic fibroblast growth hormone and plays a role in bone development and maintenance. Mutations in this gene lead to craniosynostosis and multiple types of skeletal dysplasia. [provided by RefSeq, Aug 2017].",
         "['Immunoglobulin subtype 2', 'Immunoglobulin-like domain', 'Immunoglobulin I-set', 'Immunoglobulin-like fold', 'Immunoglobulin-like domain superfamily', 'Fibroblast growth factor receptor-like', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Immunoglobulin subtype', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Fibroblast growth factor receptor family', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Receptor Tyrosine Kinase']"
        ],
        [
         "21",
         "fms related receptor tyrosine kinase 3",
         "FLT3",
         "24",
         "This gene encodes a class III receptor tyrosine kinase that regulates hematopoiesis. This receptor is activated by binding of the fms-related tyrosine kinase 3 ligand to the extracellular domain, which induces homodimer formation in the plasma membrane leading to autophosphorylation of the receptor. The activated receptor kinase subsequently phosphorylates and activates multiple cytoplasmic effector molecules in pathways involved in apoptosis, proliferation, and differentiation of hematopoietic cells in bone marrow. Mutations that result in the constitutive activation of this receptor result in acute myeloid leukemia and acute lymphoblastic leukemia. [provided by RefSeq, Jan 2015].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class III, conserved site', 'Immunoglobulin-like domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Immunoglobulin-like beta-sandwich domain', 'Immunoglobulin-like fold', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Immunoglobulin-like domain superfamily', 'Receptor Tyrosine Kinase']"
        ],
        [
         "22",
         "GATA binding protein 2",
         "GATA2",
         "25",
         "This gene encodes a member of the GATA family of zinc-finger transcription factors that are named for the consensus nucleotide sequence they bind in the promoter regions of target genes. The encoded protein plays an essential role in regulating transcription of genes involved in the development and proliferation of hematopoietic and endocrine cell lineages. Alternative splicing results in multiple transcript variants.[provided by RefSeq, Mar 2009].",
         "['Transcription factor GATA', 'Zinc finger, GATA-type', 'Zinc finger, NHR/GATA-type', 'Transcription factor, GATA-2/3']"
        ],
        [
         "23",
         "isocitrate dehydrogenase (NADP(+)) 1",
         "IDH1",
         "26",
         "Isocitrate dehydrogenases catalyze the oxidative decarboxylation of isocitrate to 2-oxoglutarate. These enzymes belong to two distinct subclasses, one of which utilizes NAD(+) as the electron acceptor and the other NADP(+). Five isocitrate dehydrogenases have been reported: three NAD(+)-dependent isocitrate dehydrogenases, which localize to the mitochondrial matrix, and two NADP(+)-dependent isocitrate dehydrogenases, one of which is mitochondrial and the other predominantly cytosolic. Each NADP(+)-dependent isozyme is a homodimer. The protein encoded by this gene is the NADP(+)-dependent isocitrate dehydrogenase found in the cytoplasm and peroxisomes. It contains the PTS-1 peroxisomal targeting signal sequence. The presence of this enzyme in peroxisomes suggests roles in the regeneration of NADPH for intraperoxisomal reductions, such as the conversion of 2, 4-dienoyl-CoAs to 3-enoyl-CoAs, as well as in peroxisomal reactions that consume 2-oxoglutarate, namely the alpha-hydroxylation of phytanic acid. The cytoplasmic enzyme serves a significant role in cytoplasmic NADPH production. Alternatively spliced transcript variants encoding the same protein have been found for this gene. [provided by RefSeq, Sep 2013].",
         "['Isocitrate dehydrogenase NADP-dependent', 'Isopropylmalate dehydrogenase-like domain', 'Isocitrate/isopropylmalate dehydrogenase, conserved site']"
        ],
        [
         "24",
         "isocitrate dehydrogenase (NADP(+)) 2",
         "IDH2",
         "27",
         "Isocitrate dehydrogenases catalyze the oxidative decarboxylation of isocitrate to 2-oxoglutarate. These enzymes belong to two distinct subclasses, one of which utilizes NAD(+) as the electron acceptor and the other NADP(+). Five isocitrate dehydrogenases have been reported: three NAD(+)-dependent isocitrate dehydrogenases, which localize to the mitochondrial matrix, and two NADP(+)-dependent isocitrate dehydrogenases, one of which is mitochondrial and the other predominantly cytosolic. Each NADP(+)-dependent isozyme is a homodimer. The protein encoded by this gene is the NADP(+)-dependent isocitrate dehydrogenase found in the mitochondria. It plays a role in intermediary metabolism and energy production. This protein may tightly associate or interact with the pyruvate dehydrogenase complex. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2014].",
         "['Isocitrate dehydrogenase NADP-dependent', 'Isocitrate/isopropylmalate dehydrogenase, conserved site', 'Isopropylmalate dehydrogenase-like domain']"
        ],
        [
         "25",
         "Janus kinase 2",
         "JAK2",
         "28",
         "This gene encodes a non-receptor tyrosine kinase that plays a central role in cytokine and growth factor signalling. The primary isoform of this protein has an N-terminal FERM domain that is required for erythropoietin receptor association, an SH2 domain that binds STAT transcription factors, a pseudokinase domain and a C-terminal tyrosine kinase domain. Cytokine binding induces autophosphorylation and activation of this kinase. This kinase then recruits and phosphorylates signal transducer and activator of transcription (STAT) proteins. Growth factors like TGF-beta 1 also induce phosphorylation and activation of this kinase and translocation of downstream STAT proteins to the nucleus where they influence gene transcription. Mutations in this gene are associated with numerous inflammatory diseases and malignancies. This gene is a downstream target of the pleiotropic cytokine IL6 that is produced by B cells, T cells, dendritic cells and macrophages to produce an immune response or inflammation. Disregulation of the IL6/JAK2/STAT3 signalling pathways produces increased cellular proliferation and myeloproliferative neoplasms of hematopoietic stem cells. A nonsynonymous mutation in the pseudokinase domain of this gene disrupts the domains inhibitory effect and results in constitutive tyrosine phosphorylation activity and hypersensitivity to cytokine signalling. This gene and the IL6/JAK2/STAT3 signalling pathway is a therapeutic target for the treatment of excessive inflammatory responses to viral infections. Alternative splicing results in multiple transcript variants encoding distinct isoforms. [provided by RefSeq, Jul 2020].",
         "['FERM domain', 'Protein kinase domain', 'SH2 domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'PH-like domain superfamily', 'Tyrosine-protein kinase, non-receptor Jak/Tyk2', 'Protein kinase, ATP binding site', 'FERM central domain', 'Band 4.1 domain', 'Tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, non-receptor Jak2', 'Janus kinase 2, pseudokinase domain', 'Janus kinase 2, catalytic domain', 'Tyrosine-protein kinase JAK2, SH2 domain', 'FERM superfamily, second domain', 'SH2 domain superfamily', 'JAK2, FERM domain C-lobe', 'JAK, FERM F2 lobe domain', 'FERM F1 lobe ubiquitin-like domain', 'JAK1-3/TYK2, pleckstrin homology-like domain', 'Janus Kinase (JAK)']"
        ],
        [
         "26",
         "KIT proto-oncogene, receptor tyrosine kinase",
         "KIT",
         "29",
         "This gene encodes a receptor tyrosine kinase. This gene was initially identified as a homolog of the feline sarcoma viral oncogene v-kit and is often referred to as proto-oncogene c-Kit. The canonical form of this glycosylated transmembrane protein has an N-terminal extracellular region with five immunoglobulin-like domains, a transmembrane region, and an intracellular tyrosine kinase domain at the C-terminus. Upon activation by its cytokine ligand, stem cell factor (SCF), this protein phosphorylates multiple intracellular proteins that play a role in in the proliferation, differentiation, migration and apoptosis of many cell types and thereby plays an important role in hematopoiesis, stem cell maintenance, gametogenesis, melanogenesis, and in mast cell development, migration and function. This protein can be a membrane-bound or soluble protein. Mutations in this gene are associated with gastrointestinal stromal tumors, mast cell disease, acute myelogenous leukemia, and piebaldism. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, May 2020].",
         "['Immunoglobulin-like beta-sandwich domain', 'Immunoglobulin-like fold', 'Immunoglobulin-like domain superfamily', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class III, conserved site', 'Immunoglobulin subtype 2', 'Immunoglobulin subtype', 'Immunoglobulin-like domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Mast/stem cell growth factor receptor', 'Receptor Tyrosine Kinase']"
        ],
        [
         "27",
         "KRAS proto-oncogene, GTPase",
         "KRAS",
         "30",
         "This gene, a Kirsten ras oncogene homolog from the mammalian ras gene family, encodes a protein that is a member of the small GTPase superfamily. A single amino acid substitution is responsible for an activating mutation. The transforming protein that results is implicated in various malignancies, including lung adenocarcinoma, mucinous adenoma, ductal carcinoma of the pancreas and colorectal carcinoma. Alternative splicing leads to variants encoding two isoforms that differ in the C-terminal region. [provided by RefSeq, Jul 2008].",
         "['Small GTPase', 'Small GTPase, Ras-type', 'P-loop containing nucleoside triphosphate hydrolase', 'Small GTP-binding protein domain']"
        ],
        [
         "28",
         "mitogen-activated protein kinase kinase 1",
         "MAP2K1",
         "31",
         "The protein encoded by this gene is a member of the dual specificity protein kinase family, which acts as a mitogen-activated protein (MAP) kinase kinase. MAP kinases, also known as extracellular signal-regulated kinases (ERKs), act as an integration point for multiple biochemical signals. This protein kinase lies upstream of MAP kinases and stimulates the enzymatic activity of MAP kinases upon wide variety of extra- and intracellular signals. As an essential component of MAP kinase signal transduction pathway, this kinase is involved in many cellular processes such as proliferation, differentiation, transcription regulation and development. [provided by RefSeq, Jul 2008].",
         "['Protein kinase domain', 'Protein kinase-like domain superfamily', 'MAP kinase kinase', 'Serine/threonine-protein kinase, active site', 'Protein kinase, ATP binding site']"
        ],
        [
         "29",
         "myocyte enhancer factor 2D",
         "MEF2D",
         "32",
         "This gene is a member of the myocyte-specific enhancer factor 2 (MEF2) family of transcription factors. Members of this family are involved in control of muscle and neuronal cell differentiation and development, and are regulated by class II histone deacetylases. Fusions of the encoded protein with Deleted in Azoospermia-Associated Protein 1 (DAZAP1) due to a translocation have been found in an acute lymphoblastic leukemia cell line, suggesting a role in leukemogenesis. The encoded protein may also be involved in Parkinson disease and myotonic dystrophy. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Oct 2012].",
         "['Transcription factor, MADS-box', 'Holliday junction regulator protein family C-terminal', 'MADS MEF2-like', 'Transcription factor, MADS-box superfamily', 'MADS-box/MEF2 Transcription Factor']"
        ],
        [
         "30",
         "colony stimulating factor 1 receptor",
         "CSF1R",
         "33",
         "The protein encoded by this gene is the receptor for colony stimulating factor 1, a cytokine which controls the production, differentiation, and function of macrophages. This receptor mediates most if not all of the biological effects of this cytokine. Ligand binding activates the receptor kinase through a process of oligomerization and transphosphorylation. The encoded protein is a tyrosine kinase transmembrane receptor and member of the CSF1/PDGF receptor family of tyrosine-protein kinases. Mutations in this gene have been associated with a predisposition to myeloid malignancy. The first intron of this gene contains a transcriptionally inactive ribosomal protein L7 processed pseudogene oriented in the opposite direction. Alternative splicing results in multiple transcript variants. Expression of a splice variant from an LTR promoter has been found in Hodgkin lymphoma (HL), HL cell lines and anaplastic large cell lymphoma. [provided by RefSeq, Mar 2017].",
         "['Immunoglobulin-like fold', 'Immunoglobulin subtype 2', 'Immunoglobulin subtype', 'Immunoglobulin-like domain', 'Immunoglobulin-like beta-sandwich domain', 'Immunoglobulin-like domain superfamily', 'Platelet-derived growth factor receptor-like protein', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class III, conserved site', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Macrophage colony-stimulating factor 1 receptor', 'Receptor Tyrosine Kinase']"
        ],
        [
         "31",
         "O-6-methylguanine-DNA methyltransferase",
         "MGMT",
         "34",
         "Alkylating agents are potent carcinogens that can result in cell death, mutation and cancer. The protein encoded by this gene is a DNA repair protein that is involved in cellular defense against mutagenesis and toxicity from alkylating agents. The protein catalyzes transfer of methyl groups from O(6)-alkylguanine and other methylated moieties of the DNA to its own molecule, which repairs the toxic lesions. Methylation of the genes promoter has been associated with several cancer types, including colorectal cancer, lung cancer, lymphoma and glioblastoma. [provided by RefSeq, Sep 2015].",
         "['Methylated-DNA-[protein]-cysteine S-methyltransferase, active site', 'Methylguanine DNA methyltransferase, ribonuclease-like domain', 'Methylated-DNA-[protein]-cysteine S-methyltransferase, DNA binding', 'Methylated DNA-protein cysteine methyltransferase, DNA binding domain', 'Winged helix-like DNA-binding domain superfamily', 'Methylated DNA-protein cysteine methyltransferase domain superfamily']"
        ],
        [
         "32",
         "nucleophosmin 1",
         "NPM1",
         "35",
         "The protein encoded by this gene is involved in several cellular processes, including centrosome duplication, protein chaperoning, and cell proliferation. The encoded phosphoprotein shuttles between the nucleolus, nucleus, and cytoplasm, chaperoning ribosomal proteins and core histones from the nucleus to the cytoplasm. This protein is also known to sequester the tumor suppressor ARF in the nucleolus, protecting it from degradation until it is needed. Mutations in this gene are associated with acute myeloid leukemia. Dozens of pseudogenes of this gene have been identified. [provided by RefSeq, Aug 2017].",
         "['Nucleoplasmin family', 'Nucleoplasmin core domain', 'Nucleophosmin, C-terminal', 'Nucleoplasmin core domain superfamily']"
        ],
        [
         "33",
         "NRAS proto-oncogene, GTPase",
         "NRAS",
         "36",
         "This is an N-ras oncogene encoding a membrane protein that shuttles between the Golgi apparatus and the plasma membrane. This shuttling is regulated through palmitoylation and depalmitoylation by the ZDHHC9-GOLGA7 complex. The encoded protein, which has intrinsic GTPase activity, is activated by a guanine nucleotide-exchange factor and inactivated by a GTPase activating protein. Mutations in this gene have been associated with somatic rectal cancer, follicular thyroid cancer, autoimmune lymphoproliferative syndrome, Noonan syndrome, and juvenile myelomonocytic leukemia. [provided by RefSeq, Jun 2011].",
         "['Small GTPase', 'Small GTP-binding protein domain', 'Small GTPase, Ras-type', 'P-loop containing nucleoside triphosphate hydrolase']"
        ],
        [
         "34",
         "phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit alpha",
         "PIK3CA",
         "37",
         "Phosphatidylinositol 3-kinase is composed of an 85 kDa regulatory subunit and a 110 kDa catalytic subunit. The protein encoded by this gene represents the catalytic subunit, which uses ATP to phosphorylate PtdIns, PtdIns4P and PtdIns(4,5)P2. This gene has been found to be oncogenic and has been implicated in cervical cancers. A pseudogene of this gene has been defined on chromosome 22. [provided by RefSeq, Apr 2016].",
         "['Phosphatidylinositol 3-kinase Ras-binding (PI3K RBD) domain', 'Phosphatidylinositol 3-/4-kinase, catalytic domain', 'Phosphoinositide 3-kinase, accessory (PIK) domain', 'C2 phosphatidylinositol 3-kinase-type domain', 'Phosphatidylinositol 3-kinase, adaptor-binding domain', 'Protein kinase-like domain superfamily', 'Phosphatidylinositol kinase', 'Armadillo-type fold', 'Phosphatidylinositol 3/4-kinase, conserved site', 'Ubiquitin-like domain superfamily', 'C2 domain superfamily', 'Phosphatidylinositol 3-/4-kinase, catalytic domain superfamily', 'PI3Kalpha, catalytic domain', 'Phosphoinositide 3-kinase, accessory (PIK) domain superfamily']"
        ],
        [
         "35",
         "platelet derived growth factor receptor alpha",
         "PDGFRA",
         "38",
         "This gene encodes a cell surface tyrosine kinase receptor for members of the platelet-derived growth factor family. These growth factors are mitogens for cells of mesenchymal origin. The identity of the growth factor bound to a receptor monomer determines whether the functional receptor is a homodimer or a heterodimer, composed of both platelet-derived growth factor receptor alpha and beta polypeptides. Studies suggest that this gene plays a role in organ development, wound healing, and tumor progression. Mutations in this gene have been associated with idiopathic hypereosinophilic syndrome, somatic and familial gastrointestinal stromal tumors, and a variety of other cancers. [provided by RefSeq, Mar 2012].",
         "['Immunoglobulin subtype', 'Immunoglobulin-like domain', 'Immunoglobulin-like beta-sandwich domain', 'Immunoglobulin-like fold', 'Immunoglobulin-like domain superfamily', 'Platelet-derived growth factor receptor-like protein', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class III, conserved site', 'Immunoglobulin subtype 2', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Immunoglobulin I-set', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Platelet-derived growth factor receptor alpha', 'Receptor Tyrosine Kinase']"
        ],
        [
         "36",
         "PML nuclear body scaffold",
         "PML",
         "39",
         "The protein encoded by this gene is a member of the tripartite motif (TRIM) family. The TRIM motif includes three zinc-binding domains, a RING, a B-box type 1 and a B-box type 2, and a coiled-coil region. This phosphoprotein localizes to nuclear bodies where it functions as a transcription factor and tumor suppressor. Its expression is cell-cycle related and it regulates the p53 response to oncogenic signals. The gene is often involved in the translocation with the retinoic acid receptor alpha gene associated with acute promyelocytic leukemia (APL). Extensive alternative splicing of this gene results in several variations of the protein's central and C-terminal regions; all variants encode the same N-terminus. Alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Jul 2008].",
         "['PML-like, coiled-coil', 'TRIM45/56/19', 'B-box-type zinc finger', 'Zinc finger, RING-type', 'Zinc finger, RING/FYVE/PHD-type', 'Zinc finger, RING-type, conserved site', 'Zinc finger, C3HC4 RING-type']"
        ],
        [
         "37",
         "phosphatase and tensin homolog",
         "PTEN",
         "41",
         "This gene was identified as a tumor suppressor that is mutated in a large number of cancers at high frequency. The protein encoded by this gene is a phosphatidylinositol-3,4,5-trisphosphate 3-phosphatase. It contains a tensin like domain as well as a catalytic domain similar to that of the dual specificity protein tyrosine phosphatases. Unlike most of the protein tyrosine phosphatases, this protein preferentially dephosphorylates phosphoinositide substrates. It negatively regulates intracellular levels of phosphatidylinositol-3,4,5-trisphosphate in cells and functions as a tumor suppressor by negatively regulating AKT/PKB signaling pathway. The use of a non-canonical (CUG) upstream initiation site produces a longer isoform that initiates translation with a leucine, and is thought to be preferentially associated with the mitochondrial inner membrane. This longer isoform may help regulate energy metabolism in the mitochondria. A pseudogene of this gene is found on chromosome 9. Alternative splicing and the use of multiple translation start codons results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Feb 2015].",
         "['Tyrosine-specific protein phosphatases domain', 'Protein-tyrosine phosphatase, catalytic', 'Tensin phosphatase, C2 domain', 'Protein-tyrosine phosphatase, active site', 'Protein-tyrosine phosphatase-like', 'Tensin-type phosphatase domain', 'C2 domain superfamily', 'PTEN, phosphatase domain', 'Dual-specificity lipid and protein phosphatase', 'Bifunctional phosphatidylinositol trisphosphate phosphatase/dual specificity phosphatase PTEN']"
        ],
        [
         "38",
         "ret proto-oncogene",
         "RET",
         "42",
         "This gene encodes a transmembrane receptor and member of the tyrosine protein kinase family of proteins. Binding of ligands such as GDNF (glial cell-line derived neurotrophic factor) and other related proteins to the encoded receptor stimulates receptor dimerization and activation of downstream signaling pathways that play a role in cell differentiation, growth, migration and survival. The encoded receptor is important in development of the nervous system, and the development of organs and tissues derived from the neural crest. This proto-oncogene can undergo oncogenic activation through both cytogenetic rearrangement and activating point mutations. Mutations in this gene are associated with Hirschsprung disease and central hypoventilation syndrome and have been identified in patients with renal agenesis. [provided by RefSeq, Sep 2017].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Cadherin-like', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Cadherin-like superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Ret, cadherin like domain 1', 'RET, cadherin-like domain 4', 'Receptor Tyrosine Kinase', 'Tyrosine-protein kinase, Ret receptor', 'Tyrosine-protein kinase receptor Ret, cadherin like domain 3']"
        ],
        [
         "39",
         "RUNX family transcription factor 1",
         "RUNX1",
         "43",
         "Core binding factor (CBF) is a heterodimeric transcription factor that binds to the core element of many enhancers and promoters. The protein encoded by this gene represents the alpha subunit of CBF and is thought to be involved in the development of normal hematopoiesis. Chromosomal translocations involving this gene are well-documented and have been associated with several types of leukemia. Three transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].",
         "['Acute myeloid leukemia 1 protein (AML1)/Runt', 'p53-like transcription factor, DNA-binding domain superfamily', 'p53/RUNT-type transcription factor, DNA-binding domain superfamily', 'Runt domain', 'Runx, C-terminal domain', 'Runt-related transcription factor RUNX', 'Runx, central domain superfamily']"
        ],
        [
         "40",
         "splicing factor 3b subunit 1",
         "SF3B1",
         "44",
         "This gene encodes subunit 1 of the splicing factor 3b protein complex. Splicing factor 3b, together with splicing factor 3a and a 12S RNA unit, forms the U2 small nuclear ribonucleoproteins complex (U2 snRNP). The splicing factor 3b/3a complex binds pre-mRNA upstream of the intron's branch site in a sequence independent manner and may anchor the U2 snRNP to the pre-mRNA. Splicing factor 3b is also a component of the minor U12-type spliceosome. The carboxy-terminal two-thirds of subunit 1 have 22 non-identical, tandem HEAT repeats that form rod-like, helical structures. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Jul 2008].",
         "['Splicing factor 3B subunit 1-like', 'Armadillo-like helical', 'Armadillo-type fold', 'Splicing factor 3B subunit 1']"
        ],
        [
         "41",
         "tumor protein p53",
         "TP53",
         "45",
         "This gene encodes a tumor suppressor protein containing transcriptional activation, DNA binding, and oligomerization domains. The encoded protein responds to diverse cellular stresses to regulate expression of target genes, thereby inducing cell cycle arrest, apoptosis, senescence, DNA repair, or changes in metabolism. Mutations in this gene are associated with a variety of human cancers, including hereditary cancers such as Li-Fraumeni syndrome. Alternative splicing of this gene and the use of alternate promoters result in multiple transcript variants and isoforms. Additional isoforms have also been shown to result from the use of alternate translation initiation codons from identical transcript variants (PMIDs: 12032546, 20937277). [provided by RefSeq, Dec 2016].",
         "['p53 tumour suppressor family', 'p53-like transcription factor, DNA-binding domain superfamily', 'p53, tetramerisation domain', 'p53, DNA-binding domain', 'p53/RUNT-type transcription factor, DNA-binding domain superfamily', 'p53 transactivation domain', 'p53-like tetramerisation domain superfamily', 'Cellular tumor antigen p53, transactivation domain 2']"
        ],
        [
         "42",
         "TSC complex subunit 1",
         "TSC1",
         "46",
         "This gene is a tumor suppressor gene that encodes the growth inhibitory protein hamartin. The encoded protein interacts with and stabilizes the GTPase activating protein tuberin. This hamartin-tuberin complex negatively regulates mammalian target of rapamycin complex 1 (mTORC1) signaling which is a major regulator of anabolic cell growth. This protein also functions as a co-chaperone for Hsp90 that inhibits its ATPase activity. This protein functions as a facilitator of Hsp90-mediated folding of kinase and non-kinase clients, including TSC2 and thereby preventing their ubiquitination and proteasomal degradation. Mutations in this gene have been associated with tuberous sclerosis and lymphangioleiomyomatosis. [provided by RefSeq, May 2022].",
         "['Hamartin', 'Armadillo-type fold']"
        ],
        [
         "43",
         "TSC complex subunit 2",
         "TSC2",
         "47",
         "This gene is a tumor suppressor gene that encodes the growth inhibitory protein tuberin. Tuberin interacts with hamartin to form the TSC protein complex which functions in the control of cell growth. This TSC protein complex negatively regulates mammalian target of rapamycin complex 1 (mTORC1) signaling which is a major regulator of anabolic cell growth. Mutations in this gene have been associated with tuberous sclerosis and lymphangioleiomyomatosis. [provided by RefSeq, May 2022].",
         "['Rap/Ran-GAP domain', 'Tuberin', 'Armadillo-like helical', 'Armadillo-type fold', 'Tuberin-type domain', 'Tuberin, N-terminal', 'Tuberin/Ral GTPase-activating protein subunit alpha', 'Rap/Ran-GAP superfamily']"
        ],
        [
         "44",
         "U2 small nuclear RNA auxiliary factor 1",
         "U2AF1",
         "48",
         "This gene belongs to the splicing factor SR family of genes. U2 auxiliary factor, comprising a large and a small subunit, is a non-snRNP protein required for the binding of U2 snRNP to the pre-mRNA branch site. This gene encodes the small subunit which plays a critical role in both constitutive and enhancer-dependent RNA splicing by directly mediating interactions between the large subunit and proteins bound to the enhancers. Alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Jul 2008].",
         "['RNA recognition motif domain', 'Zinc finger, CCCH-type', 'RNA recognition motif domain, eukaryote', 'U2 auxiliary factor small subunit', 'Nucleotide-binding alpha-beta plait domain superfamily', 'RNA-binding domain superfamily']"
        ],
        [
         "45",
         "WT1 transcription factor",
         "WT1",
         "49",
         "This gene encodes a transcription factor that contains four zinc-finger motifs at the C-terminus and a proline/glutamine-rich DNA-binding domain at the N-terminus. It has an essential role in the normal development of the urogenital system, and it is mutated in a small subset of patients with Wilms tumor. This gene exhibits complex tissue-specific and polymorphic imprinting pattern, with biallelic, and monoallelic expression from the maternal and paternal alleles in different tissues. Multiple transcript variants have been described. In several variants, there is evidence for the use of a non-AUG (CUG) translation initiation codon upstream of, and in-frame with the first AUG. Authors of PMID:7926762 also provide evidence that WT1 mRNA undergoes RNA editing in human and rat, and that this process is tissue-restricted and developmentally regulated. [provided by RefSeq, Mar 2015].",
         "[\"Wilm's tumour protein, N-terminal\", 'Zinc finger C2H2-type', 'Zinc finger C2H2 superfamily']"
        ],
        [
         "46",
         "notch receptor 1",
         "NOTCH1",
         "50",
         "This gene encodes a member of the NOTCH family of proteins. Members of this Type I transmembrane protein family share structural characteristics including an extracellular domain consisting of multiple epidermal growth factor-like (EGF) repeats, and an intracellular domain consisting of multiple different domain types. Notch signaling is an evolutionarily conserved intercellular signaling pathway that regulates interactions between physically adjacent cells through binding of Notch family receptors to their cognate ligands. The encoded preproprotein is proteolytically processed in the trans-Golgi network to generate two polypeptide chains that heterodimerize to form the mature cell-surface receptor. This receptor plays a role in the development of numerous cell and tissue types. Mutations in this gene are associated with aortic valve disease, Adams-Oliver syndrome, T-cell acute lymphoblastic leukemia, chronic lymphocytic leukemia, and head and neck squamous cell carcinoma. [provided by RefSeq, Jan 2016].",
         "['EGF-type aspartate/asparagine hydroxylation site', 'EGF-like domain', 'Notch domain', 'EGF-like calcium-binding domain', 'Ankyrin repeat', 'Notch', 'Growth factor receptor cysteine-rich domain superfamily', 'Notch, NOD domain', 'Notch, NODP domain', 'EGF-like, conserved site', 'EGF-like calcium-binding, conserved site', 'Neurogenic locus notch homolog protein 1', 'Notch, C-terminal', 'Notch-like domain superfamily', 'Ankyrin repeat-containing domain superfamily', 'NOTCH1 EGF-like calcium-binding domain', 'Notch signaling pathway']"
        ],
        [
         "47",
         "discoidin domain receptor tyrosine kinase 2",
         "DDR2",
         "51",
         "This gene encodes a member of the discoidin domain receptor subclass of the receptor tyrosine kinase (RTKs) protein family. RTKs play a key role in the communication of cells with their microenvironment. The encoded protein is a collagen-induced receptor that activates signal transduction pathways involved in cell adhesion, proliferation, and extracellular matrix remodeling. This protein is expressed in numerous cell types and may alos be involved in wound repair and regulate tumor growth and invasiveness. Mutations in this gene are the cause of short limb-hand type spondylometaepiphyseal dysplasia. [provided by RefSeq, Aug 2017].",
         "['Coagulation factor 5/8 C-terminal domain', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class II, conserved site', 'Tyrosine-protein kinase, active site', 'Galactose-binding-like domain superfamily', 'Protein kinase-like domain superfamily', 'Tyrosine-protein kinase, catalytic domain', 'Discoidin domain-containing receptor 1/2, DS-like domain', 'Receptor Tyrosine Kinase']"
        ],
        [
         "48",
         "MET proto-oncogene, receptor tyrosine kinase",
         "MET",
         "52",
         "This gene encodes a member of the receptor tyrosine kinase family of proteins and the product of the proto-oncogene MET. The encoded preproprotein is proteolytically processed to generate alpha and beta subunits that are linked via disulfide bonds to form the mature receptor. Further processing of the beta subunit results in the formation of the M10 peptide, which has been shown to reduce lung fibrosis. Binding of its ligand, hepatocyte growth factor, induces dimerization and activation of the receptor, which plays a role in cellular survival, embryogenesis, and cellular migration and invasion. Mutations in this gene are associated with papillary renal cell carcinoma, hepatocellular carcinoma, and various head and neck cancers. Amplification and overexpression of this gene are also associated with multiple human cancers. [provided by RefSeq, May 2016].",
         "['Sema domain', 'WD40/YVTN repeat-like-containing domain superfamily', 'Plexin family', 'Sema domain superfamily', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Plexin repeat', 'IPT domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Immunoglobulin-like fold', 'Immunoglobulin E-set', 'PSI domain', 'Tyrosine-protein kinase, HGF/MSP receptor', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain']"
        ],
        [
         "49",
         "EWS RNA binding protein 1",
         "EWSR1",
         "54",
         "This gene encodes a multifunctional protein that is involved in various cellular processes, including gene expression, cell signaling, and RNA processing and transport. The protein includes an N-terminal transcriptional activation domain and a C-terminal RNA-binding domain. Chromosomal translocations between this gene and various genes encoding transcription factors result in the production of chimeric proteins that are involved in tumorigenesis. These chimeric proteins usually consist of the N-terminal transcriptional activation domain of this protein fused to the C-terminal DNA-binding domain of the transcription factor protein. Mutations in this gene, specifically a t(11;22)(q24;q12) translocation, are known to cause Ewing sarcoma as well as neuroectodermal and various other tumors. Alternative splicing of this gene results in multiple transcript variants. Related pseudogenes have been identified on chromosomes 1 and 14. [provided by RefSeq, Jul 2009].",
         "['RNA recognition motif domain', 'Zinc finger, RanBP2-type', 'Nucleotide-binding alpha-beta plait domain superfamily', 'EWS, RNA recognition motif', 'TAF15/EWS/TLS family', 'RNA-binding domain superfamily', 'Zinc finger, RanBP2-type superfamily']"
        ]
       ],
       "shape": {
        "columns": 5,
        "rows": 590
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fullName</th>\n",
       "      <th>name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>summary</th>\n",
       "      <th>Protein_Domains</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALK receptor tyrosine kinase</td>\n",
       "      <td>ALK</td>\n",
       "      <td>1</td>\n",
       "      <td>This gene encodes a receptor tyrosine kinase, ...</td>\n",
       "      <td>[Protein kinase domain, MAM domain, Serine-thr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT serine/threonine kinase 1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>2</td>\n",
       "      <td>This gene encodes one of the three members of ...</td>\n",
       "      <td>[Protein kinase domain, AGC-kinase, C-terminal...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A-Raf proto-oncogene, serine/threonine kinase</td>\n",
       "      <td>ARAF</td>\n",
       "      <td>3</td>\n",
       "      <td>This proto-oncogene belongs to the RAF subfami...</td>\n",
       "      <td>[Protein kinase C-like, phorbol ester/diacylgl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL proto-oncogene 1, non-receptor tyrosine ki...</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>4</td>\n",
       "      <td>This gene is a protooncogene that encodes a pr...</td>\n",
       "      <td>[Protein kinase domain, SH2 domain, Serine-thr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>B-Raf proto-oncogene, serine/threonine kinase</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>5</td>\n",
       "      <td>This gene encodes a protein belonging to the R...</td>\n",
       "      <td>[Protein kinase domain, Serine-threonine/tyros...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>killin, p53 regulated DNA replication inhibitor</td>\n",
       "      <td>KLLN</td>\n",
       "      <td>32083</td>\n",
       "      <td>The protein encoded by this intronless gene is...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>myocyte enhancer factor 2B</td>\n",
       "      <td>MEF2B</td>\n",
       "      <td>33873</td>\n",
       "      <td>The product of this gene is a member of the MA...</td>\n",
       "      <td>[Transcription factor, MADS-box, MADS MEF2-lik...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>prostate cancer associated non-coding RNA 1</td>\n",
       "      <td>PRNCR1</td>\n",
       "      <td>41522</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>cytochrome c oxidase subunit II</td>\n",
       "      <td>COX2</td>\n",
       "      <td>56400</td>\n",
       "      <td>Contributes to cytochrome-c oxidase activity. ...</td>\n",
       "      <td>[Copper centre Cu(A), Cytochrome c oxidase sub...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>NADH dehydrogenase, subunit 1 (complex I)</td>\n",
       "      <td>ND1</td>\n",
       "      <td>56415</td>\n",
       "      <td>Enables NADH dehydrogenase (ubiquinone) activi...</td>\n",
       "      <td>[NADH:ubiquinone oxidoreductase, subunit 1/F42...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>590 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              fullName    name  gene_id  \\\n",
       "0                         ALK receptor tyrosine kinase     ALK        1   \n",
       "1                        AKT serine/threonine kinase 1    AKT1        2   \n",
       "2        A-Raf proto-oncogene, serine/threonine kinase    ARAF        3   \n",
       "3    ABL proto-oncogene 1, non-receptor tyrosine ki...    ABL1        4   \n",
       "4        B-Raf proto-oncogene, serine/threonine kinase    BRAF        5   \n",
       "..                                                 ...     ...      ...   \n",
       "585    killin, p53 regulated DNA replication inhibitor    KLLN    32083   \n",
       "586                         myocyte enhancer factor 2B   MEF2B    33873   \n",
       "587        prostate cancer associated non-coding RNA 1  PRNCR1    41522   \n",
       "588                    cytochrome c oxidase subunit II    COX2    56400   \n",
       "589          NADH dehydrogenase, subunit 1 (complex I)     ND1    56415   \n",
       "\n",
       "                                               summary  \\\n",
       "0    This gene encodes a receptor tyrosine kinase, ...   \n",
       "1    This gene encodes one of the three members of ...   \n",
       "2    This proto-oncogene belongs to the RAF subfami...   \n",
       "3    This gene is a protooncogene that encodes a pr...   \n",
       "4    This gene encodes a protein belonging to the R...   \n",
       "..                                                 ...   \n",
       "585  The protein encoded by this intronless gene is...   \n",
       "586  The product of this gene is a member of the MA...   \n",
       "587                                               None   \n",
       "588  Contributes to cytochrome-c oxidase activity. ...   \n",
       "589  Enables NADH dehydrogenase (ubiquinone) activi...   \n",
       "\n",
       "                                       Protein_Domains  \n",
       "0    [Protein kinase domain, MAM domain, Serine-thr...  \n",
       "1    [Protein kinase domain, AGC-kinase, C-terminal...  \n",
       "2    [Protein kinase C-like, phorbol ester/diacylgl...  \n",
       "3    [Protein kinase domain, SH2 domain, Serine-thr...  \n",
       "4    [Protein kinase domain, Serine-threonine/tyros...  \n",
       "..                                                 ...  \n",
       "585                                                 []  \n",
       "586  [Transcription factor, MADS-box, MADS MEF2-lik...  \n",
       "587                                                 []  \n",
       "588  [Copper centre Cu(A), Cytochrome c oxidase sub...  \n",
       "589  [NADH:ubiquinone oxidoreductase, subunit 1/F42...  \n",
       "\n",
       "[590 rows x 5 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "gene_info_df = civicdb_gene_info.process_gene_infos(gene_ids['id'])\n",
    "gene_info_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "fullName",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "name",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "gene_id",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "summary",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "Protein_Domains",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "90cacdb5-b8b3-4619-a72d-da3fa40f6843",
       "rows": [
        [
         "0",
         "ALK receptor tyrosine kinase",
         "ALK",
         "1",
         "This gene encodes a receptor tyrosine kinase, which belongs to the insulin receptor superfamily. This protein comprises an extracellular domain, an hydrophobic stretch corresponding to a single pass transmembrane region, and an intracellular kinase domain. It plays an important role in the development of the brain and exerts its effects on specific neurons in the nervous system. This gene has been found to be rearranged, mutated, or amplified in a series of tumours including anaplastic large cell lymphomas, neuroblastoma, and non-small cell lung cancer. The chromosomal rearrangements are the most common genetic alterations in this gene, which result in creation of multiple fusion genes in tumourigenesis, including ALK (chromosome 2)/EML4 (chromosome 2), ALK/RANBP2 (chromosome 2), ALK/ATIC (chromosome 2), ALK/TFG (chromosome 3), ALK/NPM1 (chromosome 5), ALK/SQSTM1 (chromosome 5), ALK/KIF5B (chromosome 10), ALK/CLTC (chromosome 17), ALK/TPM4 (chromosome 19), and ALK/MSN (chromosome X).[provided by RefSeq, Jan 2011].",
         "['Protein kinase domain', 'MAM domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class II, conserved site', 'Low-density lipoprotein (LDL) receptor class A repeat', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Concanavalin A-like lectin/glucanase domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'LDL receptor-like superfamily', 'Receptor Tyrosine Kinase']"
        ],
        [
         "1",
         "AKT serine/threonine kinase 1",
         "AKT1",
         "2",
         "This gene encodes one of the three members of the human AKT serine-threonine protein kinase family which are often referred to as protein kinase B alpha, beta, and gamma. These highly similar AKT proteins all have an N-terminal pleckstrin homology domain, a serine/threonine-specific kinase domain and a C-terminal regulatory domain. These proteins are phosphorylated by phosphoinositide 3-kinase (PI3K). AKT/PI3K forms a key component of many signalling pathways that involve the binding of membrane-bound ligands such as receptor tyrosine kinases, G-protein coupled receptors, and integrin-linked kinase. These AKT proteins therefore regulate a wide variety of cellular functions including cell proliferation, survival, metabolism, and angiogenesis in both normal and malignant cells. AKT proteins are recruited to the cell membrane by phosphatidylinositol 3,4,5-trisphosphate (PIP3) after phosphorylation of phosphatidylinositol 4,5-bisphosphate (PIP2) by PI3K. Subsequent phosphorylation of both threonine residue 308 and serine residue 473 is required for full activation of the AKT1 protein encoded by this gene. Phosphorylation of additional residues also occurs, for example, in response to insulin growth factor-1 and epidermal growth factor. Protein phosphatases act as negative regulators of AKT proteins by dephosphorylating AKT or PIP3. The PI3K/AKT signalling pathway is crucial for tumor cell survival. Survival factors can suppress apoptosis in a transcription-independent manner by activating AKT1 which then phosphorylates and inactivates components of the apoptotic machinery. AKT proteins also participate in the mammalian target of rapamycin (mTOR) signalling pathway which controls the assembly of the eukaryotic translation initiation factor 4F (eIF4E) complex and this pathway, in addition to responding to extracellular signals from growth factors and cytokines, is disregulated in many cancers. Mutations in this gene are associated with multiple types of cancer and excessive tissue growth including Proteus syndrome and Cowden syndrome 6, and breast, colorectal, and ovarian cancers. Multiple alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Jul 2020].",
         "['Protein kinase domain', 'AGC-kinase, C-terminal', 'Pleckstrin homology domain', 'Serine/threonine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'PH-like domain superfamily', 'Protein kinase, ATP binding site', 'Protein kinase, C-terminal', 'Protein kinase B alpha, catalytic domain', 'Protein Kinase B, pleckstrin homology domain']"
        ],
        [
         "2",
         "A-Raf proto-oncogene, serine/threonine kinase",
         "ARAF",
         "3",
         "This proto-oncogene belongs to the RAF subfamily of the Ser/Thr protein kinase family, and maybe involved in cell growth and development. Alternatively spliced transcript variants encoding different isoforms have been found for this gene.[provided by RefSeq, Jan 2012]",
         "['Protein kinase C-like, phorbol ester/diacylglycerol-binding domain', 'Raf-like Ras-binding', 'Diacylglycerol/phorbol-ester binding', 'Ubiquitin-like domain superfamily', 'C1-like domain superfamily', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Serine/threonine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Serine/threonine-protein kinase']"
        ],
        [
         "3",
         "ABL proto-oncogene 1, non-receptor tyrosine kinase",
         "ABL1",
         "4",
         "This gene is a protooncogene that encodes a protein tyrosine kinase involved in a variety of cellular processes, including cell division, adhesion, differentiation, and response to stress. The activity of the protein is negatively regulated by its SH3 domain, whereby deletion of the region encoding this domain results in an oncogene. The ubiquitously expressed protein has DNA-binding activity that is regulated by CDC2-mediated phosphorylation, suggesting a cell cycle function. This gene has been found fused to a variety of translocation partner genes in various leukemias, most notably the t(9;22) translocation that results in a fusion with the 5' end of the breakpoint cluster region gene (BCR; MIM:151410). Alternative splicing of this gene results in two transcript variants, which contain alternative first exons that are spliced to the remaining common exons. [provided by RefSeq, Aug 2014].",
         "['Protein kinase domain', 'SH2 domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'SH3 domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'F-actin binding', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase ABL, SH2 domain', 'SH3-like domain superfamily', 'SH2 domain superfamily', 'Non-receptor tyrosine kinases involved in cell signaling']"
        ],
        [
         "4",
         "B-Raf proto-oncogene, serine/threonine kinase",
         "BRAF",
         "5",
         "This gene encodes a protein belonging to the RAF family of serine/threonine protein kinases. This protein plays a role in regulating the MAP kinase/ERK signaling pathway, which affects cell division, differentiation, and secretion. Mutations in this gene, most commonly the V600E mutation, are the most frequently identified cancer-causing mutations in melanoma, and have been identified in various other cancers as well, including non-Hodgkin lymphoma, colorectal cancer, thyroid carcinoma, non-small cell lung carcinoma, hairy cell leukemia and adenocarcinoma of lung. Mutations in this gene are also associated with cardiofaciocutaneous, Noonan, and Costello syndromes, which exhibit overlapping phenotypes. A pseudogene of this gene has been identified on the X chromosome. [provided by RefSeq, Aug 2017].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Protein kinase C-like, phorbol ester/diacylglycerol-binding domain', 'Raf-like Ras-binding', 'Serine/threonine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Diacylglycerol/phorbol-ester binding', 'Ubiquitin-like domain superfamily', 'C1-like domain superfamily', 'Serine/threonine-protein kinase']"
        ],
        [
         "5",
         "BRCA1 DNA repair associated",
         "BRCA1",
         "6",
         "This gene encodes a 190 kD nuclear phosphoprotein that plays a role in maintaining genomic stability, and it also acts as a tumor suppressor. The BRCA1 gene contains 22 exons spanning about 110 kb of DNA. The encoded protein combines with other tumor suppressors, DNA damage sensors, and signal transducers to form a large multi-subunit protein complex known as the BRCA1-associated genome surveillance complex (BASC). This gene product associates with RNA polymerase II, and through the C-terminal domain, also interacts with histone deacetylase complexes. This protein thus plays a role in transcription, DNA repair of double-stranded breaks, and recombination. Mutations in this gene are responsible for approximately 40% of inherited breast cancers and more than 80% of inherited breast and ovarian cancers. Alternative splicing plays a role in modulating the subcellular localization and physiological function of this gene. Many alternatively spliced transcript variants, some of which are disease-associated mutations, have been described for this gene, but the full-length natures of only some of these variants has been described. A related pseudogene, which is also located on chromosome 17, has been identified. [provided by RefSeq, May 2020].",
         "['BRCT domain', 'Zinc finger, RING-type', 'Breast cancer type 1 susceptibility protein (BRCA1)', 'Zinc finger, RING/FYVE/PHD-type', 'Zinc finger, RING-type, conserved site', 'Zinc finger, C3HC4 RING-type', 'Zinc finger, RING-type, eukaryotic', 'BRCA1-associated', 'BRCT domain superfamily', 'BRCA1, serine-rich domain']"
        ],
        [
         "6",
         "BRCA2 DNA repair associated",
         "BRCA2",
         "7",
         "Inherited mutations in BRCA1 and this gene, BRCA2, confer increased lifetime risk of developing breast or ovarian cancer. Both BRCA1 and BRCA2 are involved in maintenance of genome stability, specifically the homologous recombination pathway for double-strand DNA repair. The largest exon in both genes is exon 11, which harbors the most important and frequent mutations in breast cancer patients. The BRCA2 gene was found on chromosome 13q12.3 in human. The BRCA2 protein contains several copies of a 70 aa motif called the BRC motif, and these motifs mediate binding to the RAD51 recombinase which functions in DNA repair. BRCA2 is considered a tumor suppressor gene, as tumors with BRCA2 mutations generally exhibit loss of heterozygosity (LOH) of the wild-type allele. [provided by RefSeq, May 2020].",
         "['BRCA2 repeat', 'Nucleic acid-binding, OB-fold', 'BRCA2, OB1', 'BRCA2, OB3', 'Tower domain', 'Breast cancer type 2 susceptibility protein, helical domain', 'Breast cancer type 2 susceptibility protein', 'BRCA2 helical domain superfamily', 'BRCA2, OB2 domain']"
        ],
        [
         "7",
         "cyclin D1",
         "CCND1",
         "8",
         "The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance throughout the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with and functions as a regulatory subunit of CDK4 or CDK6, whose activity is required for cell cycle G1/S transition. This protein has been shown to interact with tumor suppressor protein Rb and the expression of this gene is regulated positively by Rb. Mutations, amplification and overexpression of this gene, which alters cell cycle progression, are observed frequently in a variety of human cancers. [provided by RefSeq, Dec 2019].",
         "['Cyclin, C-terminal domain', 'Cyclin, N-terminal', 'Cyclin-like domain', 'Cyclin-like superfamily', 'Cyclin', 'Cyclins, cyclin-box']"
        ],
        [
         "8",
         "cyclin D2",
         "CCND2",
         "9",
         "The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance through the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with CDK4 or CDK6 and functions as a regulatory subunit of the complex, whose activity is required for cell cycle G1/S transition. This protein has been shown to interact with and be involved in the phosphorylation of tumor suppressor protein Rb. Knockout studies of the homologous gene in mouse suggest the essential roles of this gene in ovarian granulosa and germ cell proliferation. High level expression of this gene was observed in ovarian and testicular tumors. Mutations in this gene are associated with megalencephaly-polymicrogyria-polydactyly-hydrocephalus syndrome 3 (MPPH3). [provided by RefSeq, Sep 2014].",
         "['Cyclin, C-terminal domain', 'Cyclin, N-terminal', 'Cyclin-like domain', 'Cyclin-like superfamily', 'Cyclin', 'Cyclins, cyclin-box']"
        ],
        [
         "9",
         "cyclin D3",
         "CCND3",
         "10",
         "The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance through the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with and functions as a regulatory subunit of CDK4 or CDK6, whose activtiy is required for cell cycle G1/S transition. This protein has been shown to interact with and be involved in the phosphorylation of tumor suppressor protein Rb. The CDK4 activity associated with this cyclin was reported to be necessary for cell cycle progression through G2 phase into mitosis after UV radiation. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2008].",
         "['Cyclin, N-terminal', 'Cyclin-like superfamily', 'Cyclin', 'Cyclin, C-terminal domain', 'Cyclin-like domain', 'Cyclins, cyclin-box']"
        ],
        [
         "10",
         "cyclin E1",
         "CCNE1",
         "11",
         "The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance through the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with and functions as a regulatory subunit of CDK2, whose activity is required for cell cycle G1/S transition. This protein accumulates at the G1-S phase boundary and is degraded as cells progress through S phase. Overexpression of this gene has been observed in many tumors, which results in chromosome instability, and thus may contribute to tumorigenesis. This protein was found to associate with, and be involved in, the phosphorylation of NPAT protein (nuclear protein mapped to the ATM locus), which participates in cell-cycle regulated histone gene expression and plays a critical role in promoting cell-cycle progression in the absence of pRB. [provided by RefSeq, Apr 2016].",
         "['Cyclin, C-terminal domain', 'Cyclin, N-terminal', 'Cyclin-like domain', 'Cyclin-like superfamily', 'Cyclin', 'Cyclins, cyclin-box']"
        ],
        [
         "11",
         "cyclin dependent kinase 6",
         "CDK6",
         "12",
         "The protein encoded by this gene is a member of the CMGC family of serine/threonine protein kinases. This kinase is a catalytic subunit of the protein kinase complex that is important for cell cycle G1 phase progression and G1/S transition. The activity of this kinase first appears in mid-G1 phase, which is controlled by the regulatory subunits including D-type cyclins and members of INK4 family of CDK inhibitors. This kinase, as well as CDK4, has been shown to phosphorylate, and thus regulate the activity of, tumor suppressor protein Rb. Altered expression of this gene has been observed in multiple human cancers. A mutation in this gene resulting in reduced cell proliferation, and impaired cell motility and polarity, and has been identified in patients with primary microcephaly. [provided by RefSeq, Aug 2017].",
         "['Protein kinase domain', 'Serine/threonine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Cyclin-dependent kinase 6', 'Cyclin-dependent kinase']"
        ],
        [
         "12",
         "cyclin dependent kinase 4",
         "CDK4",
         "13",
         "The protein encoded by this gene is a member of the Ser/Thr protein kinase family. This protein is highly similar to the gene products of S. cerevisiae cdc28 and S. pombe cdc2. It is a catalytic subunit of the protein kinase complex that is important for cell cycle G1 phase progression. The activity of this kinase is restricted to the G1-S phase, which is controlled by the regulatory subunits D-type cyclins and CDK inhibitor p16(INK4a). This kinase was shown to be responsible for the phosphorylation of retinoblastoma gene product (Rb). Mutations in this gene as well as in its related proteins including D-type cyclins, p16(INK4a) and Rb were all found to be associated with tumorigenesis of a variety of cancers. Multiple polyadenylation sites of this gene have been reported. [provided by RefSeq, Jul 2008].",
         "['Protein kinase domain', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Cyclin-dependent kinase', 'Serine/threonine-protein kinase, active site']"
        ],
        [
         "13",
         "cyclin dependent kinase inhibitor 2A",
         "CDKN2A",
         "14",
         "This gene generates several transcript variants which differ in their first exons. At least three alternatively spliced variants encoding distinct proteins have been reported, two of which encode structurally related isoforms known to function as inhibitors of CDK4 kinase. The remaining transcript includes an alternate first exon located 20 Kb upstream of the remainder of the gene; this transcript contains an alternate open reading frame (ARF) that specifies a protein which is structurally unrelated to the products of the other variants. This ARF product functions as a stabilizer of the tumor suppressor protein p53 as it can interact with, and sequester, the E3 ubiquitin-protein ligase MDM2, a protein responsible for the degradation of p53. In spite of the structural and functional differences, the CDK inhibitor isoforms and the ARF product encoded by this gene, through the regulatory roles of CDK4 and p53 in cell cycle G1 progression, share a common functionality in cell cycle G1 control. This gene is frequently mutated or deleted in a wide variety of tumors, and is known to be an important tumor suppressor gene. [provided by RefSeq, Sep 2012].",
         "['Ankyrin repeat-containing domain superfamily', 'Ankyrin Repeat and Cyclin-Dependent Kinase Inhibitor', 'Tumor suppressor ARF']"
        ],
        [
         "14",
         "CCAAT enhancer binding protein alpha",
         "CEBPA",
         "15",
         "This intronless gene encodes a transcription factor that contains a basic leucine zipper (bZIP) domain and recognizes the CCAAT motif in the promoters of target genes. The encoded protein functions in homodimers and also heterodimers with CCAAT/enhancer-binding proteins beta and gamma. Activity of this protein can modulate the expression of genes involved in cell cycle regulation as well as in body weight homeostasis. Mutation of this gene is associated with acute myeloid leukemia. The use of alternative in-frame non-AUG (GUG) and AUG start codons results in protein isoforms with different lengths. Differential translation initiation is mediated by an out-of-frame, upstream open reading frame which is located between the GUG and the first AUG start codons. [provided by RefSeq, Dec 2013].",
         "['Basic-leucine zipper domain', 'CCAAT/enhancer-binding protein, chordates', 'CCAAT/enhancer-binding protein C/EBP', 'Basic-leucine zipper domain superfamily']"
        ],
        [
         "15",
         "DNA methyltransferase 3 alpha",
         "DNMT3A",
         "18",
         "CpG methylation is an epigenetic modification that is important for embryonic development, imprinting, and X-chromosome inactivation. Studies in mice have demonstrated that DNA methylation is required for mammalian development. This gene encodes a DNA methyltransferase that is thought to function in de novo methylation, rather than maintenance methylation. The protein localizes to the cytoplasm and nucleus and its expression is developmentally regulated. [provided by RefSeq, Mar 2016].",
         "['PWWP domain', 'C-5 cytosine methyltransferase', 'DNA methylase, C-5 cytosine-specific, active site', 'ADD domain', 'S-adenosyl-L-methionine-dependent methyltransferase superfamily', 'DNMT3, cysteine rich ADD domain, GATA1-like zinc finger', 'DNA (cytosine-5)-methyltransferase 3A, ADD domain', 'DNMT3, ADD domain, PHD zinc finger']"
        ],
        [
         "16",
         "epidermal growth factor receptor",
         "EGFR",
         "19",
         "The protein encoded by this gene is a transmembrane glycoprotein that is a member of the protein kinase superfamily. This protein is a receptor for members of the epidermal growth factor family. EGFR is a cell surface protein that binds to epidermal growth factor, thus inducing receptor dimerization and tyrosine autophosphorylation leading to cell proliferation. Mutations in this gene are associated with lung cancer. EGFR is a component of the cytokine storm which contributes to a severe form of Coronavirus Disease 2019 (COVID-19) resulting from infection with severe acute respiratory syndrome coronavirus-2 (SARS-CoV-2). [provided by RefSeq, Jul 2020].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Furin-like repeat', 'Tyrosine-protein kinase, active site', 'Growth factor receptor cysteine-rich domain superfamily', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Growth factor receptor domain 4', 'Epidermal growth factor receptor-like, transmembrane-juxtamembrane segment', 'Receptor Tyrosine Kinase', 'Receptor L-domain', 'Furin-like cysteine-rich domain', 'Tyrosine protein kinase, EGF/ERB/XmrK receptor', 'Receptor L-domain superfamily']"
        ],
        [
         "17",
         "erb-b2 receptor tyrosine kinase 2",
         "ERBB2",
         "20",
         "This gene encodes a member of the epidermal growth factor (EGF) receptor family of receptor tyrosine kinases. This protein has no ligand binding domain of its own and therefore cannot bind growth factors. However, it does bind tightly to other ligand-bound EGF receptor family members to form a heterodimer, stabilizing ligand binding and enhancing kinase-mediated activation of downstream signalling pathways, such as those involving mitogen-activated protein kinase and phosphatidylinositol-3 kinase. Allelic variations at amino acid positions 654 and 655 of isoform a (positions 624 and 625 of isoform b) have been reported, with the most common allele, Ile654/Ile655, shown here. Amplification and/or overexpression of this gene has been reported in numerous cancers, including breast and ovarian tumors. Alternative splicing results in several additional transcript variants, some encoding different isoforms and others that have not been fully characterized. [provided by RefSeq, Jul 2008].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Epidermal growth factor receptor-like, transmembrane-juxtamembrane segment', 'Receptor Tyrosine Kinase', 'Furin-like cysteine-rich domain', 'Furin-like repeat', 'Growth factor receptor cysteine-rich domain superfamily', 'Growth factor receptor domain 4', 'Receptor L-domain superfamily', 'Receptor L-domain', 'Tyrosine protein kinase, EGF/ERB/XmrK receptor']"
        ],
        [
         "18",
         "estrogen receptor 1",
         "ESR1",
         "21",
         "This gene encodes an estrogen receptor and ligand-activated transcription factor. The canonical protein contains an N-terminal ligand-independent transactivation domain, a central DNA binding domain, a hinge domain, and a C-terminal ligand-dependent transactivation domain. The protein localizes to the nucleus where it may form either a homodimer or a heterodimer with estrogen receptor 2. The protein encoded by this gene regulates the transcription of many estrogen-inducible genes that play a role in growth, metabolism, sexual development, gestation, and other reproductive functions and is expressed in many non-reproductive tissues. The receptor encoded by this gene plays a key role in breast cancer, endometrial cancer, and osteoporosis. This gene is reported to have dozens of transcript variants due to the use of alternate promoters and alternative splicing, however, the full-length nature of many of these variants remain uncertain. [provided by RefSeq, Jul 2020].",
         "['Estrogen receptor', 'Estrogen receptor, N-terminal', 'Nuclear hormone receptor, ligand-binding domain', 'Zinc finger, nuclear hormone receptor-type', 'Nuclear hormone receptor', 'Zinc finger, NHR/GATA-type', 'Estrogen receptor/oestrogen-related receptor', 'Oestrogen-type nuclear receptor final C-terminal domain', 'Nuclear hormone receptor-like domain superfamily', 'Nuclear hormone receptor family NR3 subfamily']"
        ],
        [
         "19",
         "fibroblast growth factor receptor 2",
         "FGFR2",
         "22",
         "The protein encoded by this gene is a member of the fibroblast growth factor receptor family, where amino acid sequence is highly conserved between members and throughout evolution. FGFR family members differ from one another in their ligand affinities and tissue distribution. A full-length representative protein consists of an extracellular region, composed of three immunoglobulin-like domains, a single hydrophobic membrane-spanning segment and a cytoplasmic tyrosine kinase domain. The extracellular portion of the protein interacts with fibroblast growth factors, setting in motion a cascade of downstream signals, ultimately influencing mitogenesis and differentiation. This particular family member is a high-affinity receptor for acidic, basic and/or keratinocyte growth factor, depending on the isoform. Mutations in this gene are associated with Crouzon syndrome, Pfeiffer syndrome, Craniosynostosis, Apert syndrome, Jackson-Weiss syndrome, Beare-Stevenson cutis gyrata syndrome, Saethre-Chotzen syndrome, and syndromic craniosynostosis. Multiple alternatively spliced transcript variants encoding different isoforms have been noted for this gene. [provided by RefSeq, Jan 2009].",
         "['Immunoglobulin subtype 2', 'Immunoglobulin subtype', 'Immunoglobulin-like domain', 'Immunoglobulin-like fold', 'Immunoglobulin-like domain superfamily', 'Immunoglobulin I-set', 'Fibroblast growth factor receptor-like', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Fibroblast growth factor receptor family', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Receptor Tyrosine Kinase']"
        ],
        [
         "20",
         "fibroblast growth factor receptor 3",
         "FGFR3",
         "23",
         "This gene encodes a member of the fibroblast growth factor receptor (FGFR) family, with its amino acid sequence being highly conserved between members and among divergent species. FGFR family members differ from one another in their ligand affinities and tissue distribution. A full-length representative protein would consist of an extracellular region, composed of three immunoglobulin-like domains, a single hydrophobic membrane-spanning segment and a cytoplasmic tyrosine kinase domain. The extracellular portion of the protein interacts with fibroblast growth factors, setting in motion a cascade of downstream signals, ultimately influencing mitogenesis and differentiation. This particular family member binds acidic and basic fibroblast growth hormone and plays a role in bone development and maintenance. Mutations in this gene lead to craniosynostosis and multiple types of skeletal dysplasia. [provided by RefSeq, Aug 2017].",
         "['Immunoglobulin subtype 2', 'Immunoglobulin-like domain', 'Immunoglobulin I-set', 'Immunoglobulin-like fold', 'Immunoglobulin-like domain superfamily', 'Fibroblast growth factor receptor-like', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Immunoglobulin subtype', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Fibroblast growth factor receptor family', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Receptor Tyrosine Kinase']"
        ],
        [
         "21",
         "fms related receptor tyrosine kinase 3",
         "FLT3",
         "24",
         "This gene encodes a class III receptor tyrosine kinase that regulates hematopoiesis. This receptor is activated by binding of the fms-related tyrosine kinase 3 ligand to the extracellular domain, which induces homodimer formation in the plasma membrane leading to autophosphorylation of the receptor. The activated receptor kinase subsequently phosphorylates and activates multiple cytoplasmic effector molecules in pathways involved in apoptosis, proliferation, and differentiation of hematopoietic cells in bone marrow. Mutations that result in the constitutive activation of this receptor result in acute myeloid leukemia and acute lymphoblastic leukemia. [provided by RefSeq, Jan 2015].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class III, conserved site', 'Immunoglobulin-like domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Immunoglobulin-like beta-sandwich domain', 'Immunoglobulin-like fold', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Immunoglobulin-like domain superfamily', 'Receptor Tyrosine Kinase']"
        ],
        [
         "22",
         "GATA binding protein 2",
         "GATA2",
         "25",
         "This gene encodes a member of the GATA family of zinc-finger transcription factors that are named for the consensus nucleotide sequence they bind in the promoter regions of target genes. The encoded protein plays an essential role in regulating transcription of genes involved in the development and proliferation of hematopoietic and endocrine cell lineages. Alternative splicing results in multiple transcript variants.[provided by RefSeq, Mar 2009].",
         "['Transcription factor GATA', 'Zinc finger, GATA-type', 'Zinc finger, NHR/GATA-type', 'Transcription factor, GATA-2/3']"
        ],
        [
         "23",
         "isocitrate dehydrogenase (NADP(+)) 1",
         "IDH1",
         "26",
         "Isocitrate dehydrogenases catalyze the oxidative decarboxylation of isocitrate to 2-oxoglutarate. These enzymes belong to two distinct subclasses, one of which utilizes NAD(+) as the electron acceptor and the other NADP(+). Five isocitrate dehydrogenases have been reported: three NAD(+)-dependent isocitrate dehydrogenases, which localize to the mitochondrial matrix, and two NADP(+)-dependent isocitrate dehydrogenases, one of which is mitochondrial and the other predominantly cytosolic. Each NADP(+)-dependent isozyme is a homodimer. The protein encoded by this gene is the NADP(+)-dependent isocitrate dehydrogenase found in the cytoplasm and peroxisomes. It contains the PTS-1 peroxisomal targeting signal sequence. The presence of this enzyme in peroxisomes suggests roles in the regeneration of NADPH for intraperoxisomal reductions, such as the conversion of 2, 4-dienoyl-CoAs to 3-enoyl-CoAs, as well as in peroxisomal reactions that consume 2-oxoglutarate, namely the alpha-hydroxylation of phytanic acid. The cytoplasmic enzyme serves a significant role in cytoplasmic NADPH production. Alternatively spliced transcript variants encoding the same protein have been found for this gene. [provided by RefSeq, Sep 2013].",
         "['Isocitrate dehydrogenase NADP-dependent', 'Isopropylmalate dehydrogenase-like domain', 'Isocitrate/isopropylmalate dehydrogenase, conserved site']"
        ],
        [
         "24",
         "isocitrate dehydrogenase (NADP(+)) 2",
         "IDH2",
         "27",
         "Isocitrate dehydrogenases catalyze the oxidative decarboxylation of isocitrate to 2-oxoglutarate. These enzymes belong to two distinct subclasses, one of which utilizes NAD(+) as the electron acceptor and the other NADP(+). Five isocitrate dehydrogenases have been reported: three NAD(+)-dependent isocitrate dehydrogenases, which localize to the mitochondrial matrix, and two NADP(+)-dependent isocitrate dehydrogenases, one of which is mitochondrial and the other predominantly cytosolic. Each NADP(+)-dependent isozyme is a homodimer. The protein encoded by this gene is the NADP(+)-dependent isocitrate dehydrogenase found in the mitochondria. It plays a role in intermediary metabolism and energy production. This protein may tightly associate or interact with the pyruvate dehydrogenase complex. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2014].",
         "['Isocitrate dehydrogenase NADP-dependent', 'Isocitrate/isopropylmalate dehydrogenase, conserved site', 'Isopropylmalate dehydrogenase-like domain']"
        ],
        [
         "25",
         "Janus kinase 2",
         "JAK2",
         "28",
         "This gene encodes a non-receptor tyrosine kinase that plays a central role in cytokine and growth factor signalling. The primary isoform of this protein has an N-terminal FERM domain that is required for erythropoietin receptor association, an SH2 domain that binds STAT transcription factors, a pseudokinase domain and a C-terminal tyrosine kinase domain. Cytokine binding induces autophosphorylation and activation of this kinase. This kinase then recruits and phosphorylates signal transducer and activator of transcription (STAT) proteins. Growth factors like TGF-beta 1 also induce phosphorylation and activation of this kinase and translocation of downstream STAT proteins to the nucleus where they influence gene transcription. Mutations in this gene are associated with numerous inflammatory diseases and malignancies. This gene is a downstream target of the pleiotropic cytokine IL6 that is produced by B cells, T cells, dendritic cells and macrophages to produce an immune response or inflammation. Disregulation of the IL6/JAK2/STAT3 signalling pathways produces increased cellular proliferation and myeloproliferative neoplasms of hematopoietic stem cells. A nonsynonymous mutation in the pseudokinase domain of this gene disrupts the domains inhibitory effect and results in constitutive tyrosine phosphorylation activity and hypersensitivity to cytokine signalling. This gene and the IL6/JAK2/STAT3 signalling pathway is a therapeutic target for the treatment of excessive inflammatory responses to viral infections. Alternative splicing results in multiple transcript variants encoding distinct isoforms. [provided by RefSeq, Jul 2020].",
         "['FERM domain', 'Protein kinase domain', 'SH2 domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'PH-like domain superfamily', 'Tyrosine-protein kinase, non-receptor Jak/Tyk2', 'Protein kinase, ATP binding site', 'FERM central domain', 'Band 4.1 domain', 'Tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, non-receptor Jak2', 'Janus kinase 2, pseudokinase domain', 'Janus kinase 2, catalytic domain', 'Tyrosine-protein kinase JAK2, SH2 domain', 'FERM superfamily, second domain', 'SH2 domain superfamily', 'JAK2, FERM domain C-lobe', 'JAK, FERM F2 lobe domain', 'FERM F1 lobe ubiquitin-like domain', 'JAK1-3/TYK2, pleckstrin homology-like domain', 'Janus Kinase (JAK)']"
        ],
        [
         "26",
         "KIT proto-oncogene, receptor tyrosine kinase",
         "KIT",
         "29",
         "This gene encodes a receptor tyrosine kinase. This gene was initially identified as a homolog of the feline sarcoma viral oncogene v-kit and is often referred to as proto-oncogene c-Kit. The canonical form of this glycosylated transmembrane protein has an N-terminal extracellular region with five immunoglobulin-like domains, a transmembrane region, and an intracellular tyrosine kinase domain at the C-terminus. Upon activation by its cytokine ligand, stem cell factor (SCF), this protein phosphorylates multiple intracellular proteins that play a role in in the proliferation, differentiation, migration and apoptosis of many cell types and thereby plays an important role in hematopoiesis, stem cell maintenance, gametogenesis, melanogenesis, and in mast cell development, migration and function. This protein can be a membrane-bound or soluble protein. Mutations in this gene are associated with gastrointestinal stromal tumors, mast cell disease, acute myelogenous leukemia, and piebaldism. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, May 2020].",
         "['Immunoglobulin-like beta-sandwich domain', 'Immunoglobulin-like fold', 'Immunoglobulin-like domain superfamily', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class III, conserved site', 'Immunoglobulin subtype 2', 'Immunoglobulin subtype', 'Immunoglobulin-like domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Mast/stem cell growth factor receptor', 'Receptor Tyrosine Kinase']"
        ],
        [
         "27",
         "KRAS proto-oncogene, GTPase",
         "KRAS",
         "30",
         "This gene, a Kirsten ras oncogene homolog from the mammalian ras gene family, encodes a protein that is a member of the small GTPase superfamily. A single amino acid substitution is responsible for an activating mutation. The transforming protein that results is implicated in various malignancies, including lung adenocarcinoma, mucinous adenoma, ductal carcinoma of the pancreas and colorectal carcinoma. Alternative splicing leads to variants encoding two isoforms that differ in the C-terminal region. [provided by RefSeq, Jul 2008].",
         "['Small GTPase', 'Small GTPase, Ras-type', 'P-loop containing nucleoside triphosphate hydrolase', 'Small GTP-binding protein domain']"
        ],
        [
         "28",
         "mitogen-activated protein kinase kinase 1",
         "MAP2K1",
         "31",
         "The protein encoded by this gene is a member of the dual specificity protein kinase family, which acts as a mitogen-activated protein (MAP) kinase kinase. MAP kinases, also known as extracellular signal-regulated kinases (ERKs), act as an integration point for multiple biochemical signals. This protein kinase lies upstream of MAP kinases and stimulates the enzymatic activity of MAP kinases upon wide variety of extra- and intracellular signals. As an essential component of MAP kinase signal transduction pathway, this kinase is involved in many cellular processes such as proliferation, differentiation, transcription regulation and development. [provided by RefSeq, Jul 2008].",
         "['Protein kinase domain', 'Protein kinase-like domain superfamily', 'MAP kinase kinase', 'Serine/threonine-protein kinase, active site', 'Protein kinase, ATP binding site']"
        ],
        [
         "29",
         "myocyte enhancer factor 2D",
         "MEF2D",
         "32",
         "This gene is a member of the myocyte-specific enhancer factor 2 (MEF2) family of transcription factors. Members of this family are involved in control of muscle and neuronal cell differentiation and development, and are regulated by class II histone deacetylases. Fusions of the encoded protein with Deleted in Azoospermia-Associated Protein 1 (DAZAP1) due to a translocation have been found in an acute lymphoblastic leukemia cell line, suggesting a role in leukemogenesis. The encoded protein may also be involved in Parkinson disease and myotonic dystrophy. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Oct 2012].",
         "['Transcription factor, MADS-box', 'Holliday junction regulator protein family C-terminal', 'MADS MEF2-like', 'Transcription factor, MADS-box superfamily', 'MADS-box/MEF2 Transcription Factor']"
        ],
        [
         "30",
         "colony stimulating factor 1 receptor",
         "CSF1R",
         "33",
         "The protein encoded by this gene is the receptor for colony stimulating factor 1, a cytokine which controls the production, differentiation, and function of macrophages. This receptor mediates most if not all of the biological effects of this cytokine. Ligand binding activates the receptor kinase through a process of oligomerization and transphosphorylation. The encoded protein is a tyrosine kinase transmembrane receptor and member of the CSF1/PDGF receptor family of tyrosine-protein kinases. Mutations in this gene have been associated with a predisposition to myeloid malignancy. The first intron of this gene contains a transcriptionally inactive ribosomal protein L7 processed pseudogene oriented in the opposite direction. Alternative splicing results in multiple transcript variants. Expression of a splice variant from an LTR promoter has been found in Hodgkin lymphoma (HL), HL cell lines and anaplastic large cell lymphoma. [provided by RefSeq, Mar 2017].",
         "['Immunoglobulin-like fold', 'Immunoglobulin subtype 2', 'Immunoglobulin subtype', 'Immunoglobulin-like domain', 'Immunoglobulin-like beta-sandwich domain', 'Immunoglobulin-like domain superfamily', 'Platelet-derived growth factor receptor-like protein', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class III, conserved site', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Macrophage colony-stimulating factor 1 receptor', 'Receptor Tyrosine Kinase']"
        ],
        [
         "31",
         "O-6-methylguanine-DNA methyltransferase",
         "MGMT",
         "34",
         "Alkylating agents are potent carcinogens that can result in cell death, mutation and cancer. The protein encoded by this gene is a DNA repair protein that is involved in cellular defense against mutagenesis and toxicity from alkylating agents. The protein catalyzes transfer of methyl groups from O(6)-alkylguanine and other methylated moieties of the DNA to its own molecule, which repairs the toxic lesions. Methylation of the genes promoter has been associated with several cancer types, including colorectal cancer, lung cancer, lymphoma and glioblastoma. [provided by RefSeq, Sep 2015].",
         "['Methylated-DNA-[protein]-cysteine S-methyltransferase, active site', 'Methylguanine DNA methyltransferase, ribonuclease-like domain', 'Methylated-DNA-[protein]-cysteine S-methyltransferase, DNA binding', 'Methylated DNA-protein cysteine methyltransferase, DNA binding domain', 'Winged helix-like DNA-binding domain superfamily', 'Methylated DNA-protein cysteine methyltransferase domain superfamily']"
        ],
        [
         "32",
         "nucleophosmin 1",
         "NPM1",
         "35",
         "The protein encoded by this gene is involved in several cellular processes, including centrosome duplication, protein chaperoning, and cell proliferation. The encoded phosphoprotein shuttles between the nucleolus, nucleus, and cytoplasm, chaperoning ribosomal proteins and core histones from the nucleus to the cytoplasm. This protein is also known to sequester the tumor suppressor ARF in the nucleolus, protecting it from degradation until it is needed. Mutations in this gene are associated with acute myeloid leukemia. Dozens of pseudogenes of this gene have been identified. [provided by RefSeq, Aug 2017].",
         "['Nucleoplasmin family', 'Nucleoplasmin core domain', 'Nucleophosmin, C-terminal', 'Nucleoplasmin core domain superfamily']"
        ],
        [
         "33",
         "NRAS proto-oncogene, GTPase",
         "NRAS",
         "36",
         "This is an N-ras oncogene encoding a membrane protein that shuttles between the Golgi apparatus and the plasma membrane. This shuttling is regulated through palmitoylation and depalmitoylation by the ZDHHC9-GOLGA7 complex. The encoded protein, which has intrinsic GTPase activity, is activated by a guanine nucleotide-exchange factor and inactivated by a GTPase activating protein. Mutations in this gene have been associated with somatic rectal cancer, follicular thyroid cancer, autoimmune lymphoproliferative syndrome, Noonan syndrome, and juvenile myelomonocytic leukemia. [provided by RefSeq, Jun 2011].",
         "['Small GTPase', 'Small GTP-binding protein domain', 'Small GTPase, Ras-type', 'P-loop containing nucleoside triphosphate hydrolase']"
        ],
        [
         "34",
         "phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit alpha",
         "PIK3CA",
         "37",
         "Phosphatidylinositol 3-kinase is composed of an 85 kDa regulatory subunit and a 110 kDa catalytic subunit. The protein encoded by this gene represents the catalytic subunit, which uses ATP to phosphorylate PtdIns, PtdIns4P and PtdIns(4,5)P2. This gene has been found to be oncogenic and has been implicated in cervical cancers. A pseudogene of this gene has been defined on chromosome 22. [provided by RefSeq, Apr 2016].",
         "['Phosphatidylinositol 3-kinase Ras-binding (PI3K RBD) domain', 'Phosphatidylinositol 3-/4-kinase, catalytic domain', 'Phosphoinositide 3-kinase, accessory (PIK) domain', 'C2 phosphatidylinositol 3-kinase-type domain', 'Phosphatidylinositol 3-kinase, adaptor-binding domain', 'Protein kinase-like domain superfamily', 'Phosphatidylinositol kinase', 'Armadillo-type fold', 'Phosphatidylinositol 3/4-kinase, conserved site', 'Ubiquitin-like domain superfamily', 'C2 domain superfamily', 'Phosphatidylinositol 3-/4-kinase, catalytic domain superfamily', 'PI3Kalpha, catalytic domain', 'Phosphoinositide 3-kinase, accessory (PIK) domain superfamily']"
        ],
        [
         "35",
         "platelet derived growth factor receptor alpha",
         "PDGFRA",
         "38",
         "This gene encodes a cell surface tyrosine kinase receptor for members of the platelet-derived growth factor family. These growth factors are mitogens for cells of mesenchymal origin. The identity of the growth factor bound to a receptor monomer determines whether the functional receptor is a homodimer or a heterodimer, composed of both platelet-derived growth factor receptor alpha and beta polypeptides. Studies suggest that this gene plays a role in organ development, wound healing, and tumor progression. Mutations in this gene have been associated with idiopathic hypereosinophilic syndrome, somatic and familial gastrointestinal stromal tumors, and a variety of other cancers. [provided by RefSeq, Mar 2012].",
         "['Immunoglobulin subtype', 'Immunoglobulin-like domain', 'Immunoglobulin-like beta-sandwich domain', 'Immunoglobulin-like fold', 'Immunoglobulin-like domain superfamily', 'Platelet-derived growth factor receptor-like protein', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class III, conserved site', 'Immunoglobulin subtype 2', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Immunoglobulin I-set', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Platelet-derived growth factor receptor alpha', 'Receptor Tyrosine Kinase']"
        ],
        [
         "36",
         "PML nuclear body scaffold",
         "PML",
         "39",
         "The protein encoded by this gene is a member of the tripartite motif (TRIM) family. The TRIM motif includes three zinc-binding domains, a RING, a B-box type 1 and a B-box type 2, and a coiled-coil region. This phosphoprotein localizes to nuclear bodies where it functions as a transcription factor and tumor suppressor. Its expression is cell-cycle related and it regulates the p53 response to oncogenic signals. The gene is often involved in the translocation with the retinoic acid receptor alpha gene associated with acute promyelocytic leukemia (APL). Extensive alternative splicing of this gene results in several variations of the protein's central and C-terminal regions; all variants encode the same N-terminus. Alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Jul 2008].",
         "['PML-like, coiled-coil', 'TRIM45/56/19', 'B-box-type zinc finger', 'Zinc finger, RING-type', 'Zinc finger, RING/FYVE/PHD-type', 'Zinc finger, RING-type, conserved site', 'Zinc finger, C3HC4 RING-type']"
        ],
        [
         "37",
         "phosphatase and tensin homolog",
         "PTEN",
         "41",
         "This gene was identified as a tumor suppressor that is mutated in a large number of cancers at high frequency. The protein encoded by this gene is a phosphatidylinositol-3,4,5-trisphosphate 3-phosphatase. It contains a tensin like domain as well as a catalytic domain similar to that of the dual specificity protein tyrosine phosphatases. Unlike most of the protein tyrosine phosphatases, this protein preferentially dephosphorylates phosphoinositide substrates. It negatively regulates intracellular levels of phosphatidylinositol-3,4,5-trisphosphate in cells and functions as a tumor suppressor by negatively regulating AKT/PKB signaling pathway. The use of a non-canonical (CUG) upstream initiation site produces a longer isoform that initiates translation with a leucine, and is thought to be preferentially associated with the mitochondrial inner membrane. This longer isoform may help regulate energy metabolism in the mitochondria. A pseudogene of this gene is found on chromosome 9. Alternative splicing and the use of multiple translation start codons results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Feb 2015].",
         "['Tyrosine-specific protein phosphatases domain', 'Protein-tyrosine phosphatase, catalytic', 'Tensin phosphatase, C2 domain', 'Protein-tyrosine phosphatase, active site', 'Protein-tyrosine phosphatase-like', 'Tensin-type phosphatase domain', 'C2 domain superfamily', 'PTEN, phosphatase domain', 'Dual-specificity lipid and protein phosphatase', 'Bifunctional phosphatidylinositol trisphosphate phosphatase/dual specificity phosphatase PTEN']"
        ],
        [
         "38",
         "ret proto-oncogene",
         "RET",
         "42",
         "This gene encodes a transmembrane receptor and member of the tyrosine protein kinase family of proteins. Binding of ligands such as GDNF (glial cell-line derived neurotrophic factor) and other related proteins to the encoded receptor stimulates receptor dimerization and activation of downstream signaling pathways that play a role in cell differentiation, growth, migration and survival. The encoded receptor is important in development of the nervous system, and the development of organs and tissues derived from the neural crest. This proto-oncogene can undergo oncogenic activation through both cytogenetic rearrangement and activating point mutations. Mutations in this gene are associated with Hirschsprung disease and central hypoventilation syndrome and have been identified in patients with renal agenesis. [provided by RefSeq, Sep 2017].",
         "['Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Cadherin-like', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Cadherin-like superfamily', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain', 'Ret, cadherin like domain 1', 'RET, cadherin-like domain 4', 'Receptor Tyrosine Kinase', 'Tyrosine-protein kinase, Ret receptor', 'Tyrosine-protein kinase receptor Ret, cadherin like domain 3']"
        ],
        [
         "39",
         "RUNX family transcription factor 1",
         "RUNX1",
         "43",
         "Core binding factor (CBF) is a heterodimeric transcription factor that binds to the core element of many enhancers and promoters. The protein encoded by this gene represents the alpha subunit of CBF and is thought to be involved in the development of normal hematopoiesis. Chromosomal translocations involving this gene are well-documented and have been associated with several types of leukemia. Three transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].",
         "['Acute myeloid leukemia 1 protein (AML1)/Runt', 'p53-like transcription factor, DNA-binding domain superfamily', 'p53/RUNT-type transcription factor, DNA-binding domain superfamily', 'Runt domain', 'Runx, C-terminal domain', 'Runt-related transcription factor RUNX', 'Runx, central domain superfamily']"
        ],
        [
         "40",
         "splicing factor 3b subunit 1",
         "SF3B1",
         "44",
         "This gene encodes subunit 1 of the splicing factor 3b protein complex. Splicing factor 3b, together with splicing factor 3a and a 12S RNA unit, forms the U2 small nuclear ribonucleoproteins complex (U2 snRNP). The splicing factor 3b/3a complex binds pre-mRNA upstream of the intron's branch site in a sequence independent manner and may anchor the U2 snRNP to the pre-mRNA. Splicing factor 3b is also a component of the minor U12-type spliceosome. The carboxy-terminal two-thirds of subunit 1 have 22 non-identical, tandem HEAT repeats that form rod-like, helical structures. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Jul 2008].",
         "['Splicing factor 3B subunit 1-like', 'Armadillo-like helical', 'Armadillo-type fold', 'Splicing factor 3B subunit 1']"
        ],
        [
         "41",
         "tumor protein p53",
         "TP53",
         "45",
         "This gene encodes a tumor suppressor protein containing transcriptional activation, DNA binding, and oligomerization domains. The encoded protein responds to diverse cellular stresses to regulate expression of target genes, thereby inducing cell cycle arrest, apoptosis, senescence, DNA repair, or changes in metabolism. Mutations in this gene are associated with a variety of human cancers, including hereditary cancers such as Li-Fraumeni syndrome. Alternative splicing of this gene and the use of alternate promoters result in multiple transcript variants and isoforms. Additional isoforms have also been shown to result from the use of alternate translation initiation codons from identical transcript variants (PMIDs: 12032546, 20937277). [provided by RefSeq, Dec 2016].",
         "['p53 tumour suppressor family', 'p53-like transcription factor, DNA-binding domain superfamily', 'p53, tetramerisation domain', 'p53, DNA-binding domain', 'p53/RUNT-type transcription factor, DNA-binding domain superfamily', 'p53 transactivation domain', 'p53-like tetramerisation domain superfamily', 'Cellular tumor antigen p53, transactivation domain 2']"
        ],
        [
         "42",
         "TSC complex subunit 1",
         "TSC1",
         "46",
         "This gene is a tumor suppressor gene that encodes the growth inhibitory protein hamartin. The encoded protein interacts with and stabilizes the GTPase activating protein tuberin. This hamartin-tuberin complex negatively regulates mammalian target of rapamycin complex 1 (mTORC1) signaling which is a major regulator of anabolic cell growth. This protein also functions as a co-chaperone for Hsp90 that inhibits its ATPase activity. This protein functions as a facilitator of Hsp90-mediated folding of kinase and non-kinase clients, including TSC2 and thereby preventing their ubiquitination and proteasomal degradation. Mutations in this gene have been associated with tuberous sclerosis and lymphangioleiomyomatosis. [provided by RefSeq, May 2022].",
         "['Hamartin', 'Armadillo-type fold']"
        ],
        [
         "43",
         "TSC complex subunit 2",
         "TSC2",
         "47",
         "This gene is a tumor suppressor gene that encodes the growth inhibitory protein tuberin. Tuberin interacts with hamartin to form the TSC protein complex which functions in the control of cell growth. This TSC protein complex negatively regulates mammalian target of rapamycin complex 1 (mTORC1) signaling which is a major regulator of anabolic cell growth. Mutations in this gene have been associated with tuberous sclerosis and lymphangioleiomyomatosis. [provided by RefSeq, May 2022].",
         "['Rap/Ran-GAP domain', 'Tuberin', 'Armadillo-like helical', 'Armadillo-type fold', 'Tuberin-type domain', 'Tuberin, N-terminal', 'Tuberin/Ral GTPase-activating protein subunit alpha', 'Rap/Ran-GAP superfamily']"
        ],
        [
         "44",
         "U2 small nuclear RNA auxiliary factor 1",
         "U2AF1",
         "48",
         "This gene belongs to the splicing factor SR family of genes. U2 auxiliary factor, comprising a large and a small subunit, is a non-snRNP protein required for the binding of U2 snRNP to the pre-mRNA branch site. This gene encodes the small subunit which plays a critical role in both constitutive and enhancer-dependent RNA splicing by directly mediating interactions between the large subunit and proteins bound to the enhancers. Alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Jul 2008].",
         "['RNA recognition motif domain', 'Zinc finger, CCCH-type', 'RNA recognition motif domain, eukaryote', 'U2 auxiliary factor small subunit', 'Nucleotide-binding alpha-beta plait domain superfamily', 'RNA-binding domain superfamily']"
        ],
        [
         "45",
         "WT1 transcription factor",
         "WT1",
         "49",
         "This gene encodes a transcription factor that contains four zinc-finger motifs at the C-terminus and a proline/glutamine-rich DNA-binding domain at the N-terminus. It has an essential role in the normal development of the urogenital system, and it is mutated in a small subset of patients with Wilms tumor. This gene exhibits complex tissue-specific and polymorphic imprinting pattern, with biallelic, and monoallelic expression from the maternal and paternal alleles in different tissues. Multiple transcript variants have been described. In several variants, there is evidence for the use of a non-AUG (CUG) translation initiation codon upstream of, and in-frame with the first AUG. Authors of PMID:7926762 also provide evidence that WT1 mRNA undergoes RNA editing in human and rat, and that this process is tissue-restricted and developmentally regulated. [provided by RefSeq, Mar 2015].",
         "[\"Wilm's tumour protein, N-terminal\", 'Zinc finger C2H2-type', 'Zinc finger C2H2 superfamily']"
        ],
        [
         "46",
         "notch receptor 1",
         "NOTCH1",
         "50",
         "This gene encodes a member of the NOTCH family of proteins. Members of this Type I transmembrane protein family share structural characteristics including an extracellular domain consisting of multiple epidermal growth factor-like (EGF) repeats, and an intracellular domain consisting of multiple different domain types. Notch signaling is an evolutionarily conserved intercellular signaling pathway that regulates interactions between physically adjacent cells through binding of Notch family receptors to their cognate ligands. The encoded preproprotein is proteolytically processed in the trans-Golgi network to generate two polypeptide chains that heterodimerize to form the mature cell-surface receptor. This receptor plays a role in the development of numerous cell and tissue types. Mutations in this gene are associated with aortic valve disease, Adams-Oliver syndrome, T-cell acute lymphoblastic leukemia, chronic lymphocytic leukemia, and head and neck squamous cell carcinoma. [provided by RefSeq, Jan 2016].",
         "['EGF-type aspartate/asparagine hydroxylation site', 'EGF-like domain', 'Notch domain', 'EGF-like calcium-binding domain', 'Ankyrin repeat', 'Notch', 'Growth factor receptor cysteine-rich domain superfamily', 'Notch, NOD domain', 'Notch, NODP domain', 'EGF-like, conserved site', 'EGF-like calcium-binding, conserved site', 'Neurogenic locus notch homolog protein 1', 'Notch, C-terminal', 'Notch-like domain superfamily', 'Ankyrin repeat-containing domain superfamily', 'NOTCH1 EGF-like calcium-binding domain', 'Notch signaling pathway']"
        ],
        [
         "47",
         "discoidin domain receptor tyrosine kinase 2",
         "DDR2",
         "51",
         "This gene encodes a member of the discoidin domain receptor subclass of the receptor tyrosine kinase (RTKs) protein family. RTKs play a key role in the communication of cells with their microenvironment. The encoded protein is a collagen-induced receptor that activates signal transduction pathways involved in cell adhesion, proliferation, and extracellular matrix remodeling. This protein is expressed in numerous cell types and may alos be involved in wound repair and regulate tumor growth and invasiveness. Mutations in this gene are the cause of short limb-hand type spondylometaepiphyseal dysplasia. [provided by RefSeq, Aug 2017].",
         "['Coagulation factor 5/8 C-terminal domain', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Tyrosine-protein kinase, receptor class II, conserved site', 'Tyrosine-protein kinase, active site', 'Galactose-binding-like domain superfamily', 'Protein kinase-like domain superfamily', 'Tyrosine-protein kinase, catalytic domain', 'Discoidin domain-containing receptor 1/2, DS-like domain', 'Receptor Tyrosine Kinase']"
        ],
        [
         "48",
         "MET proto-oncogene, receptor tyrosine kinase",
         "MET",
         "52",
         "This gene encodes a member of the receptor tyrosine kinase family of proteins and the product of the proto-oncogene MET. The encoded preproprotein is proteolytically processed to generate alpha and beta subunits that are linked via disulfide bonds to form the mature receptor. Further processing of the beta subunit results in the formation of the M10 peptide, which has been shown to reduce lung fibrosis. Binding of its ligand, hepatocyte growth factor, induces dimerization and activation of the receptor, which plays a role in cellular survival, embryogenesis, and cellular migration and invasion. Mutations in this gene are associated with papillary renal cell carcinoma, hepatocellular carcinoma, and various head and neck cancers. Amplification and overexpression of this gene are also associated with multiple human cancers. [provided by RefSeq, May 2016].",
         "['Sema domain', 'WD40/YVTN repeat-like-containing domain superfamily', 'Plexin family', 'Sema domain superfamily', 'Protein kinase domain', 'Serine-threonine/tyrosine-protein kinase, catalytic domain', 'Plexin repeat', 'IPT domain', 'Tyrosine-protein kinase, active site', 'Protein kinase-like domain superfamily', 'Immunoglobulin-like fold', 'Immunoglobulin E-set', 'PSI domain', 'Tyrosine-protein kinase, HGF/MSP receptor', 'Protein kinase, ATP binding site', 'Tyrosine-protein kinase, catalytic domain']"
        ],
        [
         "49",
         "EWS RNA binding protein 1",
         "EWSR1",
         "54",
         "This gene encodes a multifunctional protein that is involved in various cellular processes, including gene expression, cell signaling, and RNA processing and transport. The protein includes an N-terminal transcriptional activation domain and a C-terminal RNA-binding domain. Chromosomal translocations between this gene and various genes encoding transcription factors result in the production of chimeric proteins that are involved in tumorigenesis. These chimeric proteins usually consist of the N-terminal transcriptional activation domain of this protein fused to the C-terminal DNA-binding domain of the transcription factor protein. Mutations in this gene, specifically a t(11;22)(q24;q12) translocation, are known to cause Ewing sarcoma as well as neuroectodermal and various other tumors. Alternative splicing of this gene results in multiple transcript variants. Related pseudogenes have been identified on chromosomes 1 and 14. [provided by RefSeq, Jul 2009].",
         "['RNA recognition motif domain', 'Zinc finger, RanBP2-type', 'Nucleotide-binding alpha-beta plait domain superfamily', 'EWS, RNA recognition motif', 'TAF15/EWS/TLS family', 'RNA-binding domain superfamily', 'Zinc finger, RanBP2-type superfamily']"
        ]
       ],
       "shape": {
        "columns": 5,
        "rows": 590
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fullName</th>\n",
       "      <th>name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>summary</th>\n",
       "      <th>Protein_Domains</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALK receptor tyrosine kinase</td>\n",
       "      <td>ALK</td>\n",
       "      <td>1</td>\n",
       "      <td>This gene encodes a receptor tyrosine kinase, ...</td>\n",
       "      <td>[Protein kinase domain, MAM domain, Serine-thr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT serine/threonine kinase 1</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>2</td>\n",
       "      <td>This gene encodes one of the three members of ...</td>\n",
       "      <td>[Protein kinase domain, AGC-kinase, C-terminal...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A-Raf proto-oncogene, serine/threonine kinase</td>\n",
       "      <td>ARAF</td>\n",
       "      <td>3</td>\n",
       "      <td>This proto-oncogene belongs to the RAF subfami...</td>\n",
       "      <td>[Protein kinase C-like, phorbol ester/diacylgl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL proto-oncogene 1, non-receptor tyrosine ki...</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>4</td>\n",
       "      <td>This gene is a protooncogene that encodes a pr...</td>\n",
       "      <td>[Protein kinase domain, SH2 domain, Serine-thr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>B-Raf proto-oncogene, serine/threonine kinase</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>5</td>\n",
       "      <td>This gene encodes a protein belonging to the R...</td>\n",
       "      <td>[Protein kinase domain, Serine-threonine/tyros...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>killin, p53 regulated DNA replication inhibitor</td>\n",
       "      <td>KLLN</td>\n",
       "      <td>32083</td>\n",
       "      <td>The protein encoded by this intronless gene is...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>myocyte enhancer factor 2B</td>\n",
       "      <td>MEF2B</td>\n",
       "      <td>33873</td>\n",
       "      <td>The product of this gene is a member of the MA...</td>\n",
       "      <td>[Transcription factor, MADS-box, MADS MEF2-lik...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>prostate cancer associated non-coding RNA 1</td>\n",
       "      <td>PRNCR1</td>\n",
       "      <td>41522</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>cytochrome c oxidase subunit II</td>\n",
       "      <td>COX2</td>\n",
       "      <td>56400</td>\n",
       "      <td>Contributes to cytochrome-c oxidase activity. ...</td>\n",
       "      <td>[Copper centre Cu(A), Cytochrome c oxidase sub...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>NADH dehydrogenase, subunit 1 (complex I)</td>\n",
       "      <td>ND1</td>\n",
       "      <td>56415</td>\n",
       "      <td>Enables NADH dehydrogenase (ubiquinone) activi...</td>\n",
       "      <td>[NADH:ubiquinone oxidoreductase, subunit 1/F42...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>590 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              fullName    name  gene_id  \\\n",
       "0                         ALK receptor tyrosine kinase     ALK        1   \n",
       "1                        AKT serine/threonine kinase 1    AKT1        2   \n",
       "2        A-Raf proto-oncogene, serine/threonine kinase    ARAF        3   \n",
       "3    ABL proto-oncogene 1, non-receptor tyrosine ki...    ABL1        4   \n",
       "4        B-Raf proto-oncogene, serine/threonine kinase    BRAF        5   \n",
       "..                                                 ...     ...      ...   \n",
       "585    killin, p53 regulated DNA replication inhibitor    KLLN    32083   \n",
       "586                         myocyte enhancer factor 2B   MEF2B    33873   \n",
       "587        prostate cancer associated non-coding RNA 1  PRNCR1    41522   \n",
       "588                    cytochrome c oxidase subunit II    COX2    56400   \n",
       "589          NADH dehydrogenase, subunit 1 (complex I)     ND1    56415   \n",
       "\n",
       "                                               summary  \\\n",
       "0    This gene encodes a receptor tyrosine kinase, ...   \n",
       "1    This gene encodes one of the three members of ...   \n",
       "2    This proto-oncogene belongs to the RAF subfami...   \n",
       "3    This gene is a protooncogene that encodes a pr...   \n",
       "4    This gene encodes a protein belonging to the R...   \n",
       "..                                                 ...   \n",
       "585  The protein encoded by this intronless gene is...   \n",
       "586  The product of this gene is a member of the MA...   \n",
       "587                                               None   \n",
       "588  Contributes to cytochrome-c oxidase activity. ...   \n",
       "589  Enables NADH dehydrogenase (ubiquinone) activi...   \n",
       "\n",
       "                                       Protein_Domains  \n",
       "0    [Protein kinase domain, MAM domain, Serine-thr...  \n",
       "1    [Protein kinase domain, AGC-kinase, C-terminal...  \n",
       "2    [Protein kinase C-like, phorbol ester/diacylgl...  \n",
       "3    [Protein kinase domain, SH2 domain, Serine-thr...  \n",
       "4    [Protein kinase domain, Serine-threonine/tyros...  \n",
       "..                                                 ...  \n",
       "585                                                 []  \n",
       "586  [Transcription factor, MADS-box, MADS MEF2-lik...  \n",
       "587                                                 []  \n",
       "588  [Copper centre Cu(A), Cytochrome c oxidase sub...  \n",
       "589  [NADH:ubiquinone oxidoreductase, subunit 1/F42...  \n",
       "\n",
       "[590 rows x 5 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_info_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_info_df.to_csv(path + '\\\\' + 'civic_gene_info.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
